From dfb53a64b57b005f9c3ab4991fcfc24b435137a3 Mon Sep 17 00:00:00 2001
From: Jaroslaw Zabiello <hipertarcker@gmail.com>
Date: Thu, 1 May 2008 00:06:50 +0100
Subject: [PATCH] Fixed 'package' task.

---
 railties/Rakefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/Rakefile b/railties/Rakefile
index ef673c2..8ea8709 100644
--- a/railties/Rakefile
+++ b/railties/Rakefile
@@ -255,7 +255,7 @@ task :generate_rails_framework_doc do
 end
 
 task :generate_app_doc do
-  File.cp "doc/README_FOR_APP", "#{PKG_DESTINATION}/doc/README_FOR_APP"
+  FileUtils.cp "doc/README_FOR_APP", "#{PKG_DESTINATION}/doc/README_FOR_APP"
   system %{cd #{PKG_DESTINATION}; rake doc:app}
 end
 
-- 
1.5.4.5

