From 7326162401eb736a22931832069025ad8ed79bca Mon Sep 17 00:00:00 2001
From: Gaspard Bucher <gaspard@teti.ch>
Date: Fri, 13 Feb 2009 23:20:06 +0100
Subject: [PATCH] Mysql connection should reflect ActiveRecord's default_timezone.

---
 .../connection_adapters/mysql_adapter.rb           |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
index 9300df2..de5f7c9 100644
--- a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
@@ -584,6 +584,9 @@ module ActiveRecord
         def configure_connection
           encoding = @config[:encoding]
           execute("SET NAMES '#{encoding}'") if encoding
+          
+          tz = ActiveRecord::Base.default_timezone == :utc ? "+0:0" : "SYSTEM"
+          execute("SET time_zone = '#{tz}'")
 
           # By default, MySQL 'where id is null' selects the last inserted id.
           # Turn this off. http://dev.rubyonrails.org/ticket/6778
-- 
1.6.0

