From d09a89006928a75cbeb747a98b2f77fe302866da Mon Sep 17 00:00:00 2001
From: Joe Van Dyk <joe@fixieconsulting.com>
Date: Sat, 26 Sep 2009 16:35:33 -0700
Subject: [PATCH] There is no 'test_sqlite' task.  There is, however, a 'test_sqlite3' task

---
 activerecord/RUNNING_UNIT_TESTS |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/RUNNING_UNIT_TESTS b/activerecord/RUNNING_UNIT_TESTS
index 39fc867..8559d72 100644
--- a/activerecord/RUNNING_UNIT_TESTS
+++ b/activerecord/RUNNING_UNIT_TESTS
@@ -14,7 +14,7 @@ connection.rb otherwise (on Postgres, at least) tests for default values will fa
 
 The easiest way to run the unit tests is through Rake. The default task runs
 the entire test suite for all the adapters. You can also run the suite on just
-one adapter by using the tasks test_mysql, test_sqlite, test_postgresql or any
+one adapter by using the tasks test_mysql, test_sqlite3, test_postgresql or any
 of the other test_ tasks. For more information, checkout the full array of rake
 tasks with "rake -T"
 
-- 
1.6.3.2

