From c124679e209f35faa1ee926a416ff5eedc63b825 Mon Sep 17 00:00:00 2001
From: Aaron <aaronchi@gmail.com>
Date: Thu, 18 Feb 2010 09:21:01 -0800
Subject: [PATCH] fix called_from under windows

---
 railties/lib/rails/engine.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/rails/engine.rb b/railties/lib/rails/engine.rb
index 53e54d7..a158bb7 100644
--- a/railties/lib/rails/engine.rb
+++ b/railties/lib/rails/engine.rb
@@ -15,7 +15,7 @@ module Rails
       def inherited(base)
         unless abstract_railtie?(base)
           base.called_from = begin
-            call_stack = caller.map { |p| p.split(':').first }
+            call_stack = caller.map { |p| p.split(':')[0..-2].join(':') }
             File.dirname(call_stack.detect { |p| p !~ %r[railties[\w\-]*/lib/rails|rack[\w\-]*/lib/rack] })
           end
         end
-- 
1.6.5.1.1367.gcd48

