diff --git a/activerecord/lib/active_record/reflection.rb b/activerecord/lib/active_record/reflection.rb
index dbff4f2..9d81ff1 100644
--- a/activerecord/lib/active_record/reflection.rb
+++ b/activerecord/lib/active_record/reflection.rb
@@ -175,11 +175,11 @@ module ActiveRecord
       end
 
       def table_name
-        @table_name ||= klass.table_name
+        klass.table_name
       end
 
       def quoted_table_name
-        @quoted_table_name ||= klass.quoted_table_name
+        klass.quoted_table_name
       end
 
       def primary_key_name
