From b0e905d02c41994cc636bee1242853891b3395ab Mon Sep 17 00:00:00 2001
From: Ryan Bigg <radarlistener@gmail.com>
Date: Fri, 9 Apr 2010 20:42:38 +1000
Subject: [PATCH] If we're going to call it beta3 everywhere else, may as well do it in here too.

---
 version.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/version.rb b/version.rb
index 7c47cbe..c108761 100644
--- a/version.rb
+++ b/version.rb
@@ -3,7 +3,7 @@ module Rails
     MAJOR = 3
     MINOR = 0
     TINY  = 0
-    BUILD = "beta2"
+    BUILD = "beta3"
 
     STRING = [MAJOR, MINOR, TINY, BUILD].join('.')
   end
-- 
1.6.4.1

