From 59ae8c486f1d61f453be87edd62b8de0f93d1b58 Mon Sep 17 00:00:00 2001
From: Chris Lloyd <christopher.lloyd@gmail.com>
Date: Thu, 7 Aug 2008 11:00:18 +1000
Subject: [PATCH] `rake doc:plugins' now uses UTF-8 charset.

---
 railties/lib/tasks/documentation.rake |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/railties/lib/tasks/documentation.rake b/railties/lib/tasks/documentation.rake
index 331b245..f4927a0 100644
--- a/railties/lib/tasks/documentation.rake
+++ b/railties/lib/tasks/documentation.rake
@@ -62,6 +62,7 @@ namespace :doc do
         options << "-o doc/plugins/#{plugin}"
         options << "--title '#{plugin.titlecase} Plugin Documentation'"
         options << '--line-numbers' << '--inline-source'
+        options << '--charset' << 'utf-8'
         options << '-T html'
 
         files.include("#{plugin_base}/lib/**/*.rb")
-- 
1.5.6.1

