From 67db843501af5c8c67c931d2faf6854aedb85bd1 Mon Sep 17 00:00:00 2001
From: Stephen Touset <stephen@touset.org>
Date: Thu, 12 Mar 2009 16:27:33 -0400
Subject: [PATCH] Load StringInquirer in Initializer#env

---
 railties/lib/initializer.rb |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index edea4e5..ca487dd 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -53,7 +53,10 @@ module Rails
     end
 
     def env
-      @_env ||= ActiveSupport::StringInquirer.new(RAILS_ENV)
+      @_env ||= begin
+        require 'active_support/string_inquirer'
+        ActiveSupport::StringInquirer.new(RAILS_ENV)
+      end
     end
 
     def cache
-- 
1.6.1

