From 3215029c0f74d76b6630ca66fa3e927ecd95a4da Mon Sep 17 00:00:00 2001
From: Christopher Currie <christopher@sixslicestudios.com>
Date: Thu, 30 Oct 2008 16:02:54 -0700
Subject: [PATCH] Don't rewrite ruby shebang paths for BIN_FILES

---
 railties/Rakefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/Rakefile b/railties/Rakefile
index 872ea83..3e78f35 100644
--- a/railties/Rakefile
+++ b/railties/Rakefile
@@ -209,7 +209,7 @@ end
 task :copy_binfiles do
   BIN_FILES.each do |file|
     dest_file = File.join(PKG_DESTINATION, 'script', file)
-    copy_with_rewritten_ruby_path(File.join('bin', file), dest_file)
+    cp File.join('bin', file), dest_file
     chmod 0755, dest_file
   end
 end
-- 
1.5.4.3

