From c1924d5dec168a5cc97290914bd77f25f4c6afe4 Mon Sep 17 00:00:00 2001
From: ujihisa <ujihisa@gmail.com>
Date: Fri, 8 Oct 2010 12:58:40 -0700
Subject: [PATCH] http://localhost:3000/rails/info/properties shows more details for ease of bug reporting

---
 railties/lib/rails/info.rb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/railties/lib/rails/info.rb b/railties/lib/rails/info.rb
index 6cbd1f2..ca2de3c 100644
--- a/railties/lib/rails/info.rb
+++ b/railties/lib/rails/info.rb
@@ -62,8 +62,8 @@ module Rails
       end
     end
 
-    # The Ruby version and platform, e.g. "1.8.2 (powerpc-darwin8.2.0)".
-    property 'Ruby version', "#{RUBY_VERSION} (#{RUBY_PLATFORM})"
+    # Details of the Ruby, e.g. "ruby 1.9.2p14 (2010-10-02 revision 29393) [x86_64-darwin10.4.1]"
+    property 'Ruby description', RUBY_DESCRIPTION
 
     # The RubyGems version, if it's installed.
     property 'RubyGems version' do
-- 
1.7.3

