diff --git a/activerecord/lib/active_record/base.rb b/activerecord/lib/active_record/base.rb
index 8dd07eb..c6dcd4f 100755
--- a/activerecord/lib/active_record/base.rb
+++ b/activerecord/lib/active_record/base.rb
@@ -377,7 +377,7 @@ module ActiveRecord #:nodoc:
           next
         end
         klass.instance_variables.each { |var| klass.send(:remove_instance_variable, var) }
-        klass.instance_methods(false).each { |m| klass.send :undef_method, m }
+        klass.instance_methods(false).each { |m| klass.send :remove_method, m }
       end
       @@subclasses = {}
       nonreloadables.each { |klass| (@@subclasses[klass.superclass] ||= []) << klass }
