From e76ded9924a755b21bfef0a13d508508a4a25150 Mon Sep 17 00:00:00 2001
From: Pivotal Labs <opensource@pivotallabs.com>
Date: Tue, 31 Mar 2009 09:54:56 -0700
Subject: [PATCH] New instructions for installing mysql ruby bindings since the gem leaks memory

---
 .../connection_adapters/mysql_adapter.rb           |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
index 9300df2..270f983 100644
--- a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
@@ -64,7 +64,7 @@ module ActiveRecord
         begin
           require_library_or_gem('mysql')
         rescue LoadError
-          $stderr.puts '!!! The bundled mysql.rb driver has been removed from Rails 2.2. Please install the mysql gem and try again: gem install mysql.'
+          $stderr.puts '!!! The bundled mysql.rb driver has been removed from Rails 2.2. Please install the ruby mysql bindings from http://www.tmtm.org/en/mysql/ruby/ and try again.'
           raise
         end
       end
-- 
1.6.0.4

