diff --git a/actionpack/lib/action_view/helpers/prototype_helper.rb b/actionpack/lib/action_view/helpers/prototype_helper.rb
index 81938ac..44a68bb 100644
--- a/actionpack/lib/action_view/helpers/prototype_helper.rb
+++ b/actionpack/lib/action_view/helpers/prototype_helper.rb
@@ -1009,9 +1009,14 @@ module ActionView
       
         js_options['asynchronous'] = options[:type] != :synchronous
         js_options['method']       = method_option_to_s(options[:method]) if options[:method]
-        js_options['insertion']    = "Insertion.#{options[:position].to_s.camelize}" if options[:position]
         js_options['evalScripts']  = options[:script].nil? || options[:script]
 
+        if options[:position]
+          js_options['insertion'] = "'#{options[:position]}'"
+        elsif options[:insertion]
+          js_options['insertion'] = options[:insertion]
+        end
+
         if options[:form]
           js_options['parameters'] = 'Form.serialize(this)'
         elsif options[:submit]
