From 696feed4020588ed79b039f549e7f0eb655bd583 Mon Sep 17 00:00:00 2001
From: Jeffrey Chupp <jchupp@politics4all.com>
Date: Sun, 5 Apr 2009 15:39:02 -0500
Subject: [PATCH] quote index_name on postgresql remove_index

---
 .../connection_adapters/postgresql_adapter.rb      |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
index 913bb52..c97cd84 100644
--- a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
@@ -839,7 +839,7 @@ module ActiveRecord
 
       # Drops an index from a table.
       def remove_index(table_name, options = {})
-        execute "DROP INDEX #{index_name(table_name, options)}"
+        execute "DROP INDEX #{quote_column_name(index_name(table_name, options))}"
       end
 
       # Maps logical Rails types to PostgreSQL-specific data types.
-- 
1.6.0.4

