From 910c92cd59a823396a3137cda06a4016bea18826 Mon Sep 17 00:00:00 2001
From: Greg Hazel <ghazel@gmail.com>
Date: Wed, 1 Sep 2010 00:56:37 -0700
Subject: [PATCH] remove use of echo [#4410 state:resolved]

---
 activerecord/Rakefile |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/activerecord/Rakefile b/activerecord/Rakefile
index c2d63cd..395c72d 100644
--- a/activerecord/Rakefile
+++ b/activerecord/Rakefile
@@ -75,8 +75,8 @@ end
 namespace :mysql do
   desc 'Build the MySQL test databases'
   task :build_databases do
-    %x( echo "create DATABASE activerecord_unittest DEFAULT CHARACTER SET utf8 DEFAULT COLLATE utf8_unicode_ci " | mysql --user=#{MYSQL_DB_USER})
-    %x( echo "create DATABASE activerecord_unittest2 DEFAULT CHARACTER SET utf8 DEFAULT COLLATE utf8_unicode_ci " | mysql --user=#{MYSQL_DB_USER})
+    %x( mysql --user=#{MYSQL_DB_USER} -e "create DATABASE activerecord_unittest DEFAULT CHARACTER SET utf8 DEFAULT COLLATE utf8_unicode_ci ")
+    %x( mysql --user=#{MYSQL_DB_USER} -e "create DATABASE activerecord_unittest2 DEFAULT CHARACTER SET utf8 DEFAULT COLLATE utf8_unicode_ci ")
   end
 
   desc 'Drop the MySQL test databases'
-- 
1.7.1.msysgit.0

