From 1b52c703da899430fff5569a93d5806931e7e397 Mon Sep 17 00:00:00 2001
From: Hongli Lai (Phusion) <hongli@phusion.nl>
Date: Thu, 1 May 2008 20:47:09 +0200
Subject: [PATCH] Prevent AssetTagHelper from crashing if RAILS_ROOT is not defined. Fixes compatibility with Passenger.

---
 railties/lib/initializer.rb |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index ea61d4e..09a98d5 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -28,7 +28,11 @@ module Rails
     end
   
     def root
-      RAILS_ROOT
+      if defined?(RAILS_ROOT)
+        RAILS_ROOT
+      else
+        nil
+      end
     end
   
     def env
@@ -40,7 +44,7 @@ module Rails
     end
 
     def public_path
-      @@public_path ||= File.join(self.root, "public")
+      @@public_path ||= self.root ? File.join(self.root, "public") : "public"
     end
 
     def public_path=(path)
-- 
1.5.4

