From 523a178292ae11578493fe0757f448281a7e22aa Mon Sep 17 00:00:00 2001
From: Sam Granieri <sam@samgranieri.com>
Date: Thu, 24 Jul 2008 13:25:35 -0500
Subject: [PATCH] Fixed email template loading

---
 railties/lib/initializer.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index 44863ab..48d20b3 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -341,7 +341,7 @@ Run `rake gems:install` to install the missing gems.
 
     def load_view_paths
       ActionView::PathSet::Path.eager_load_templates! if configuration.cache_classes
-      ActionMailer::Base.template_root.load
+      ActionMailer::Base.template_root.load if configuration.frameworks.include?(:action_mailer)
       ActionController::Base.view_paths.load
     end
 
-- 
1.5.6.4

