From 17400206cbe68fd92e406b75c3cd0f2b72a7b71a Mon Sep 17 00:00:00 2001
From: Philip R. Dutton <fooledbyprimes@gmail.com>
Date: Sun, 10 Oct 2010 22:31:30 -0400
Subject: [PATCH] I corrected a code snippet typo in the railties/guides/source/initialization.textile file [#5779 state:resolved]

---
 railties/guides/source/initialization.textile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/guides/source/initialization.textile b/railties/guides/source/initialization.textile
index 599ddcc..3e02bc0 100644
--- a/railties/guides/source/initialization.textile
+++ b/railties/guides/source/initialization.textile
@@ -3070,7 +3070,7 @@ The +Rails::Plugin::Configuration+ class may be a bit difficult to find at first
 
 <ruby>
   module Rails
-    class Plugin < Railtie
+    class Plugin < Engine
       ...
     end
   end
-- 
1.7.0.4

