From 701049a035e1fcb478199dbe6942f23fabca3fe5 Mon Sep 17 00:00:00 2001
From: Bryan Ray <bryansray@gmail.com>
Date: Sat, 12 Jul 2008 15:03:57 -0500
Subject: [PATCH] Added some options for sqlite3.

---
 railties/lib/commands/dbconsole.rb |   16 +++++++++++++++-
 1 files changed, 15 insertions(+), 1 deletions(-)

diff --git a/railties/lib/commands/dbconsole.rb b/railties/lib/commands/dbconsole.rb
index 17acb7b..3080344 100644
--- a/railties/lib/commands/dbconsole.rb
+++ b/railties/lib/commands/dbconsole.rb
@@ -3,12 +3,21 @@ require 'yaml'
 require 'optparse'
 
 include_password = false
+options = {}
 
 OptionParser.new do |opt|
   opt.banner = "Usage: dbconsole [options] [environment]"
   opt.on("-p", "--include-password", "Automatically provide the database from database.yml") do |v|
     include_password = true
   end
+  opt.on("--mode [MODE]", ['html', 'list', 'line', 'column'], 
+        "Automatically put the sqlite3 database in a specified mode (html, list, line, column).") do |mode|
+    options['mode'] = mode
+  end
+  opt.on("-h", "--header") do |h|
+    options['header'] = h
+  end
+  
   opt.parse!(ARGV)
   abort opt.to_s unless (0..1).include?(ARGV.size)
 end
@@ -58,8 +67,13 @@ when "sqlite"
   exec(find_cmd('sqlite'), config["database"])
 
 when "sqlite3"
-  exec(find_cmd('sqlite3'), config["database"])
+  args = []
+  
+  args << "-#{options['mode']}" if options['mode']
+  args << "-header" if options['header']
+  args << config['database']
 
+  exec(find_cmd('sqlite3'), *args)
 else
   abort "Unknown command-line client for #{config['database']}. Submit a Rails patch to add support!"
 end
-- 
1.5.6

