From 19d911aedaabbeeb7d3505b4085f726f5830bbe8 Mon Sep 17 00:00:00 2001
From: Charles Nutter <charles.nutter@sun.com>
Date: Sat, 28 Feb 2009 17:47:00 -0600
Subject: [PATCH] Move away from using exec to launch console.

---
 railties/lib/commands/console.rb |   18 +++++++++---------
 1 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/railties/lib/commands/console.rb b/railties/lib/commands/console.rb
index 63df834..c9a7018 100644
--- a/railties/lib/commands/console.rb
+++ b/railties/lib/commands/console.rb
@@ -1,6 +1,5 @@
-irb = RUBY_PLATFORM =~ /(:?mswin|mingw)/ ? 'irb.bat' : 'irb'
-
 require 'optparse'
+require 'irb'
 
 options = { :sandbox => false, :irb => irb }
 OptionParser.new do |opt|
@@ -11,16 +10,15 @@ OptionParser.new do |opt|
   opt.parse!(ARGV)
 end
 
-libs =  " -r irb/completion"
-libs << %( -r "#{RAILS_ROOT}/config/environment")
-libs << " -r console_app"
-libs << " -r console_sandbox" if options[:sandbox]
-libs << " -r console_with_helpers"
+require "irb/completion"
+require "#{RAILS_ROOT}/config/environment"
+require "console_app"
+require "console_sandbox" if options[:sandbox]
+require "console_with_helpers"
 
 if options[:debugger]
   begin
     require 'ruby-debug'
-    libs << " -r ruby-debug"
     puts "=> Debugger enabled"
   rescue Exception
     puts "You need to install ruby-debug to run the console in debugging mode. With gems, use 'gem install ruby-debug'"
@@ -42,4 +40,6 @@ if options[:sandbox]
 else
   puts "Loading #{ENV['RAILS_ENV']} environment (Rails #{Rails.version})"
 end
-exec "#{options[:irb]} #{libs} --simple-prompt"
+
+ARGV.replace(["--simple-prompt"])
+IRB.start
-- 
1.6.0.2

