From e97f583f3bed2437587434dc68eb32f10dd125ab Mon Sep 17 00:00:00 2001
From: George Ogata <george.ogata@gmail.com>
Date: Wed, 18 Mar 2009 00:55:06 +1100
Subject: [PATCH] Make script/server use a different pid file for each port.

---
 railties/lib/commands/server.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/commands/server.rb b/railties/lib/commands/server.rb
index ebe34a4..98d8132 100644
--- a/railties/lib/commands/server.rb
+++ b/railties/lib/commands/server.rb
@@ -60,7 +60,7 @@ end
 
 if options[:detach]
   Process.daemon
-  pid = "#{RAILS_ROOT}/tmp/pids/server.pid"
+  pid = "#{RAILS_ROOT}/tmp/pids/server.#{options[:Port]}.pid"
   File.open(pid, 'w'){ |f| f.write(Process.pid) }
   at_exit { File.delete(pid) if File.exist?(pid) }
 end
-- 
1.5.6.5

