diff --git a/actionpack/lib/action_view/helpers/number_helper.rb b/actionpack/lib/action_view/helpers/number_helper.rb
index 05a9c5b..55c980f 100644
--- a/actionpack/lib/action_view/helpers/number_helper.rb
+++ b/actionpack/lib/action_view/helpers/number_helper.rb
@@ -115,7 +115,9 @@ module ActionView
         currency  = I18n.translate(:'number.currency.format', :locale => options[:locale], :default => {})
 
         defaults  = DEFAULT_CURRENCY_VALUES.merge(defaults).merge!(currency)
-        defaults[:negative_format] = "-" + options[:format] if options[:format]
+        if (format = options[:format] || currency[:format]) && !currency[:negative_format]
+          defaults[:negative_format] = "-" + format
+        end
         options   = defaults.merge!(options)
 
         unit      = options.delete(:unit)
