From f3d196e2d4ce73595a58a06b7e8e877e2245d2a2 Mon Sep 17 00:00:00 2001
From: ZhangJinzhu <wosmvp@gmail.com>
Date: Sat, 6 Mar 2010 20:19:15 +0800
Subject: [PATCH] Read version from RAILS_VERSION, so 'rake install' will be run correctly

---
 Rakefile |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/Rakefile b/Rakefile
index cc12696..d8274fa 100644
--- a/Rakefile
+++ b/Rakefile
@@ -49,12 +49,14 @@ end
 
 desc "Install gems for all projects."
 task :install => :gem do
+  version = File.read("RAILS_VERSION").strip
+
   (PROJECTS - ["railties"]).each do |project|
     puts "INSTALLING #{project}"
-    system("gem install #{project}/pkg/#{project}-#{ActionPack::VERSION::STRING}.gem --no-ri --no-rdoc")
+    system("gem install #{project}/pkg/#{project}-#{version}.gem --no-ri --no-rdoc")
   end
-  system("gem install railties/pkg/railties-#{ActionPack::VERSION::STRING}.gem --no-ri --no-rdoc")
-  system("gem install pkg/rails-#{ActionPack::VERSION::STRING}.gem --no-ri --no-rdoc")
+  system("gem install railties/pkg/railties-#{version}.gem --no-ri --no-rdoc")
+  system("gem install pkg/rails-#{version}.gem --no-ri --no-rdoc")
 end
 
 desc "Generate documentation for the Rails framework"
-- 
1.7.0

