diff --git a/railties/lib/rails/configuration.rb b/railties/lib/rails/configuration.rb
index 59132ef..523e8c2 100644
--- a/railties/lib/rails/configuration.rb
+++ b/railties/lib/rails/configuration.rb
@@ -76,7 +76,7 @@ module Rails
       @paths.config.locales      = "config/locales"
       @paths.config.environments = "config/environments"
 
-      @paths.app.controllers.push *builtin_directories
+      builtin_directories.each {|dir| @paths.app.controllers.push dir}
 
       @paths.app.load_path!
       @paths.app.metals.load_path!
@@ -264,4 +264,4 @@ module Rails
       @reload_plugins
     end
   end
-end
\ No newline at end of file
+end
