diff --git a/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb b/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
index 59a51c0..87e46b6 100644
--- a/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
@@ -124,6 +124,14 @@ module ActiveRecord
       def quote_column_name(name) #:nodoc:
         %Q("#{name}")
       end
+      
+      def quoted_true
+        "'True'"
+      end
+      
+      def quoted_false
+        "'False'"
+      end
 
 
       # DATABASE STATEMENTS ======================================
