From b31c5d19702cdd0cfef02a9709c849d1938f089b Mon Sep 17 00:00:00 2001
From: Nick Hoffman <nick@deadorange.com>
Date: Tue, 10 Mar 2009 12:41:08 -0400
Subject: [PATCH] Before loading the database configuration and establishing a database connection, Rails now checks that the database configuration file exists.

---
 railties/lib/initializer.rb |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index edea4e5..cc019de 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -406,11 +406,12 @@ Run `rake gems:install` to install the missing gems.
     end
 
     # This initialization routine does nothing unless <tt>:active_record</tt>
-    # is one of the frameworks to load (Configuration#frameworks). If it is,
+    # is one of the frameworks to load (Configuration#frameworks), and the
+    # database configuration file exists. If both requirements are satisfied,
     # this sets the database configuration from Configuration#database_configuration
     # and then establishes the connection.
     def initialize_database
-      if configuration.frameworks.include?(:active_record)
+      if configuration.frameworks.include?(:active_record) and File.exist?(configuration.database_configuration_file)
         ActiveRecord::Base.configurations = configuration.database_configuration
         ActiveRecord::Base.establish_connection
       end
-- 
1.6.0.4

