From a56f53e445bece8ce2f2329bae80549bd9b6f6ba Mon Sep 17 00:00:00 2001
From: Matt Jones <al2o3cr@gmail.com>
Date: Thu, 10 Dec 2009 13:00:26 -0500
Subject: [PATCH] specify correct Rack version in actionpack gemspec

---
 actionpack/Rakefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/actionpack/Rakefile b/actionpack/Rakefile
index 73f9efe..b2546fa 100644
--- a/actionpack/Rakefile
+++ b/actionpack/Rakefile
@@ -80,7 +80,7 @@ spec = Gem::Specification.new do |s|
   s.requirements << 'none'
 
   s.add_dependency('activesupport', '= 2.3.5' + PKG_BUILD)
-  s.add_dependency('rack', '~> 1.0.0')
+  s.add_dependency('rack', '~> 1.0.1')
 
   s.require_path = 'lib'
   s.autorequire = 'action_controller'
-- 
1.5.3.1

