From c3ea0f81cfc2ebe49b0b062b0bd3fa15d37496a7 Mon Sep 17 00:00:00 2001
From: Chris Moore <cdmwebs@gmail.com>
Date: Wed, 26 Nov 2008 11:17:52 -0500
Subject: [PATCH] create the db directory if it doesn't exist (rake db:create)

---
 railties/lib/tasks/databases.rake |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index 5cb27f1..facfd32 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -29,6 +29,8 @@ namespace :db do
   def create_database(config)
     begin
       if config['adapter'] =~ /sqlite/
+        FileUtils.mkdir("#{RAILS_ROOT}/db") unless File.exists?("#{RAILS_ROOT}/db")
+
         if File.exist?(config['database'])
           $stderr.puts "#{config['database']} already exists"
         else
-- 
1.6.0.2

