diff -purN --exclude '*.svn' /usr/lib/ruby/gems/1.8/gems/activerecord-2.3.2/lib/active_record/associations/association_collection.rb ./activerecord/lib/active_record/associations/association_collection.rb
--- /usr/lib/ruby/gems/1.8/gems/activerecord-2.3.2/lib/active_record/associations/association_collection.rb	2009-06-10 10:31:24.000000000 -0700
+++ ./activerecord/lib/active_record/associations/association_collection.rb	2009-07-08 12:11:40.000000000 -0700
@@ -112,14 +112,23 @@ module ActiveRecord
         result = true
         load_target if @owner.new_record?
 
-        transaction do
-          flatten_deeper(records).each do |record|
-            raise_on_type_mismatch(record)
-            add_record_to_target_with_callbacks(record) do |r|
-              result &&= insert_record(record) unless @owner.new_record?
-            end
-          end
-        end
+				if @owner.new_record? 
+					flatten_deeper(records).each do |record|
+						raise_on_type_mismatch(record)
+						add_record_to_target_with_callbacks(record) do |r|
+							result &&= insert_record(record) unless @owner.new_record?
+						end
+					end
+				else 
+					transaction do
+						flatten_deeper(records).each do |record|
+							raise_on_type_mismatch(record)
+							add_record_to_target_with_callbacks(record) do |r|
+								result &&= insert_record(record) unless @owner.new_record?
+							end
+						end
+					end
+				end
 
         result && self
       end
