diff --git a/activerecord/test/cases/method_scoping_test.rb b/activerecord/test/cases/method_scoping_test.rb
index af6fcd3..c16b633 100644
--- a/activerecord/test/cases/method_scoping_test.rb
+++ b/activerecord/test/cases/method_scoping_test.rb
@@ -137,6 +137,17 @@ class MethodScopingTest < ActiveRecord::TestCase
     assert_equal 1, scoped_authors.size
     assert_equal authors(:david).attributes, scoped_authors.first.attributes
   end
+  
+  def test_scoped_find_works_with_same_join_and_include
+    scoped_authors = Author.with_scope(:find => { :joins => :posts }) do
+      Author.find(:all, :select => 'DISTINCT authors.*', :include => :posts, :conditions => 'comments.id = 1')
+    end
+    assert scoped_authors.include?(authors(:david))
+    assert !scoped_authors.include?(authors(:mary))
+    assert_equal 1, scoped_authors.size
+    assert_equal authors(:david).attributes, scoped_authors.first.attributes
+  end
+  
 
   def test_scoped_count_include
     # with the include, will retrieve only developers for the given project
