diff --git a/activesupport/lib/active_support.rb b/activesupport/lib/active_support.rb
index f2baa5a..ec23eae 100644
--- a/activesupport/lib/active_support.rb
+++ b/activesupport/lib/active_support.rb
@@ -69,7 +69,7 @@ module ActiveSupport
   end
 end
 
-require 'active_support/vendor'
+require 'active_support/core_ext/kernel/requires'
 
-require 'i18n'
+require_library_or_gem 'i18n'
 I18n.load_path << "#{File.dirname(__FILE__)}/active_support/locale/en.yml"
diff --git a/railties/bin/rails b/railties/bin/rails
index 0f51d57..179b34e 100755
--- a/railties/bin/rails
+++ b/railties/bin/rails
@@ -20,6 +20,7 @@ end
 
 ARGV << "--help" if ARGV.empty?
 
+require 'rubygems'
 require 'rails'
 require 'rails/generators'
 require 'rails/generators/rails/app/app_generator'
