From 590271e61adea271e231579cd68b56a7ca50f675 Mon Sep 17 00:00:00 2001
From: zhengjia <jiazheng@live.com>
Date: Mon, 15 Nov 2010 22:35:41 -0600
Subject: [PATCH] Minor fix on Rails:Railtie documentation

---
 railties/lib/rails/railtie.rb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/railties/lib/rails/railtie.rb b/railties/lib/rails/railtie.rb
index c76bc83..030a838 100644
--- a/railties/lib/rails/railtie.rb
+++ b/railties/lib/rails/railtie.rb
@@ -97,7 +97,7 @@ module Rails
   # If your railtie has rake tasks, you can tell Rails to load them through the method
   # rake tasks:
   #
-  #   class MyRailtie < Railtie
+  #   class MyRailtie < Rails::Railtie
   #     rake_tasks do
   #       load "path/to/my_railtie.tasks"
   #     end
@@ -107,7 +107,7 @@ module Rails
   # your generators at a different location, you can specify in your Railtie a block which
   # will load them during normal generators lookup:
   #
-  #   class MyRailtie < Railtie
+  #   class MyRailtie < Rails::Railtie
   #     generators do
   #       require "path/to/my_railtie_generator"
   #     end
-- 
1.6.4.2

