diff --git a/activerecord/test/cases/method_scoping_test.rb b/activerecord/test/cases/method_scoping_test.rb
index 6372b4f..39e0429 100644
--- a/activerecord/test/cases/method_scoping_test.rb
+++ b/activerecord/test/cases/method_scoping_test.rb
@@ -27,6 +27,23 @@ class MethodScopingTest < ActiveRecord::TestCase
     end
   end
 
+  def test_scoped_find_last
+    highest_salary = Developer.find(:first, :order => "salary DESC")
+    Developer.with_scope(:find => { :order => "salary" }) do
+      assert_equal highest_salary, Developer.find(:last)
+    end
+  end
+
+  def test_scoped_find_last_preserves_scope
+    lowest_salary = Developer.find(:first, :order => "salary ASC")
+    highest_salary = Developer.find(:first, :order => "salary DESC")
+    Developer.with_scope(:find => { :order => "salary" }) do
+      assert_equal highest_salary, Developer.find(:last)
+      # should still work (i.e. the :last call should not have side effects)
+      assert_equal lowest_salary, Developer.find(:first)
+    end
+  end
+
   def test_scoped_find_combines_conditions
     Developer.with_scope(:find => { :conditions => "salary = 9000" }) do
       assert_equal developers(:poor_jamis), Developer.find(:first, :conditions => "name = 'Jamis'")
