From 0d7f46d2c4a92a771102d494bd180919ec2c5429 Mon Sep 17 00:00:00 2001
From: Aaron Patterson <aaron.patterson@gmail.com>
Date: Thu, 8 Jul 2010 11:44:19 -0700
Subject: [PATCH] removing useless code. [#5070 state:resolved]

---
 .../connection_adapters/sqlite3_adapter.rb         |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb b/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb
index 0d9a86a..e4fa412 100644
--- a/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb
@@ -16,8 +16,7 @@ module ActiveRecord
 
       db = SQLite3::Database.new(
         config[:database],
-        :results_as_hash => true,
-        :type_translation => false
+        :results_as_hash => true
       )
 
       db.busy_timeout(config[:timeout]) unless config[:timeout].nil?
-- 
1.7.0.6

