From 24b998b0be416d64a8598632d86184c74f3edae4 Mon Sep 17 00:00:00 2001
From: David Reese <david@whatcould.com>
Date: Thu, 6 Aug 2009 16:06:30 -0500
Subject: [PATCH] Make BlankSlate play nice with Ruby 1.9#methods

---
 .../vendor/builder-2.1.2/blankslate.rb             |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/activesupport/lib/active_support/vendor/builder-2.1.2/blankslate.rb b/activesupport/lib/active_support/vendor/builder-2.1.2/blankslate.rb
index da6034d..d0939ea 100644
--- a/activesupport/lib/active_support/vendor/builder-2.1.2/blankslate.rb
+++ b/activesupport/lib/active_support/vendor/builder-2.1.2/blankslate.rb
@@ -20,10 +20,11 @@ class BlankSlate
     # Hide the method named +name+ in the BlankSlate class.  Don't
     # hide +instance_eval+ or any method beginning with "__".
     def hide(name)
-      if instance_methods.include?(name.to_s) and
-        name !~ /^(__|instance_eval)/
+      method_name = RUBY_VERSION < '1.9' ? name.to_s : name.to_sym
+      if instance_methods.include?(method_name) and
+        name !~ /^(__|instance_eval|object_id)/
         @hidden_methods ||= {}
-        @hidden_methods[name.to_sym] = instance_method(name)
+        @hidden_methods[name.to_sym] = instance_method(method_name)
         undef_method name
       end
     end
-- 
1.6.0

