From f7089664355e7437178eaa2568f1c55d3ef946d9 Mon Sep 17 00:00:00 2001
From: Priit Tamboom <priit@mx.ee>
Date: Sun, 16 May 2010 15:11:07 +0200
Subject: [PATCH] Added nokogiri dependency to Gemfile

---
 Gemfile |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/Gemfile b/Gemfile
index 5a8879d..aef091c 100644
--- a/Gemfile
+++ b/Gemfile
@@ -9,6 +9,7 @@ gem "mocha", ">= 0.9.8"
 group :mri do
   gem 'json'
   gem 'yajl-ruby'
+  gem "nokogiri", ">= 1.4.0"
 
   if RUBY_VERSION < '1.9'
     gem "system_timer"
-- 
1.7.0.4

