diff --git a/activerecord/lib/active_record/fixtures.rb b/activerecord/lib/active_record/fixtures.rb
index c650111..8181a88 100644
--- a/activerecord/lib/active_record/fixtures.rb
+++ b/activerecord/lib/active_record/fixtures.rb
@@ -860,9 +860,16 @@ module ActiveRecord
 
       def require_fixture_classes(table_names = nil)
         (table_names || fixture_table_names).each do |table_name|
-          file_name = table_name.to_s
-          file_name = file_name.singularize if ActiveRecord::Base.pluralize_table_names
-          try_to_load_dependency(file_name)
+          #require 'ruby-debug'; debugger if table_name.match(/warehouse/)
+          if klass = self.fixture_class_names[table_name.to_sym]
+            next if klass.is_a?(Class)
+            file_name = klass.to_s.underscore
+          else
+            file_name = table_name.to_s.underscore
+            file_name = file_name.singularize if ActiveRecord::Base.pluralize_table_names
+            klass     = file_name.classify
+          end
+          try_to_load_dependency(file_name) unless Object.const_defined?(klass.to_sym)
         end
       end
 
diff --git a/activerecord/test/cases/fixtures_test.rb b/activerecord/test/cases/fixtures_test.rb
index 252bf4f..4ad35c8 100644
--- a/activerecord/test/cases/fixtures_test.rb
+++ b/activerecord/test/cases/fixtures_test.rb
@@ -544,6 +544,14 @@ class FoxyFixturesTest < ActiveRecord::TestCase
     end
   end
 
+  def test_trys_to_load_dependency_only_if_necessary
+    class_names = {:a => Book, :b => 'Book', :c => 'DoesNotExist'}
+    #Test::Unit::TestCase.stubs(:fixture_class_names).returns(class_names)
+    #Test::Unit::TestCase.expects(:try_to_load_dependency).with(class_names[:c].underscore)
+    require 'ruby-debug'; debugger
+    Test::Unit::TestCase.require_fixture_classes(class_names.keys)
+  end
+
   def test_only_populates_columns_that_exist
     assert_not_nil(pirates(:blackbeard).created_on)
     assert_not_nil(pirates(:blackbeard).updated_on)
