diff --git a/actionpack/test/template/asset_tag_helper_test.rb b/actionpack/test/template/asset_tag_helper_test.rb
index 7e40a55..e197d90 100644
--- a/actionpack/test/template/asset_tag_helper_test.rb
+++ b/actionpack/test/template/asset_tag_helper_test.rb
@@ -224,6 +224,12 @@ class AssetTagHelperTest < ActionView::TestCase
     ImageLinkToTag.each { |method, tag| assert_dom_equal(tag, eval(method)) }
   end
 
+  def test_image_tag_with_asset_host
+    ENV["RAILS_ASSET_ID"] = ""
+    ActionController::Base.asset_host = 'http://a0.example.com/asd'
+    assert_dom_equal(%(<img alt="Mouse" onmouseover="this.src='http://a0.example.com/asd/images/mouse_over.png'" onmouseout="this.src='http://a0.example.com/asd/images/mouse.png'" src="http://a0.example.com/asd/images/mouse.png" />), image_tag("mouse.png", :mouseover => "mouse_over.png"))
+  end
+
   def test_timebased_asset_id
     expected_time = File.stat(File.expand_path(File.dirname(__FILE__) + "/../fixtures/public/images/rails.png")).mtime.to_i.to_s
     assert_equal %(<img alt="Rails" src="/images/rails.png?#{expected_time}" />), image_tag("rails.png")
