--- lib/mongrel/cgi.rb.orig	2009-01-27 16:10:20.263513800 +0000
+++ lib/mongrel/cgi.rb	2010-06-10 11:39:07.664959500 +0100
@@ -104,8 +104,13 @@
           cookie.each {|c| to['Set-Cookie'] = c.to_s }
         when Hash
           cookie.each_value {|c| to['Set-Cookie'] = c.to_s}
+        when String
+          cookies = cookie.split(/\n+/)
+          cookies.each do |c|
+            to['Set-Cookie'] = c
+          end
         else
-          to['Set-Cookie'] = options['cookie'].to_s
+          to['Set-Cookie'] = cookie.to_s
         end
         
         @head.delete('cookie')
