diff --git a/activesupport/lib/active_support/memoizable.rb b/activesupport/lib/active_support/memoizable.rb
index cd5c01c..6f3434d 100644
--- a/activesupport/lib/active_support/memoizable.rb
+++ b/activesupport/lib/active_support/memoizable.rb
@@ -39,6 +39,16 @@ module ActiveSupport
           end
         end
       end
+
+      def unmemoize(*symbols)
+        memoized_methods = methods.grep(/^_unmemoized/)
+        symbols.each do |symbol|
+          if memoized_methods.include?("#{symbol}")
+            ivar = MEMOIZED_IVAR.call(symbol)
+            instance_variable_get(ivar).clear if instance_variable_defined?(ivar)
+          end
+        end
+      end
     end
 
     def memoize(*symbols)
