From 54fd6a2d1adac545432d866f5fbd8a8e7bec7641 Mon Sep 17 00:00:00 2001
From: Tim Connor <timocratic@gmail.com>
Date: Thu, 21 May 2009 14:51:34 -0700
Subject: [PATCH] fix gem task dependency crash

---
 railties/lib/initializer.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index 3c0d594..2927c6f 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -304,7 +304,7 @@ module Rails
     end
 
     def load_gems
-      unless $gems_build_rake_task
+      unless $gems_rake_task
         @configuration.gems.each { |gem| gem.load }
       end
     end
-- 
1.6.1

