From a446ce697db6cfb53a0bdba5e935961b7e84888e Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Jakub=20Ku=C5=BAma?= <qoobaa@gmail.com>
Date: Sat, 22 Nov 2008 18:23:49 +0100
Subject: [PATCH] Removing SQLite3 columns with custom types

---
 .../connection_adapters/sqlite_adapter.rb          |   12 +++++++++---
 1 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb b/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
index 84f8c02..625e69b 100644
--- a/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
@@ -63,6 +63,12 @@ module ActiveRecord
           end
         end
       end
+
+      private
+
+        def simplified_type(field_type)
+          super or field_type
+        end
     end
 
     # The SQLite adapter works with both the 2.x and 3.x series of SQLite with the sqlite-ruby drivers (available both as gems and
@@ -83,7 +89,7 @@ module ActiveRecord
       def requires_reloading?
         true
       end
- 
+
       def disconnect!
         super
         @connection.close rescue nil
@@ -217,7 +223,7 @@ module ActiveRecord
         if @connection.respond_to?(:transaction_active?) && @connection.transaction_active?
           raise StatementInvalid, 'Cannot add columns to a SQLite database while inside a transaction'
         end
-        
+
         super(table_name, column_name, type, options)
         # See last paragraph on http://www.sqlite.org/lang_altertable.html
         execute "VACUUM"
@@ -314,7 +320,7 @@ module ActiveRecord
                 (options[:rename][column.name] ||
                  options[:rename][column.name.to_sym] ||
                  column.name) : column.name
-              
+
               @definition.column(column_name, column.type,
                 :limit => column.limit, :default => column.default,
                 :null => column.null)
-- 
1.5.6.3

