From f431d24fd1cbe06f90c0b5588f0c903f4d1d614a Mon Sep 17 00:00:00 2001
From: Chris Kampmeier <chris@kampers.net>
Date: Sun, 3 May 2009 18:09:06 -0700
Subject: [PATCH] Don't use #tap before Active Support is available, since older versions of ruby don't have native implementations

---
 activerecord/lib/active_record.rb |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/activerecord/lib/active_record.rb b/activerecord/lib/active_record.rb
index 1982ffc..06d6c87 100644
--- a/activerecord/lib/active_record.rb
+++ b/activerecord/lib/active_record.rb
@@ -21,9 +21,8 @@
 # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #++
 
-"#{File.dirname(__FILE__)}/../../activesupport/lib".tap do |path|
-  $:.unshift(path) if File.directory?(path)
-end
+activesupport_path = "#{File.dirname(__FILE__)}/../../activesupport/lib"
+$:.unshift(activesupport_path) if File.directory?(activesupport_path)
 require 'active_support'
 require 'active_support/core/all'
 
-- 
1.6.1.3

