From d1862a12ae951c2781c06d66fd0998465906aae8 Mon Sep 17 00:00:00 2001
From: Mike Breen <hardbap@gmail.com>
Date: Mon, 20 Jul 2009 16:48:08 -0400
Subject: [PATCH] Add option to routes task to target a specific controller with CONTROLLER=x.

---
 railties/lib/tasks/routes.rake |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/railties/lib/tasks/routes.rake b/railties/lib/tasks/routes.rake
index 39b7139..abbf325 100644
--- a/railties/lib/tasks/routes.rake
+++ b/railties/lib/tasks/routes.rake
@@ -1,6 +1,7 @@
-desc 'Print out all defined routes in match order, with names.'
+desc 'Print out all defined routes in match order, with names. Target specific controller with CONTROLLER=x.'
 task :routes => :environment do
-  routes = ActionController::Routing::Routes.routes.collect do |route|
+  all_routes = ENV['CONTROLLER'] ? ActionController::Routing::Routes.routes.select { |route| route.defaults[:controller] == ENV['CONTROLLER'] } : ActionController::Routing::Routes.routes
+  routes = all_routes.collect do |route|
     name = ActionController::Routing::Routes.named_routes.routes.index(route).to_s
     verb = route.conditions[:method].to_s.upcase
     segs = route.segments.inject("") { |str,s| str << s.to_s }
@@ -14,4 +15,4 @@ task :routes => :environment do
   routes.each do |r|
     puts "#{r[:name].rjust(name_width)} #{r[:verb].ljust(verb_width)} #{r[:segs].ljust(segs_width)} #{r[:reqs]}"
   end
-end
\ No newline at end of file
+end
-- 
1.6.3

