From 27e7b450e6add6e465cf7da3842a581ce31e0b98 Mon Sep 17 00:00:00 2001
From: Will Bryant <will.bryant@gmail.com>
Date: Wed, 19 Nov 2008 22:44:54 +1300
Subject: [PATCH] re-enable db:schema:dump so db:migrate:redo dumps the schema after migrating back up

---
 railties/lib/tasks/databases.rake |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index 5cb27f1..96ec8a5 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -245,6 +245,7 @@ namespace :db do
       File.open(ENV['SCHEMA'] || "#{RAILS_ROOT}/db/schema.rb", "w") do |file|
         ActiveRecord::SchemaDumper.dump(ActiveRecord::Base.connection, file)
       end
+      Rake::Task["db:schema:dump"].reenable
     end
 
     desc "Load a schema.rb file into the database"
-- 
1.5.5.1

