--- activerecord-3.0.3/lib/active_record/associations/has_many_association.rb.orig	2010-12-15 19:36:47.000000000 -0200
+++ activerecord-3.0.3/lib/active_record/associations/has_many_association.rb	2010-12-15 19:35:27.000000000 -0200
@@ -13,7 +13,9 @@
       protected
         def owner_quoted_id
           if @reflection.options[:primary_key]
-            quote_value(@owner.send(@reflection.options[:primary_key]))
+            @owner.class.quote_value(@owner.send(@reflection.options[:primary_key]))
+          elsif @reflection.source_reflection && @reflection.source_reflection.options[:primary_key]
+            @owner.class.quote_value(@owner.send(@reflection.source_reflection.options[:primary_key]))
           else
             @owner.quoted_id
           end
