From 9f7f64b7ea9d7fd4e2f8951e36edfe06baf191db Mon Sep 17 00:00:00 2001
From: Jaroslaw Zabiello <hipertracker@gmail.com>
Date: Wed, 28 May 2008 00:03:14 +0100
Subject: [PATCH] Added install task for Rake.

---
 Rakefile |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/Rakefile b/Rakefile
index 11d205a..8715378 100644
--- a/Rakefile
+++ b/Rakefile
@@ -19,3 +19,12 @@ task :default => :test
     end
   end
 end
+
+desc "Install all projects" 
+task 'install' do
+  Rake::Task['package'].invoke   
+  sudo = PLATFORM =~ /win32/ ? '' : 'sudo'
+  Dir.glob("*/pkg/*.gem").each do |g| 
+    system "#{sudo} gem install -f -l #{g}"
+  end                                    
+end
-- 
1.5.5.1

