From e2cedf286f0a4770d26f22e0c4139a50c06f0d62 Mon Sep 17 00:00:00 2001
From: Chad Woolley <thewoolleyman@gmail.com>
Date: Sat, 29 Aug 2009 23:35:30 -0700
Subject: [PATCH] CI config updates: do not send CI emails unless explicitly enabled, use 'gem update --system', and send emails from an address which can post to the core list

---
 ci/cruise_config.rb |   10 +++++++---
 1 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/ci/cruise_config.rb b/ci/cruise_config.rb
index 15a1bd6..9c7fa98 100644
--- a/ci/cruise_config.rb
+++ b/ci/cruise_config.rb
@@ -1,5 +1,9 @@
 Project.configure do |project|
- project.build_command = 'ruby ci/ci_build.rb'
- project.email_notifier.emails = ['rubyonrails-core@googlegroups.com', 'thewoolleyman@gmail.com','michael@koziarski.com', 'david@loudthinking.com', 'jeremy@bitsweat.net', 'josh@joshpeek.com', 'pratiknaik@gmail.com', 'wycats@gmail.com']
- project.email_notifier.from = 'thewoolleyman+railsci@gmail.com'
+  # Send email notifications about broken and fixed builds to core mailing list
+  if Socket.gethostname =~ /ci.rubyonrails.org/ && ENV['ENABLE_RAILS_CI_EMAILS'] == 'true'
+    project.email_notifier.emails = ['rubyonrails-core@googlegroups.com']
+  end
+
+  project.build_command = 'sudo gem update --system && ruby ci/ci_build.rb'
+  project.email_notifier.from = 'thewoolleyman@gmail.com'
 end
-- 
1.6.3.3

