From 334d1403f5bd565d4476ac816742bc7858a744b8 Mon Sep 17 00:00:00 2001
From: Neil Middleton <neil@wagonlabs.com>
Date: Thu, 12 Aug 2010 17:39:27 +0100
Subject: [PATCH] Added change that demonstrates issue in LH#5365

---
 activerecord/test/models/reply.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/test/models/reply.rb b/activerecord/test/models/reply.rb
index 110d540..52ad961 100644
--- a/activerecord/test/models/reply.rb
+++ b/activerecord/test/models/reply.rb
@@ -3,7 +3,7 @@ require 'models/topic'
 class Reply < Topic
   scope :base
 
-  belongs_to :topic, :foreign_key => "parent_id", :counter_cache => true
+  belongs_to :topic, :foreign_key => "parent_id", :counter_cache => true, :touch => true 
   belongs_to :topic_with_primary_key, :class_name => "Topic", :primary_key => "title", :foreign_key => "parent_title", :counter_cache => "replies_count"
   has_many :replies, :class_name => "SillyReply", :dependent => :destroy, :foreign_key => "parent_id"
 
-- 
1.7.0
