From 31c3a3961fad3417b35ff5b93fb7840fc8b7ff05 Mon Sep 17 00:00:00 2001
From: Scott Taylor <scott@railsnewbie.com>
Date: Wed, 16 Jun 2010 21:56:19 -0400
Subject: [PATCH] add missing shebang to rails bin.  LH [#4885 state:resolved]

---
 bin/rails |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/bin/rails b/bin/rails
index 4a74726..a628cc3 100755
--- a/bin/rails
+++ b/bin/rails
@@ -1,3 +1,5 @@
+#!/usr/bin/env ruby
+
 begin
   require "rails/cli"
 rescue LoadError
-- 
1.6.5.2

