From b49589d8bfc4ee92717f7318d65b980f4840ec17 Mon Sep 17 00:00:00 2001
From: Usuario de Programacion <pairprog@pairprog-10.(none)>
Date: Mon, 9 Mar 2009 13:57:04 +0100
Subject: [PATCH] I made a small change so that escape characters work properly in xml with postgres

---
 .../connection_adapters/postgresql_adapter.rb      |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
index 913bb52..cffc70a 100644
--- a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
@@ -358,7 +358,7 @@ module ActiveRecord
         if value.kind_of?(String) && column && column.type == :binary
           "#{quoted_string_prefix}'#{escape_bytea(value)}'"
         elsif value.kind_of?(String) && column && column.sql_type =~ /^xml$/
-          "xml '#{quote_string(value)}'"
+          "xml E'#{quote_string(value)}'"
         elsif value.kind_of?(Numeric) && column && column.sql_type =~ /^money$/
           # Not truly string input, so doesn't require (or allow) escape string syntax.
           "'#{value.to_s}'"
-- 
1.5.6.3

