From 62a8ceb9945e579eab34fd7a699fea529e0bd0f3 Mon Sep 17 00:00:00 2001
From: Dieter Komendera <dieter@komendera.com>
Date: Mon, 21 Jul 2008 10:34:50 +0200
Subject: [PATCH] Make rake rails:freeze:edge download files directly from github.com

---
 railties/lib/tasks/framework.rake |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/railties/lib/tasks/framework.rake b/railties/lib/tasks/framework.rake
index 71aea09..b6bd6b2 100644
--- a/railties/lib/tasks/framework.rake
+++ b/railties/lib/tasks/framework.rake
@@ -38,17 +38,16 @@ namespace :rails do
       end
     end
 
-    desc 'Lock to latest Edge Rails, for a specific release use RELEASE=1.2.0'
+    desc 'Lock to latest Edge Rails, for a specific release use RELEASE=v2.1.0'
     task :edge do
       require 'open-uri'
-      version = ENV["RELEASE"] || "edge"
-      target  = "rails_#{version}.zip"
-      url     = "http://dev.rubyonrails.org/archives/#{target}"
+      version = ENV["RELEASE"] || "master"
+      url     = "http://github.com/rails/rails/zipball/#{version}"
 
       chdir 'vendor' do
         puts "Downloading Rails from #{url}"
         File.open('rails.zip', 'wb') do |dst|
-          open url do |src|
+          open(url) do |src|
             while chunk = src.read(4096)
               dst << chunk
             end
@@ -58,6 +57,7 @@ namespace :rails do
         puts 'Unpacking Rails'
         rm_rf 'rails'
         `unzip rails.zip`
+        mv Dir.glob('rails-rails-*').first, 'rails'
         %w(rails.zip rails/Rakefile rails/cleanlogs.sh rails/pushgems.rb rails/release.rb).each do |goner|
           rm_f goner
         end
-- 
1.5.5.3

