diff --git a/Rakefile b/Rakefile
index 3bb0e85..d3133a3 100644
--- a/Rakefile
+++ b/Rakefile
@@ -8,3 +8,9 @@ require 'rake/testtask'
 require 'rake/rdoctask'
 
 require 'tasks/rails'
+
+# temporary work around for
+# http://rails.lighthouseapp.com/projects/8994/tickets/59-when-loading-a-plugin-via-rubygems-rake-tasks-aren-t-included
+File.read("#{RAILS_ROOT}/config/environment.rb").scan(/config.gem \"(.+)\".*:version => \"([\d\.]+)\"/) do |m|
+  Dir["#{Gem.path}/gems/#{m.first}-#{m.last}/tasks/**/*.rake"].sort.each { |ext| load ext }
+end
