Index: response.rb
===================================================================
--- response.rb	(revision 1055)
+++ response.rb	(revision 1056)
@@ -31,7 +31,7 @@
   #    end
   #  end
   class Response < Rack::Response
-    DEFAULT_HEADERS = { "Cache-Control" => "no-cache" }
+    DEFAULT_HEADERS = {  }
     attr_accessor :request
 
     attr_accessor :session, :assigns, :template, :layout
@@ -199,7 +199,7 @@
       end
 
       def set_conditional_cache_control!
-        if headers['Cache-Control'] == DEFAULT_HEADERS['Cache-Control']
+        if session.send(:loaded?) && headers['Cache-Control'] == DEFAULT_HEADERS['Cache-Control']
           headers['Cache-Control'] = 'private, max-age=0, must-revalidate'
         end
       end

