diff --git a/actionpack/lib/action_dispatch/middleware/session/abstract_store.rb b/actionpack/lib/action_dispatch/middleware/session/abstract_store.rb
index 6d109f4..22db949 100644
--- a/actionpack/lib/action_dispatch/middleware/session/abstract_store.rb
+++ b/actionpack/lib/action_dispatch/middleware/session/abstract_store.rb
@@ -155,7 +155,7 @@ module ActionDispatch
         if !session_data.is_a?(AbstractStore::SessionHash) || session_data.send(:loaded?) || options[:expire_after]
           if session_data.is_a?(AbstractStore::SessionHash)
             session_data.send(:load!) if !session_data.send(:loaded?)
-            return response if !session_data.send(:updated?)
+            # return response if !session_data.send(:updated?)
           end
 
           sid = options[:id] || generate_sid
