From 0c0e0b407ac8640409105baf41ad387ac8a9e932 Mon Sep 17 00:00:00 2001
From: Luke Chadwick <luke.a.chadwick@gmail.com>
Date: Tue, 11 May 2010 08:56:31 +1000
Subject: [PATCH] removed code in railties/lib/rails/info.rb that no longer worked because of the change to gem bundler

---
 railties/lib/rails/info.rb |   11 +++--------
 1 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/railties/lib/rails/info.rb b/railties/lib/rails/info.rb
index 5a496f6..8079982 100644
--- a/railties/lib/rails/info.rb
+++ b/railties/lib/rails/info.rb
@@ -36,17 +36,12 @@ module Rails
       end
 
       def edge_rails_revision(info = git_info)
-        info[/commit ([a-z0-9-]+)/, 1] || freeze_edge_version
+        info[/commit ([a-z0-9-]+)/, 1] || "unknown"
       end
 
       def freeze_edge_version
-        if File.exist?(rails_vendor_root)
-          begin
-            Dir[File.join(rails_vendor_root, 'REVISION_*')].first.scan(/_(\d+)$/).first.first
-          rescue
-            Dir[File.join(rails_vendor_root, 'TAG_*')].first.scan(/_(.+)$/).first.first rescue 'unknown'
-          end
-        end
+        Rails.logger.warn 'The freeze:edge command has been deprecated, specify the path setting in your app Gemfile instead and bundle install'
+        return "unknown"
       end
 
       def to_s
-- 
1.7.0.6

