diff --git a/activesupport/lib/active_support/json/encoders/object.rb b/activesupport/lib/active_support/json/encoders/object.rb
index 79d556d..c33039b 100644
--- a/activesupport/lib/active_support/json/encoders/object.rb
+++ b/activesupport/lib/active_support/json/encoders/object.rb
@@ -4,5 +4,7 @@ class Object
     ActiveSupport::JSON.encode(instance_values, options)
   end
 
-  alias to_json rails_to_json
+  def to_json
+    rails_to_json
+  end
 end
