From ad1d4b1e1eba157ba0cc7f1cc8c0a25ebd71db33 Mon Sep 17 00:00:00 2001
From: Santiago Pastorino <santiago@wyeworks.com>
Date: Mon, 29 Mar 2010 00:23:02 -0300
Subject: [PATCH] ruby-debug19 added to Gemfile [#4290 state:committed]

---
 Gemfile |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/Gemfile b/Gemfile
index ea9842c..951137b 100644
--- a/Gemfile
+++ b/Gemfile
@@ -11,6 +11,8 @@ group :mri do
   if RUBY_VERSION < '1.9'
     gem "system_timer"
     gem "ruby-debug", ">= 0.10.3"
+  else
+    gem "ruby-debug19"
   end
 end
 
-- 
1.7.0

