diff --git a/railties/lib/rails/generators/actions.rb b/railties/lib/rails/generators/actions.rb
index e39fa3f..7dec4d4 100644
--- a/railties/lib/rails/generators/actions.rb
+++ b/railties/lib/rails/generators/actions.rb
@@ -89,12 +89,12 @@ module Rails
       #
       # ==== Example
       #
-      #   source "http://gems.github.com/"
+      #   add_source "http://gems.github.com/"
       def add_source(source, options={})
         log :source, source
 
         in_root do
-          prepend_file "Gemfile", "source #{source.inspect}", :verbose => false
+          prepend_file "Gemfile", "source #{source.inspect}\n", :verbose => false
         end
       end
 
