From eaac2210fa4d3c496911d6b1d0e1d853f479d738 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C5=81ukasz=20Strza=C5=82kowski?= <lukasz.strzalkowski@gmail.com>
Date: Tue, 13 Jul 2010 09:57:15 +0200
Subject: [PATCH] Bump bundler to 1.0.0.beta.5

---
 rails.gemspec |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/rails.gemspec b/rails.gemspec
index 2d10ce7..e3f8119 100644
--- a/rails.gemspec
+++ b/rails.gemspec
@@ -25,5 +25,5 @@ Gem::Specification.new do |s|
   s.add_dependency('activeresource', version)
   s.add_dependency('actionmailer',   version)
   s.add_dependency('railties',       version)
-  s.add_dependency('bundler',        '>= 1.0.0.beta.3')
+  s.add_dependency('bundler',        '>= 1.0.0.beta.5')
 end
-- 
1.7.1

