From 2fa6b192709ad3a513af0764f649a48c3648d2cf Mon Sep 17 00:00:00 2001
From: Logan Bowers <logan@datacurrent.com>
Date: Thu, 29 Apr 2010 11:24:10 -0700
Subject: [PATCH] Fix an precision issue when parsing timestamps [#4498]

---
 .../abstract/schema_definitions.rb                 |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb b/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb
index 520f3c8..2a2e38a 100644
--- a/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb
+++ b/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb
@@ -170,7 +170,7 @@ module ActiveRecord
           # '0.123456' -> 123456
           # '1.123456' -> 123456
           def microseconds(time)
-            ((time[:sec_fraction].to_f % 1) * 1_000_000).to_i
+            (time[:sec_fraction] * 1_000_000).to_i
           end
 
           def new_date(year, mon, mday)
-- 
1.6.6.1

