From 49cb23496f12206f17c9bf4c38e6e25511af92a7 Mon Sep 17 00:00:00 2001
From: Chirag Viradiya <chirag.viradiya@gmail.com>
Date: Tue, 23 Nov 2010 22:04:24 +0530
Subject: [PATCH] Fixed improper quoting of column names in postgresql adapter [#6046 state:resolved]

---
 .../connection_adapters/postgresql_adapter.rb      |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
index ccc5085..dc8ba5a 100644
--- a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
@@ -382,7 +382,11 @@ module ActiveRecord
 
       # Quotes column names for use in SQL queries.
       def quote_column_name(name) #:nodoc:
-        PGconn.quote_ident(name.to_s)
+        if name.is_a?(Array)
+          name.collect { |n| PGconn.quote_ident(n.to_s) }.join(',')          
+        else
+          PGconn.quote_ident(name.to_s)
+        end        
       end
 
       # Quote date/time values for use in SQL input. Includes microseconds
-- 
1.7.3.1.msysgit.0

