diff --git a/actionpack/lib/action_view/helpers/asset_tag_helpers/asset_include_tag.rb b/actionpack/lib/action_view/helpers/asset_tag_helpers/asset_include_tag.rb
index e527970..73bbcbc 100644
--- a/actionpack/lib/action_view/helpers/asset_tag_helpers/asset_include_tag.rb
+++ b/actionpack/lib/action_view/helpers/asset_tag_helpers/asset_include_tag.rb
@@ -10,9 +10,6 @@ module ActionView
       class AssetIncludeTag
         attr_reader :config, :asset_paths
 
-        class_attribute :expansions
-        self.expansions = { }
-
         def initialize(config, asset_paths)
           @config = config
           @asset_paths = asset_paths
diff --git a/actionpack/lib/action_view/helpers/asset_tag_helpers/javascript_tag_helpers.rb b/actionpack/lib/action_view/helpers/asset_tag_helpers/javascript_tag_helpers.rb
index 6581e1d..4b35041 100644
--- a/actionpack/lib/action_view/helpers/asset_tag_helpers/javascript_tag_helpers.rb
+++ b/actionpack/lib/action_view/helpers/asset_tag_helpers/javascript_tag_helpers.rb
@@ -10,6 +10,9 @@ module ActionView
       class JavascriptIncludeTag < AssetIncludeTag
         include TagHelper
 
+        class_attribute :expansions
+        self.expansions = { }
+
         def asset_name
           'javascript'
         end
diff --git a/actionpack/lib/action_view/helpers/asset_tag_helpers/stylesheet_tag_helpers.rb b/actionpack/lib/action_view/helpers/asset_tag_helpers/stylesheet_tag_helpers.rb
index d02b28d..7fe1f47 100644
--- a/actionpack/lib/action_view/helpers/asset_tag_helpers/stylesheet_tag_helpers.rb
+++ b/actionpack/lib/action_view/helpers/asset_tag_helpers/stylesheet_tag_helpers.rb
@@ -10,6 +10,9 @@ module ActionView
       class StylesheetIncludeTag < AssetIncludeTag
         include TagHelper
 
+        class_attribute :expansions
+        self.expansions = { }
+
         def asset_name
           'stylesheet'
         end
