diff --git a/railties/lib/rails_generator/generators/applications/app/template_runner.rb b/railties/lib/rails_generator/generators/applications/app/template_runner.rb
index c611364..9dc1314 100644
--- a/railties/lib/rails_generator/generators/applications/app/template_runner.rb
+++ b/railties/lib/rails_generator/generators/applications/app/template_runner.rb
@@ -322,7 +322,7 @@ module Rails
     #
     def ask(string)
       puts string
-      gets.strip
+      STDIN.gets.strip
     end
 
     # Do something in the root of the Rails application or
