From 7b33258cf66adcd3b63c6329c18e5d2f1d182911 Mon Sep 17 00:00:00 2001
From: Juan Lupion <juan.lupion@the-cocktail.com>
Date: Sat, 29 Nov 2008 20:10:14 +0100
Subject: [PATCH] Make rails:update copy any new initializers to config/initializers

---
 railties/lib/tasks/framework.rake |   13 +++++++++++--
 1 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/railties/lib/tasks/framework.rake b/railties/lib/tasks/framework.rake
index df080e9..3ea7389 100644
--- a/railties/lib/tasks/framework.rake
+++ b/railties/lib/tasks/framework.rake
@@ -77,8 +77,8 @@ namespace :rails do
     rm_rf "vendor/rails"
   end
 
-  desc "Update both configs, scripts and public/javascripts from Rails"
-  task :update => [ "update:scripts", "update:javascripts", "update:configs", "update:application_controller" ]
+  desc "Update both configs, scripts, and public/javascripts from Rails"
+  task :update => [ "update:scripts", "update:javascripts", "update:configs", "update:application_controller", "update:initializers" ]
 
   namespace :update do
     desc "Add new scripts to the application script/ directory"
@@ -115,6 +115,15 @@ namespace :rails do
       FileUtils.cp(RAILTIES_PATH + '/environments/boot.rb', RAILS_ROOT + '/config/boot.rb')
     end
     
+    desc "Update new initializers from your current rails install"
+    task :initializers do 
+      require 'railties_path'
+      project_dir = RAILS_ROOT + '/config/initializers/'
+      initializers = Dir[RAILTIES_PATH + '/configs/initializers/*.rb']
+      initializers.reject!{|s| File.exist?(project_dir + File.basename(s))}
+      FileUtils.cp(initializers, project_dir)
+    end
+    
     desc "Rename application.rb to application_controller.rb"
     task :application_controller do
       old_style = RAILS_ROOT + '/app/controllers/application.rb'
-- 
1.5.4.5

