From c822219c0b225930e75e31a6f61c8fc7f78cbc9e Mon Sep 17 00:00:00 2001
From: David Trasbo <me@dtrasbo.com>
Date: Sun, 27 Jun 2010 14:02:34 +0200
Subject: [PATCH] Re-raise Gem::LoadError when missing Rails gem for easier debugging [#2901 state:committed]

---
 railties/environments/boot.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/environments/boot.rb b/railties/environments/boot.rb
index dd5e3b6..68b4075 100644
--- a/railties/environments/boot.rb
+++ b/railties/environments/boot.rb
@@ -63,7 +63,7 @@ module Rails
       end
     rescue Gem::LoadError => load_error
       $stderr.puts %(Missing the Rails #{version} gem. Please `gem install -v=#{version} rails`, update your RAILS_GEM_VERSION setting in config/environment.rb for the Rails version you do have installed, or comment out RAILS_GEM_VERSION to use the latest version installed.)
-      exit 1
+      raise
     end
 
     class << self
-- 
1.7.1

