diff --git a/railties/test/generators/rails_controller_generator_test.rb b/railties/test/generators/rails_controller_generator_test.rb
index 0090d21..4e18b39 100644
--- a/railties/test/generators/rails_controller_generator_test.rb
+++ b/railties/test/generators/rails_controller_generator_test.rb
@@ -17,4 +17,24 @@ class RailsControllerGeneratorTest < GeneratorTestCase
     assert_generated_functional_test_for "admin::products"
     assert_generated_helper_for "admin::products"
   end
+  
+  def test_controller_generates_namespaced_and_not_namespaced_controllers
+      run_generator('controller', %w(products))
+      
+      # We have to require the generated helper to show the problem because
+      # the test helpers just check for generated files and contents but
+      # do not actually load them. But they have to be loaded (as in a real environment)
+      # to make the second generator run fail
+      require "#{RAILS_ROOT}/app/helpers/products_helper"
+
+      assert_nothing_raised do
+        begin
+          run_generator('controller', %w(admin::products))
+        ensure
+          # cleanup
+          Object.send(:remove_const, :ProductsHelper)
+        end
+      end
+  end
+  
 end
