From ab69cd43ebc175231c10f8572de3c90b77f30c7e Mon Sep 17 00:00:00 2001
From: Ben Scofield <git@turrean.com>
Date: Thu, 29 May 2008 11:30:26 -0700
Subject: [PATCH] include rake tasks from frozen gems

---
 railties/lib/tasks/rails.rb |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/railties/lib/tasks/rails.rb b/railties/lib/tasks/rails.rb
index bfcf5bc..2816bd7 100644
--- a/railties/lib/tasks/rails.rb
+++ b/railties/lib/tasks/rails.rb
@@ -6,3 +6,4 @@ Dir["#{File.dirname(__FILE__)}/*.rake"].each { |ext| load ext }
 # Load any custom rakefile extensions
 Dir["#{RAILS_ROOT}/lib/tasks/**/*.rake"].sort.each { |ext| load ext }
 Dir["#{RAILS_ROOT}/vendor/plugins/*/**/tasks/**/*.rake"].sort.each { |ext| load ext }
+Dir["#{RAILS_ROOT}/vendor/gems/*/**/tasks/**/*.rake"].sort.each { |ext| load ext }
-- 
1.5.4.1

