--- actionpack-2.3.2/lib/action_controller/verification.rb.orig	2010-02-15 16:34:33.000000000 +0100
+++ actionpack-2.3.2/lib/action_controller/verification.rb	2010-02-15 18:52:42.000000000 +0100
@@ -80,7 +80,7 @@
       #   array (may also be a single value).
       def verify(options={})
         before_filter :only => options[:only], :except => options[:except] do |c|
-          c.__send__ :verify_action, options
+          c.__send__ :verify_action, Marshal.load(Marshal.dump(options))
         end
       end
     end
