From 3dac72732a69aedfa8411071a6c538e581049e10 Mon Sep 17 00:00:00 2001
From: Rizwan Reza <rizwanreza@gmail.com>
Date: Sat, 16 Jan 2010 11:32:05 +0330
Subject: [PATCH] Better readability of script/generate output without args

---
 railties/lib/rails/generators.rb |   10 +++++++---
 1 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/railties/lib/rails/generators.rb b/railties/lib/rails/generators.rb
index 3713a38..776d22f 100644
--- a/railties/lib/rails/generators.rb
+++ b/railties/lib/rails/generators.rb
@@ -222,9 +222,13 @@ module Rails
       others -= Rails::Generators.builtin
       others.sort!
 
-      puts "Please select a generator."
-      puts "Builtin: #{builtin.join(', ')}."
-      puts "Others: #{others.join(', ')}." unless others.empty?
+      puts "Usage\n  script/generate [generator] [generate-specific options]"
+      puts "\nBuiltin Generators:"
+      puts "  #{builtin.join("\n  ")}"
+      unless others.empty?
+        puts "\nOther Generatorss:"
+        puts "  #{others.join("\n  ")}"
+      end
     end
 
     protected
-- 
1.6.5.2

