diff --git a/activerecord/test/cases/associations_test.rb b/activerecord/test/cases/associations_test.rb
index d99fb44..bdda396 100644
--- a/activerecord/test/cases/associations_test.rb
+++ b/activerecord/test/cases/associations_test.rb

@@ -75,6 +75,17 @@ class AssociationsTest < ActiveRecord::TestCase
     end
   end
 
+  def test_force_reload_is_uncached_for_polymorphic_relation
+    person = Person.new :first_name => 'clark'
+    sponsor = Sponsor.create!(:sponsorable => person)
+    ActiveRecord::Base.cache do
+      sponsor.sponsorable
+      assert_nothing_raised { sponsor.sponsorable(true) }
+      assert_queries(0) { assert_not_nil sponsor.sponsorable }
+      assert_queries(1) { assert_not_nil sponsor.sponsorable(true) }
+    end
+  end
+
 end
 
 class AssociationProxyTest < ActiveRecord::TestCase
