From 53d719d340dcc94ec24793f07a7da0b4ce38201b Mon Sep 17 00:00:00 2001
From: Phil Smith <phil.h.smith@gmail.com>
Date: Mon, 19 Apr 2010 12:43:42 -0700
Subject: [PATCH] Regexp.escape(load_path) in add_routing_namespaces initializer [#4442 state:resolved]

---
 railties/lib/rails/engine.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/rails/engine.rb b/railties/lib/rails/engine.rb
index 0f33b40..98da7e2 100644
--- a/railties/lib/rails/engine.rb
+++ b/railties/lib/rails/engine.rb
@@ -166,7 +166,7 @@ module Rails
       paths.app.controllers.to_a.each do |load_path|
         load_path = File.expand_path(load_path)
         Dir["#{load_path}/*/**/*_controller.rb"].collect do |path|
-          namespace = File.dirname(path).sub(/#{load_path}\/?/, '')
+          namespace = File.dirname(path).sub(/#{Regexp.escape(load_path)}\/?/, '')
           app.routes.controller_namespaces << namespace unless namespace.empty?
         end
       end
-- 
1.6.6

