diff --git a/activesupport/test/json/decoding_test.rb b/activesupport/test/json/decoding_test.rb
index 09fd0d0..aefe4f6 100644
--- a/activesupport/test/json/decoding_test.rb
+++ b/activesupport/test/json/decoding_test.rb
@@ -24,6 +24,7 @@ class TestJSONDecoding < ActiveSupport::TestCase
     %([])    => [],
     %({})    => {},
     %({"a":1})     => {"a" => 1},
+    %("a":"1\\\\")     => {"a" => "1\\"},
     %({"a": ""})    => {"a" => ""},
     %({"a":"\\""}) => {"a" => "\""},
     %({"a": null})  => {"a" => nil},
