From 99464feddc512c061303ea33ef5b6b5c1a9ae7ed Mon Sep 17 00:00:00 2001
From: Ripta Pasay <github@r8y.org>
Date: Wed, 3 Dec 2008 11:37:45 -0500
Subject: [PATCH] Add failing lexical lookup test case

---
 activesupport/test/inflector_test.rb |   15 +++++++++++++++
 1 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/activesupport/test/inflector_test.rb b/activesupport/test/inflector_test.rb
index d8c93dc..84f3a4e 100644
--- a/activesupport/test/inflector_test.rb
+++ b/activesupport/test/inflector_test.rb
@@ -2,8 +2,21 @@ require 'abstract_unit'
 require 'inflector_test_cases'
 
 module Ace
+  module Extension
+    def self.included(base)
+      base.extend(ClassMethods)
+    end
+    
+    module ClassMethods
+      def mission_accomplished?
+        false
+      end
+    end
+  end
+
   module Base
     class Case
+      include Extension
     end
   end
 end
@@ -168,6 +181,8 @@ class InflectorTest < Test::Unit::TestCase
 
   def test_constantize_does_lexical_lookup
     assert_raises(NameError) { ActiveSupport::Inflector.constantize("Ace::Base::InflectorTest") }
+    assert_nothing_raised { Ace::Base::Case::ClassMethods }
+    assert_nothing_raised { assert_equal Ace::Base::Case::ClassMethods, ActiveSupport::Inflector.constantize("Ace::Base::Case::ClassMethods") }
   end
 
   def test_ordinal
-- 
1.5.6.4

