From 49859b0bb1dc41dca33331f198eaf2ccf034a6b4 Mon Sep 17 00:00:00 2001
From: Sven Fuchs <svenfuchs@artweb-design.de>
Date: Mon, 25 Aug 2008 11:48:03 +0200
Subject: [PATCH] I18n: fix activerecord i18n test for classy backend

---
 activerecord/test/cases/i18n_test.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/test/cases/i18n_test.rb b/activerecord/test/cases/i18n_test.rb
index 9f934ab..0603673 100644
--- a/activerecord/test/cases/i18n_test.rb
+++ b/activerecord/test/cases/i18n_test.rb
@@ -40,7 +40,7 @@ class ActiveRecordI18nTests < Test::Unit::TestCase
 
   private
     def reset_translations
-      I18n.backend = I18n::Backend::Simple
+      I18n.backend = I18n::Backend::Simple.new
     end
 end
 
-- 
1.5.3.7

