From 2893a791b5463c0382b44bded273a4a48c422c4b Mon Sep 17 00:00:00 2001
From: Greg DeVore <greg@bluemangolearning.com>
Date: Wed, 18 Mar 2009 21:27:27 -0400
Subject: [PATCH] Updated Rack::MethodOverride so that it would also look in the query params for the _method=put even if there is form data present.

---
 .../vendor/rack-1.0/rack/methodoverride.rb         |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/actionpack/lib/action_controller/vendor/rack-1.0/rack/methodoverride.rb b/actionpack/lib/action_controller/vendor/rack-1.0/rack/methodoverride.rb
index 0eed29f..20c99e9 100644
--- a/actionpack/lib/action_controller/vendor/rack-1.0/rack/methodoverride.rb
+++ b/actionpack/lib/action_controller/vendor/rack-1.0/rack/methodoverride.rb
@@ -13,6 +13,7 @@ module Rack
       if env["REQUEST_METHOD"] == "POST"
         req = Request.new(env)
         method = req.POST[METHOD_OVERRIDE_PARAM_KEY] ||
+          req.GET[METHOD_OVERRIDE_PARAM_KEY] ||
           env[HTTP_METHOD_OVERRIDE_HEADER]
         method = method.to_s.upcase
         if HTTP_METHODS.include?(method)
-- 
1.5.4.4

