From c2b8d1e539426ae93217062bfb82cd3031fa7241 Mon Sep 17 00:00:00 2001
From: Matt Conway <wr0ngway@yahoo.com>
Date: Wed, 22 Apr 2009 22:25:46 -0400
Subject: [PATCH] add options for setting mysql connection timeouts

---
 .../connection_adapters/mysql_adapter.rb           |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
index 9300df2..4542fe6 100644
--- a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
@@ -573,6 +573,10 @@ module ActiveRecord
             @connection.ssl_set(@config[:sslkey], @config[:sslcert], @config[:sslca], @config[:sslcapath], @config[:sslcipher])
           end
 
+          @connection.options(Mysql::OPT_CONNECT_TIMEOUT, @config[:connect_timeout]) if @config[:connect_timeout]
+          @connection.options(Mysql::OPT_READ_TIMEOUT, @config[:read_timeout]) if @config[:read_timeout]
+          @connection.options(Mysql::OPT_WRITE_TIMEOUT, @config[:write_timeout]) if @config[:write_timeout]
+
           @connection.real_connect(*@connection_options)
 
           # reconnect must be set after real_connect is called, because real_connect sets it to false internally
-- 
1.5.6.4

