From f1a1ae9a62dfba6370b0fec390c2a017c712bc03 Mon Sep 17 00:00:00 2001
From: Francesc Esplugas <fesplugas@intraducibles.net>
Date: Tue, 23 Sep 2008 13:25:02 +0200
Subject: [PATCH] Preinitializers folder

---
 railties/environments/boot.rb |   12 ++++++++++--
 1 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/railties/environments/boot.rb b/railties/environments/boot.rb
index 6a30b54..5de6243 100644
--- a/railties/environments/boot.rb
+++ b/railties/environments/boot.rb
@@ -25,12 +25,20 @@ module Rails
     end
 
     def preinitialize
-      load(preinitializer_path) if File.exist?(preinitializer_path)
+      load(preinitializer_file) if File.exist?(preinitializer_file)
+      if File.exist?(preinitializer_path)
+        Dir["#{RAILS_ROOT}/config/preinitializers/**/*.rb"].each { |file| require file }
+      end
     end
 
-    def preinitializer_path
+    def preinitializer_file
       "#{RAILS_ROOT}/config/preinitializer.rb"
     end
+
+    def preinitializer_path
+      "#{RAILS_ROOT}/config/preinitializers"
+    end
+
   end
 
   class Boot
-- 
1.6.0.2

