From 25bd356ceced3b13f5c7d47fbc2acf65a460f106 Mon Sep 17 00:00:00 2001
From: Adam Byrtek <adambyrtek@gmail.com>
Date: Sat, 9 Aug 2008 18:29:46 +0200
Subject: [PATCH] Missing connection management methods implemented for the SQLite adapter.

---
 .../connection_adapters/sqlite_adapter.rb          |   22 +++++++++++++++----
 1 files changed, 17 insertions(+), 5 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb b/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
index 84f8c02..e69fc96 100644
--- a/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
@@ -84,11 +84,6 @@ module ActiveRecord
         true
       end
  
-      def disconnect!
-        super
-        @connection.close rescue nil
-      end
-
       def supports_count_distinct? #:nodoc:
         sqlite_version >= '3.2.6'
       end
@@ -126,6 +121,23 @@ module ActiveRecord
       end
 
 
+      # CONNECTION MANAGEMENT ====================================
+
+      def active?
+        !@connection.closed?
+      end
+
+      def reconnect!
+        unless active?
+          raise ActiveRecord::ActiveRecordError, "Closed SQLite connection cannot be reconnected"
+        end
+      end
+
+      def disconnect!
+        @connection.close rescue nil
+      end
+
+
       # DATABASE STATEMENTS ======================================
 
       def execute(sql, name = nil) #:nodoc:
-- 
1.5.4.4

