From 6cfa42bae7ce041e252a4f114f43fa215f7c3f79 Mon Sep 17 00:00:00 2001
From: Mike Gunderloy <MikeG1@larkfarm.com>
Date: Wed, 4 Feb 2009 08:35:42 -0600
Subject: [PATCH] 1.8.7+ compatibility for info section in CI build file

---
 ci/ci_build.rb |   14 ++++++++++++--
 1 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/ci/ci_build.rb b/ci/ci_build.rb
index 010f78b..e3ec99e 100755
--- a/ci/ci_build.rb
+++ b/ci/ci_build.rb
@@ -98,9 +98,19 @@ puts "[CruiseControl]   #{`mysql --version`}"
 puts "[CruiseControl]   #{`pg_config --version`}"
 puts "[CruiseControl]   SQLite2: #{`sqlite -version`}"
 puts "[CruiseControl]   SQLite3: #{`sqlite3 -version`}"
-`gem env`.each {|line| print "[CruiseControl]   #{line}"}
+gem_env = `gem env`
+if gem_env.respond_to?(:lines)
+  gem_env.lines.each {|line| print "[CruiseControl]   #{line}"}
+else
+  gem_env.each {|line| print "[CruiseControl]   #{line}"}
+end
 puts "[CruiseControl]   Local gems:"
-`gem list`.each {|line| print "[CruiseControl]     #{line}"}
+gem_list = `gem list`
+if gem_list.respond_to?(:lines)
+  gem_list.lines.each {|line| print "[CruiseControl]   #{line}"}
+else
+  gem_list.each {|line| print "[CruiseControl]   #{line}"}
+end
 
 failures = build_results.select { |key, value| value == false }
 
-- 
1.6.1

