=== modified file 'vendor/rails/activerecord/test/cases/autosave_association_test.rb'
--- vendor/rails/activerecord/test/cases/autosave_association_test.rb	2009-04-03 13:23:10 +0000
+++ vendor/rails/activerecord/test/cases/autosave_association_test.rb	2009-04-21 16:08:04 +0000
@@ -431,6 +431,13 @@
     assert !@pirate.ship.marked_for_destruction?
   end
 
+  def test_an_object_marked_for_destruction_should_not_be_destroyed_twice
+    @pirate.ship.mark_for_destruction
+    assert @pirate.save
+    @pirate.ship.expects(:destroy).never
+    assert @pirate.save
+  end
+
   # has_one
   def test_should_destroy_a_child_association_as_part_of_the_save_transaction_if_it_was_marked_for_destroyal
     assert !@pirate.ship.marked_for_destruction?
@@ -898,4 +905,4 @@
   end
 
   include AutosaveAssociationOnACollectionAssociationTests
-end
\ No newline at end of file
+end

