From 579b594d3795bf70dacbdf4f9dc355b36d29c1f5 Mon Sep 17 00:00:00 2001
From: Santiago Pastorino <santiago@wyeworks.com>
Date: Fri, 19 Mar 2010 22:26:57 -0300
Subject: [PATCH] ActionView::SafeBuffer should be there for backwards compatibility [#4241 state:committed]

---
 actionpack/lib/action_view.rb |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/actionpack/lib/action_view.rb b/actionpack/lib/action_view.rb
index e1adb48..9b7d0c4 100644
--- a/actionpack/lib/action_view.rb
+++ b/actionpack/lib/action_view.rb
@@ -53,5 +53,6 @@ end
 
 require 'active_support/core_ext/string/output_safety'
 
+ActionView::SafeBuffer = ActiveSupport::Deprecation::DeprecatedConstantProxy.new('ActionView::SafeBuffer', 'ActiveSupport::SafeBuffer')
 
 I18n.load_path << "#{File.dirname(__FILE__)}/action_view/locale/en.yml"
-- 
1.7.0

