From 65eae9fef6b8251851bf6c57b74d5ba2c2270c58 Mon Sep 17 00:00:00 2001
From: Tekin Suleyman <tekin@tekin.co.uk>
Date: Thu, 29 May 2008 22:19:26 +0100
Subject: [PATCH] Updated plugin install to support scp syntax git repository urls

---
 railties/lib/commands/plugin.rb |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/railties/lib/commands/plugin.rb b/railties/lib/commands/plugin.rb
index 105819c..d2465b3 100644
--- a/railties/lib/commands/plugin.rb
+++ b/railties/lib/commands/plugin.rb
@@ -80,7 +80,7 @@ class RailsEnvironment
   
   def install(name_uri_or_plugin)
     if name_uri_or_plugin.is_a? String
-      if name_uri_or_plugin =~ /:\/\// 
+      if name_uri_or_plugin =~ /\:/ 
         plugin = Plugin.new(name_uri_or_plugin)
       else
         plugin = Plugins[name_uri_or_plugin]
@@ -151,7 +151,7 @@ class Plugin
   end
   
   def self.find(name)
-    name =~ /\// ? new(name) : Repositories.instance.find_plugin(name)
+    name =~ /\:/ ? new(name) : Repositories.instance.find_plugin(name)
   end
   
   def to_s
@@ -280,7 +280,7 @@ class Plugin
     end
 
     def guess_name(url)
-      @name = File.basename(url)
+      @name = File.basename(url.split(':').last)
       if @name == 'trunk' || @name.empty?
         @name = File.basename(File.dirname(url))
       end
-- 
1.5.5.1

