diff --git a/actionpack/test/controller/base_test.rb b/actionpack/test/controller/base_test.rb
index 8416754..7a9bccf 100644
--- a/actionpack/test/controller/base_test.rb
+++ b/actionpack/test/controller/base_test.rb
@@ -167,4 +167,17 @@ class DefaultUrlOptionsTest < Test::Unit::TestCase
   ensure
     ActionController::Routing::Routes.load!
   end
+end
+
+class EnsureNamedRoutesWorksTicket22BugTest < Test::Unit::TestCase
+  def test_named_routes_still_work
+    ActionController::Routing::Routes.draw do |map|
+      map.resources :things
+    end
+    EmptyController.send :include, ActionController::UrlWriter
+
+    assert_equal '/things', EmptyController.new.send(:things_path)
+  ensure
+    ActionController::Routing::Routes.load!
+  end
 end
\ No newline at end of file
