diff -ur activerecord-2.0.2/lib/active_record/connection_adapters/postgresql_adapter.rb activerecord-2.0.2.changed/lib/active_record/connection_adapters/postgresql_adapter.rb
--- activerecord-2.0.2/lib/active_record/connection_adapters/postgresql_adapter.rb	Fri May  2 08:03:39 2008
+++ activerecord-2.0.2.changed/lib/active_record/connection_adapters/postgresql_adapter.rb	Fri May  2 07:48:43 2008
@@ -346,7 +346,7 @@
           # There are some incorrectly compiled postgres drivers out there
           # that don't define PGconn.escape.
           self.class.instance_eval do
-            undef_method(:quote_string)
+            remove_method(:quote_string)
           end
         end
         quote_string(s)
