--- /var/lib/gems/1.8/gems/actionpack-2.3.5/lib/action_controller/integration.rb.org	2009-11-30 12:54:03.298833961 +0100
+++ /var/lib/gems/1.8/gems/actionpack-2.3.5/lib/action_controller/integration.rb	2010-01-18 14:40:59.257623293 +0100
@@ -417,6 +417,12 @@
                 multipart_requestify(value, false).each do |subkey, subvalue|
                   p[k + subkey] = subvalue
                 end
+              elsif Array === value
+                value.each_with_index do |element, i|
+                  multipart_requestify(element, false).each do |subkey, subvalue|
+                    p[k + "[#{i}]" + subkey] = subvalue
+                  end
+                end
               else
                 p[k] = value
               end
