Index: vendor/rails/activerecord/lib/active_record/base.rb
===================================================================
--- vendor/rails/activerecord/lib/active_record/base.rb (revision 25646)
+++ vendor/rails/activerecord/lib/active_record/base.rb (working copy)
@@ -1832,6 +1832,8 @@
                         hash[method][key] = merge_conditions(params[key], hash[method][key])
                       elsif key == :include && merge
                         hash[method][key] = merge_includes(hash[method][key], params[key]).uniq
+                      elsif key == :order && merge
+                        hash[method][key] = params[key] + ", #{hash[method][key]}"
                       else
                         hash[method][key] = hash[method][key] || params[key]
                       end

