From 26c0432fd0ebb81bf3898f9dbba66190a9857df0 Mon Sep 17 00:00:00 2001
From: Diego Carrion <dc.rec1@gmail.com>
Date: Sun, 27 Feb 2011 22:54:02 -0300
Subject: [PATCH] added activesupport/test/tmp to gitignore

---
 .gitignore |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/.gitignore b/.gitignore
index 35402a2..8daa1e4 100644
--- a/.gitignore
+++ b/.gitignore
@@ -10,6 +10,7 @@ activerecord/doc
 actionpack/doc
 actionmailer/doc
 activesupport/doc
+activesupport/test/tmp
 activemodel/test/fixtures/fixture_database.sqlite3
 actionpack/test/tmp
 activesupport/test/fixtures/isolation_test
-- 
1.6.5.2

