From 7c13e71c27f755bfd0a1911f3b3861e3784504fd Mon Sep 17 00:00:00 2001
From: Bruno Michel <bmichel@menfin.info>
Date: Wed, 29 Sep 2010 22:27:40 +0200
Subject: [PATCH] Failing test case for #5735

---
 activesupport/test/dependencies_test.rb |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/activesupport/test/dependencies_test.rb b/activesupport/test/dependencies_test.rb
index bc7f597..53a4042 100644
--- a/activesupport/test/dependencies_test.rb
+++ b/activesupport/test/dependencies_test.rb
@@ -153,6 +153,13 @@ class DependenciesTest < Test::Unit::TestCase
     end
   end
 
+  def test_does_not_find_class_in_the_bad_namespace
+    with_autoloading_fixtures do
+      assert_raise(NameError) { A::E }
+      assert_raise(NameError) { A::B::E }
+    end
+  end
+
   def test_directories_manifest_as_modules_unless_const_defined
     with_autoloading_fixtures do
       assert_kind_of Module, ModuleFolder
-- 
1.7.1

