From 389338ad1c812e871b64819c87773cb5e33a5619 Mon Sep 17 00:00:00 2001
From: Steve Purcell <steve@sanityinc.com>
Date: Tue, 21 Sep 2010 11:37:41 +0100
Subject: [PATCH] Add foreign key on developers_projects.developer_id, illustrating problem in ticket 5674

---
 activerecord/test/schema/schema.rb |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/activerecord/test/schema/schema.rb b/activerecord/test/schema/schema.rb
index ea62833..27e37ce 100644
--- a/activerecord/test/schema/schema.rb
+++ b/activerecord/test/schema/schema.rb
@@ -184,6 +184,10 @@ ActiveRecord::Schema.define do
     t.integer :access_level, :default => 1
   end
 
+  except 'SQLite' do
+    execute "ALTER TABLE developers_projects ADD CONSTRAINT developers_projects_developer_id_fkey FOREIGN KEY (developer_id) REFERENCES developers(id)"
+  end
+
   create_table :edges, :force => true, :id => false do |t|
     t.column :source_id, :integer, :null => false
     t.column :sink_id,   :integer, :null => false
-- 
1.7.3

