From 2a598322fcf2fa8600cdb0d502613dbf3b52d746 Mon Sep 17 00:00:00 2001
From: Dan Croak <dcroak@thoughtbot.com>
Date: Sun, 9 Aug 2009 12:42:43 -0400
Subject: [PATCH] [#3022] 2-3-stable fix tests when rails in a directory with a space

---
 activesupport/test/isolation_test.rb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/activesupport/test/isolation_test.rb b/activesupport/test/isolation_test.rb
index 5a1f285..0d297cd 100644
--- a/activesupport/test/isolation_test.rb
+++ b/activesupport/test/isolation_test.rb
@@ -73,7 +73,7 @@ else
     File.open(File.join(File.dirname(__FILE__), "fixtures", "isolation_test"), "w") {}
 
     ENV["CHILD"] = "1"
-    OUTPUT = `#{Gem.ruby} -I#{File.dirname(__FILE__)} #{File.expand_path(__FILE__)} -v`
+    OUTPUT = `#{Gem.ruby} -I#{File.dirname(__FILE__)} "#{File.expand_path(__FILE__)}" -v`
     ENV.delete("CHILD")
 
     def setup
@@ -153,4 +153,4 @@ else
     end
 
   end
-end
\ No newline at end of file
+end
-- 
1.6.2.2

