Index: activesupport/lib/active_support/json/encoders/date_time.rb
===================================================================
--- activesupport/lib/active_support/json/encoders/date_time.rb	(revision 9250)
+++ activesupport/lib/active_support/json/encoders/date_time.rb	(working copy)
@@ -1,5 +1,5 @@
 class DateTime
   def to_json(options = nil) #:nodoc:
-    %("#{strftime("%Y/%m/%d %H:%M:%S %z")}")
+    strftime('"%Y/%m/%d %H:%M:%S %z"')
   end
 end
