From 3d2f9b93717bd1ab90571511d6ba5d1035367e0f Mon Sep 17 00:00:00 2001
From: Sam Elliott <sam@lenary.co.uk>
Date: Mon, 28 Feb 2011 22:17:38 +0000
Subject: [PATCH] Remove warnings about redefined test methods

---
 activesupport/test/inflector_test.rb |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/activesupport/test/inflector_test.rb b/activesupport/test/inflector_test.rb
index f55116d..1670d9e 100644
--- a/activesupport/test/inflector_test.rb
+++ b/activesupport/test/inflector_test.rb
@@ -51,21 +51,21 @@ class InflectorTest < Test::Unit::TestCase
   end
 
   SingularToPlural.each do |singular, plural|
-    define_method "test_pluralize_#{singular}" do
+    define_method "test_pluralize_singular_#{singular}" do
       assert_equal(plural, ActiveSupport::Inflector.pluralize(singular))
       assert_equal(plural.capitalize, ActiveSupport::Inflector.pluralize(singular.capitalize))
     end
   end
 
   SingularToPlural.each do |singular, plural|
-    define_method "test_singularize_#{plural}" do
+    define_method "test_singularize_plural_#{plural}" do
       assert_equal(singular, ActiveSupport::Inflector.singularize(plural))
       assert_equal(singular.capitalize, ActiveSupport::Inflector.singularize(plural.capitalize))
     end
   end
-  
+
   SingularToPlural.each do |singular, plural|
-    define_method "test_pluralize_#{plural}" do
+    define_method "test_pluralize_plural_#{plural}" do
       assert_equal(plural, ActiveSupport::Inflector.pluralize(plural))
       assert_equal(plural.capitalize, ActiveSupport::Inflector.pluralize(plural.capitalize))
     end
-- 
1.7.4.1

