--- text_helper.rb~	2010-10-04 01:23:59.964357134 +0300
+++ text_helper.rb	2010-10-04 01:25:34.431373632 +0300
@@ -14,9 +14,9 @@
 
       include SanitizeHelper
       # The preferred method of outputting text in your views is to use the
-      # <%= "text" %> eRuby syntax. The regular _puts_ and _print_ methods
+      # \<\%= "text" %\> eRuby syntax. The regular _puts_ and _print_ methods
       # do not operate as expected in an eRuby code block. If you absolutely must
-      # output text within a non-output code block (i.e., <% %>), you can use the concat method.
+      # output text within a non-output code block (i.e., \<\% %\>), you can use the concat method.
       #
       # ==== Examples
       #   <%
