diff --git a/activerecord/lib/active_record/schema_dumper.rb b/activerecord/lib/active_record/schema_dumper.rb
index 557a554..9f45ad0 100644
--- a/activerecord/lib/active_record/schema_dumper.rb
+++ b/activerecord/lib/active_record/schema_dumper.rb
@@ -84,7 +84,7 @@ HEADER
           pk ||= 'id'
 
           tbl.print "  create_table #{table.inspect}"
-          if columns.detect { |c| c.name == pk }
+          if columns.detect { |c| c.name == pk && c.type == :integer }
             if pk != 'id'
               tbl.print %Q(, :primary_key => "#{pk}")
             end
@@ -96,7 +96,7 @@ HEADER
 
           column_specs = columns.map do |column|
             raise StandardError, "Unknown type '#{column.sql_type}' for column '#{column.name}'" if @types[column.type].nil?
-            next if column.name == pk
+            next if column.name == pk && column.type == :integer
             spec = {}
             spec[:name]      = column.name.inspect
             spec[:type]      = column.type.to_s
