From c795215ac19ac46d7d62598f727cea22baaa2e3f Mon Sep 17 00:00:00 2001
From: Jeff Cohen <cohen.jeff@gmail.com>
Date: Mon, 12 Jan 2009 20:33:35 -0600
Subject: [PATCH] Make --freeze option work properly.

---
 railties/bin/rails |    7 +++----
 1 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/railties/bin/rails b/railties/bin/rails
index 6a0c675..bc7b13e 100755
--- a/railties/bin/rails
+++ b/railties/bin/rails
@@ -7,9 +7,8 @@ if %w(--version -v).include? ARGV.first
   exit(0)
 end
 
-freeze   = ARGV.any? { |option| %w(--freeze -f).include?(option) }
-
-app_path = ARGV.first
+freeze_option =  ARGV.index('--freeze') || ARGV.index('-f')
+app_path = ARGV[freeze_option+1] if freeze_option
 
 require File.dirname(__FILE__) + '/../lib/rails_generator'
 
@@ -17,4 +16,4 @@ require 'rails_generator/scripts/generate'
 Rails::Generator::Base.use_application_sources!
 Rails::Generator::Scripts::Generate.new.run(ARGV, :generator => 'app')
 
-Dir.chdir(app_path) { `rake rails:freeze:gems`; puts "froze" } if freeze
\ No newline at end of file
+Dir.chdir(app_path) { `rake rails:freeze:gems`; puts "Successfully frozen to gems." } if freeze_option
\ No newline at end of file
-- 
1.6.0.2

