From dcd4b772f861d7a24042ab47a141318584f1cdea Mon Sep 17 00:00:00 2001
From: Burc Yildirim <burcyildirim@gmail.com>
Date: Sat, 16 Jan 2010 21:49:43 +0200
Subject: [PATCH] Load initializers before routing, otherwise inflections fail with wrong routes.

---
 railties/lib/rails/application.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/rails/application.rb b/railties/lib/rails/application.rb
index c95316a..01aa504 100644
--- a/railties/lib/rails/application.rb
+++ b/railties/lib/rails/application.rb
@@ -104,7 +104,7 @@ module Rails
       @app.call(env)
     end
 
-    initializer :load_application_initializers do
+    initializer :load_application_initializers, :before => "action_controller.initialize_routing" do
       Dir["#{root}/config/initializers/**/*.rb"].sort.each do |initializer|
         load(initializer)
       end
-- 
1.6.5.rc2

