From 95d8ae09c914058afa3f6f8cc2876870fe8629b4 Mon Sep 17 00:00:00 2001
From: Mike Gunderloy <MikeG1@larkfarm.com>
Date: Mon, 29 Dec 2008 20:46:20 -0600
Subject: [PATCH] Add transaction check to SQLite2 adapter

---
 .../connection_adapters/sqlite_adapter.rb          |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb b/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
index 84f8c02..c717a6d 100644
--- a/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
@@ -402,6 +402,9 @@ module ActiveRecord
       end
 
       def add_column(table_name, column_name, type, options = {}) #:nodoc:
+        if @connection.respond_to?(:transaction_active?) && @connection.transaction_active?
+          raise StatementInvalid, 'Cannot add columns to a SQLite database while inside a transaction'
+        end
         alter_table(table_name) do |definition|
           definition.column(column_name, type, options)
         end
-- 
1.5.4.3

