From 94a8767720e1202a884b53cb93f0e49beae0aa3f Mon Sep 17 00:00:00 2001
From: Tim Connor <timocratic@gmail.com>
Date: Thu, 21 May 2009 14:57:30 -0700
Subject: [PATCH] fix gem task dependency crash

---
 railties/lib/initializer.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index 4d34b82..034b3d6 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -303,7 +303,7 @@ module Rails
     end
 
     def load_gems
-      unless $gems_build_rake_task
+      unless $gems_rake_task
         @configuration.gems.each { |gem| gem.load }
       end
     end
-- 
1.6.1

