diff --git a/activerecord/Rakefile b/activerecord/Rakefile
index 50c1201..d6033a9 100755
--- a/activerecord/Rakefile
+++ b/activerecord/Rakefile
@@ -72,8 +72,6 @@ namespace :postgresql do
   task :build_databases do
     %x( createdb activerecord_unittest )
     %x( createdb activerecord_unittest2 )
-    %x( psql activerecord_unittest -f #{File.join(SCHEMA_ROOT, 'postgresql.sql')} )
-    %x( psql activerecord_unittest2 -f #{File.join(SCHEMA_ROOT, 'postgresql2.sql')} )
   end
 
   desc 'Drop the PostgreSQL test databases'
diff --git a/activerecord/test/schema/postgresql_specific_schema.rb b/activerecord/test/schema/postgresql_specific_schema.rb
index 35e9ecf..576a4d0 100644
--- a/activerecord/test/schema/postgresql_specific_schema.rb
+++ b/activerecord/test/schema/postgresql_specific_schema.rb
@@ -2,7 +2,7 @@ ActiveRecord::Schema.define do
 
   %w(postgresql_arrays postgresql_moneys postgresql_numbers postgresql_times postgresql_network_addresses postgresql_bit_strings
       postgresql_oids defaults geometrics).each do |table_name|
-        drop_table table_name
+    execute "DROP TABLE  IF EXISTS #{quote_table_name table_name}"
   end
 
   execute 'DROP SEQUENCE IF EXISTS companies_nonstd_seq CASCADE'
