diff --git a/activesupport/lib/active_support/notifications/fanout.rb b/activesupport/lib/active_support/notifications/fanout.rb
index 300ec84..d3936a3 100644
--- a/activesupport/lib/active_support/notifications/fanout.rb
+++ b/activesupport/lib/active_support/notifications/fanout.rb
@@ -24,11 +24,11 @@ module ActiveSupport
       end
 
       def publish(name, *args)
-        if listeners = @listeners_for[name]
-          listeners.each { |s| s.publish(name, *args) }
-        else
-          @listeners_for[name] = @subscribers.select { |s| s.publish(name, *args) }
-        end
+        listeners_for(name).each { |s| s.publish(name, *args) }
+      end
+
+      def listeners_for(name)
+        @listeners_for[name] ||= @subscribers.select { |s| s.subscribed_to?(name) }
       end
 
       # This is a sync queue, so there is not waiting.
@@ -60,9 +60,7 @@ module ActiveSupport
         end
 
         def publish(*args)
-          return unless subscribed_to?(args.first)
           push(*args)
-          true
         end
 
         def drained?
