From a5e915ac14e40c3a5c19e3bd4ff6f7e2358f0b1b Mon Sep 17 00:00:00 2001
From: Blythe Dunham <blythe@snowgiraffe.com>
Date: Thu, 24 Dec 2009 13:52:50 -0800
Subject: [PATCH] ActiveRecord Tests cannot load arel

---
 activerecord/lib/active_record.rb |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/activerecord/lib/active_record.rb b/activerecord/lib/active_record.rb
index 196b87c..bde195d 100644
--- a/activerecord/lib/active_record.rb
+++ b/activerecord/lib/active_record.rb
@@ -28,6 +28,9 @@ $:.unshift(activesupport_path) if File.directory?(activesupport_path) && !$:.inc
 activemodel_path = File.expand_path('../../../activemodel/lib', __FILE__)
 $:.unshift(activemodel_path) if File.directory?(activemodel_path) && !$:.include?(activemodel_path)
 
+arel_path = File.expand_path('../../../arel/lib', __FILE__)
+$:.unshift(arel_path) if File.directory?(arel_path) && !$:.include?(arel_path)
+
 require 'active_support'
 require 'active_model'
 require 'arel'
-- 
1.6.5

