--- /var/ruby/1.8/gem_home/gems/activesupport-2.2.2/lib/active_support/inflector.rb     2009-03-10 07:02:36.710001758 -0700
+++ /var/ruby/1.8/gem_home/gems/activesupport-2.3.1/lib/active_support/inflector.rb     2009-03-10 06:49:24.155519095 -0700
@@ -275,9 +275,16 @@
       Iconv.iconv('ascii//ignore//translit', 'utf-8', string).to_s
     end
 
+    if RUBY_VERSION >= '1.9'
+      undef_method :transliterate
+      def transliterate(string)
+        warn "Ruby 1.9 doesn't support Unicode normalization yet"
+        string.dup
+      end
+
     # The iconv transliteration code doesn't function correctly
     # on some platforms, but it's very fast where it does function.
-    if "foo" != Inflector.transliterate("föö")
+    elsif "foo" != (Inflector.transliterate("föö") rescue nil)
       undef_method :transliterate
       def transliterate(string)
         string.mb_chars.normalize(:kd). # Decompose accented characters