From 2f8e625f3fdae310df4f7da99ef1532ba29124f1 Mon Sep 17 00:00:00 2001
From: moro <moronatural@gmail.com>
Date: Fri, 27 Feb 2009 12:40:17 +0900
Subject: [PATCH] I want script/console to invoke used Ruby's irb.

---
 railties/lib/commands/console.rb |   13 ++++++++++++-
 1 files changed, 12 insertions(+), 1 deletions(-)

diff --git a/railties/lib/commands/console.rb b/railties/lib/commands/console.rb
index 63df834..9f5e8ad 100644
--- a/railties/lib/commands/console.rb
+++ b/railties/lib/commands/console.rb
@@ -1,4 +1,15 @@
-irb = RUBY_PLATFORM =~ /(:?mswin|mingw)/ ? 'irb.bat' : 'irb'
+require 'rbconfig'
+
+def irb(base = "irb", config = RbConfig::CONFIG)
+  irb_name = [
+    config["configure_args"].scan(/--program-prefix=(\w+)/),
+    base,
+    config["configure_args"].scan(/--program-suffix=(\w+)/),
+    (RUBY_PLATFORM =~ /(:?mswin|mingw)/) ? '.bat' : config["EXEEXT"]
+  ].flatten.compact.join
+
+  File.expand_path(irb_name, config["bindir"])
+end
 
 require 'optparse'
 
-- 
1.6.0

