diff --git a/actionpack/test/template/form_helper_test.rb b/actionpack/test/template/form_helper_test.rb
index 654eee4..f8a59f4 100644
--- a/actionpack/test/template/form_helper_test.rb
+++ b/actionpack/test/template/form_helper_test.rb
@@ -98,6 +98,41 @@ class FormHelperTest < ActionView::TestCase
     assert_dom_equal('<label for="my_for">Title</label>', label(:post, :title, nil, "for" => "my_for"))
   end
 
+  def test_select
+    expected =  '<select id="post_title" name="post[title]">' +
+                '<option value="OPT2">OPTion Two</option>' + "\n" +
+                '<option value="OPT1">OPTion One</option>' +
+                '</select>'
+    assert_dom_equal(expected, select(:post, :title, 
+                {  'OPTion Two' => 'OPT2', 'OPTion One' => 'OPT1'  }
+                                     ))
+  end
+
+  def test_select_with_blank
+    expected =  '<select id="post_title" name="post[title]">' +
+                '<option value=""></option>' + "\n" +
+                '<option value="OPT2">OPTion Two</option>' + "\n" +
+                '<option value="OPT1">OPTion One</option>' +
+                '</select>'
+    assert_dom_equal(expected, select(:post, :title, 
+                {  'OPTion Two' => 'OPT2', 'OPTion One' => 'OPT1'  },
+                :include_blank => true
+                                     ))
+  end
+
+  def test_select_with_blank_and_css_id
+    expected =  '<select id="my_id" name="post[title]">' +
+                '<option value=""></option>' + "\n" +
+                '<option value="OPT2">OPTion Two</option>' + "\n" +
+                '<option value="OPT1">OPTion One</option>' +
+                '</select>'
+    assert_dom_equal(expected, select(:post, :title, 
+                {  'OPTion Two' => 'OPT2', 'OPTion One' => 'OPT1'  },
+                {  :include_blank => true  },
+                :id => :my_id
+                                     ))
+  end
+
   def test_text_field
     assert_dom_equal(
       '<input id="post_title" name="post[title]" size="30" type="text" value="Hello World" />', text_field("post", "title")
@@ -1173,4 +1208,4 @@ class FormHelperTest < ActionView::TestCase
     def protect_against_forgery?
       false
     end
-end
\ No newline at end of file
+end
