From be99d5b8ecbbf1a1b8616a99703e77e16ac90541 Mon Sep 17 00:00:00 2001
From: Akira Matsuda <ronnie@dio.jp>
Date: Tue, 9 Sep 2008 17:24:23 +0900
Subject: [PATCH] Ruby 1.9 compat: use String#lines instead of to_a in Ruby 1.9

---
 railties/lib/commands/plugin.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/commands/plugin.rb b/railties/lib/commands/plugin.rb
index 980244a..5c6cb76 100644
--- a/railties/lib/commands/plugin.rb
+++ b/railties/lib/commands/plugin.rb
@@ -901,7 +901,7 @@ class RecursiveHTTPFetcher
   def initialize(urls_to_fetch, level = 1, cwd = ".")
     @level = level
     @cwd = cwd
-    @urls_to_fetch = urls_to_fetch.to_a
+    @urls_to_fetch = RUBY_VERSION >= '1.9' ? urls_to_fetch.lines : urls_to_fetch.to_a
     @quiet = false
   end
 
-- 
1.6.0.1

