--- url_helper.rb
+++ (clipboard)
@@ -561,10 +561,8 @@
           confirm, popup = html_options.delete("confirm"), html_options.delete("popup")
 
           method, href = html_options.delete("method"), html_options['href']
-          
+
           html_options["onclick"] = case
-            when (popup || method || confirm) && html_options.has_key?("onclick")
-              raise ActionView::ActionViewError, "You can't use :popup, :method, or :confirm with :onclick in the same link" 
             when popup && method
               raise ActionView::ActionViewError, "You can't use :popup and :method in the same link"
             when confirm && popup
