From fa4c97aee19e026f165f9cbb71fd89eedcb055f8 Mon Sep 17 00:00:00 2001
From: Nic Benders <nic.benders@gmail.com>
Date: Sun, 26 Sep 2010 11:45:20 -0700
Subject: [PATCH] db:structure:dump should list current RAILS_ENV adapter in errors, instead of always the test adapter

---
 railties/lib/tasks/databases.rake |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index 1cf2434..db1e18c 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -296,7 +296,7 @@ namespace :db do
         db_string = firebird_db_string(abcs[RAILS_ENV])
         sh "isql -a #{db_string} > #{RAILS_ROOT}/db/#{RAILS_ENV}_structure.sql"
       else
-        raise "Task not supported by '#{abcs["test"]["adapter"]}'"
+        raise "Task not supported by '#{abcs[RAILS_ENV]["adapter"]}'"
       end
 
       if ActiveRecord::Base.connection.supports_migrations?
-- 
1.7.2.2

