diff --git a/activerecord/test/cases/relations_test.rb b/activerecord/test/cases/relations_test.rb
index aa75aa2..2107031 100644
--- a/activerecord/test/cases/relations_test.rb
+++ b/activerecord/test/cases/relations_test.rb
@@ -667,4 +667,8 @@ class RelationTest < ActiveRecord::TestCase
   def test_relations_limit_with_conditions_or_limit
     assert_equal Post.limit(2).size, Post.limit(2).all.size
   end
+  
+  def test_chaining_scopes_using_order_method_is_equivalent_to_chaining_scopes_using_scope_with_that_order
+    assert_equal Post.order("id").order("comments_count DESC").order_values, Post.order("id").ranked_by_comments.order_values
+  end  
 end
