diff --git a/actionpack/lib/action_controller/caching/sweeping.rb b/actionpack/lib/action_controller/caching/sweeping.rb
index cf16417..7edee80 100644
--- a/actionpack/lib/action_controller/caching/sweeping.rb
+++ b/actionpack/lib/action_controller/caching/sweeping.rb
@@ -56,7 +56,11 @@ module ActionController #:nodoc:
 
         def before(controller)
           self.controller = controller
-          callback(:before) if controller.perform_caching
+          if controller.perform_caching
+            callback(:before)
+          else
+            true
+          end
         end
 
         def after(controller)
