diff --git a/railties/lib/rails/generators/test_unit/controller/controller_generator.rb b/railties/lib/rails/generators/test_unit/controller/controller_generator.rb
index 39816d9..20f3bd8 100644
--- a/railties/lib/rails/generators/test_unit/controller/controller_generator.rb
+++ b/railties/lib/rails/generators/test_unit/controller/controller_generator.rb
@@ -3,6 +3,7 @@ require 'rails/generators/test_unit'
 module TestUnit
   module Generators
     class ControllerGenerator < Base
+      argument :actions, :type => :array, :default => [], :banner => "action action"
       check_class_collision :suffix => "ControllerTest"
 
       def create_test_files
diff --git a/railties/lib/rails/generators/test_unit/controller/templates/functional_test.rb b/railties/lib/rails/generators/test_unit/controller/templates/functional_test.rb
index 62fa5d8..0d41858 100644
--- a/railties/lib/rails/generators/test_unit/controller/templates/functional_test.rb
+++ b/railties/lib/rails/generators/test_unit/controller/templates/functional_test.rb
@@ -1,8 +1,18 @@
 require 'test_helper'
 
 class <%= class_name %>ControllerTest < ActionController::TestCase
+<% if actions.empty? -%>
   # Replace this with your real tests.
   test "the truth" do
     assert true
   end
+<% else -%>
+<% for action in actions -%>
+  test "should get <%= action %>" do
+    get :<%= action %>
+    assert_response :success
+  end
+
+<% end -%>
+<% end -%>
 end
