diff --git a/activerecord/lib/active_record/associations.rb b/activerecord/lib/active_record/associations.rb
index f2feac0..e8782a5 100644
--- a/activerecord/lib/active_record/associations.rb
+++ b/activerecord/lib/active_record/associations.rb
@@ -1416,8 +1416,9 @@ module ActiveRecord
         include Module.new {
           class_eval <<-RUBY, __FILE__, __LINE__ + 1
             def destroy                     # def destroy
-              super                         #   super
+              result = super                #   result = super
               #{reflection.name}.clear      #   posts.clear
+              result                        #   result
             end                             # end
           RUBY
         }
