diff --git a/actionpack/lib/action_dispatch/routing/url_for.rb b/actionpack/lib/action_dispatch/routing/url_for.rb
index ec78f53..b8c02f4 100644
--- a/actionpack/lib/action_dispatch/routing/url_for.rb
+++ b/actionpack/lib/action_dispatch/routing/url_for.rb
@@ -72,7 +72,7 @@ module ActionDispatch
     # you can do that by including ActionController::UrlFor in your class:
     #
     #   class User < ActiveRecord::Base
-    #     include ActionController::UrlFor
+    #     include Rails.application.routes.url_helpers
     #
     #     def base_uri
     #       user_path(self)
