b/active_support/core_ext/string/output_safety.rb.old     2010-12-13 21:35:02.000000000 +0100
--- lib/active_support/core_ext/string/output_safety.rb 2010-12-13 21:45:25.000000000 +0100
***************
*** 63,68 ****
--- 63,78 ----
    end
  end
  
+ class Array
+   #
+   # ['<a>', <b>.html_safe].safe_join(' ')
+   # => "&lt;a&gt; <b>" 
+   #
+   def safe_join(sep = '')
+     collect{|v| v.html_safe? ? v : ERB::Util::h(v)}.join(sep).html_safe
+   end
+ end
+ 
  module ActiveSupport #:nodoc:
    class SafeBuffer < String
      alias safe_concat concat

