From b72263b08468744869b1901f89125492756a99ff Mon Sep 17 00:00:00 2001
From: Jake Howerton <jake.howerton@gmail.com>
Date: Fri, 16 May 2008 01:29:55 -0400
Subject: [PATCH] add rdoc task to activemodel

---
 activemodel/Rakefile |   15 ++++++++++++++-
 1 files changed, 14 insertions(+), 1 deletions(-)

diff --git a/activemodel/Rakefile b/activemodel/Rakefile
index cb9a617..0f77b66 100644
--- a/activemodel/Rakefile
+++ b/activemodel/Rakefile
@@ -1,4 +1,17 @@
 #!/usr/bin/env ruby
 $LOAD_PATH << File.join(File.dirname(__FILE__), 'vendor', 'rspec', 'lib')
 require 'rake'
-require 'spec/rake/spectask'
\ No newline at end of file
+require 'spec/rake/spectask'
+require 'rake/rdoctask'
+
+# Generate the RDoc documentation
+Rake::RDocTask.new { |rdoc|
+  rdoc.rdoc_dir = 'doc'
+  rdoc.title    = "Active Model"
+  rdoc.options << '--line-numbers' << '--inline-source' << '-A cattr_accessor=object'
+  rdoc.options << '--charset' << 'utf-8'
+  rdoc.template = "#{ENV['template']}.rb" if ENV['template']
+  rdoc.rdoc_files.include('README', 'CHANGES')
+  rdoc.rdoc_files.include('lib/active_model.rb')
+  rdoc.rdoc_files.include('lib/active_model/*.rb')
+}
\ No newline at end of file
-- 
1.5.5.1

