From 345ca7fcdff6bdb89ef27720179431f8fc5df14a Mon Sep 17 00:00:00 2001
From: Chad Woolley <thewoolleyman@gmail.com>
Date: Tue, 11 Aug 2009 22:53:55 -0700
Subject: [PATCH] make mysql and postgresql rebuild databases on every CI build, to prevent breakages such as collation and character set changing

---
 ci/ci_build.rb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/ci/ci_build.rb b/ci/ci_build.rb
index 0b9bd3d..7d81fa8 100755
--- a/ci/ci_build.rb
+++ b/ci/ci_build.rb
@@ -28,14 +28,14 @@ cd "#{root_dir}/activerecord" do
   puts
   puts "[CruiseControl] Building ActiveRecord with MySQL"
   puts
-  build_results[:activerecord_mysql] = system 'rake test_mysql'
+  build_results[:activerecord_mysql] = system 'rake mysql:rebuild_databases && rake test_mysql'
 end
 
 cd "#{root_dir}/activerecord" do
   puts
   puts "[CruiseControl] Building ActiveRecord with PostgreSQL"
   puts
-  build_results[:activerecord_postgresql8] = system 'rake test_postgresql'
+  build_results[:activerecord_postgresql8] = system 'rake postgresql:rebuild_databases && rake test_postgresql'
 end
 
 cd "#{root_dir}/activerecord" do
-- 
1.6.3.3

