From d57773998dc4efb24c641513e871f0e36da02bca Mon Sep 17 00:00:00 2001
From: Ken Collins <ken@metaskills.net>
Date: Fri, 23 Jul 2010 16:02:23 -0400
Subject: [PATCH] Conversion of a two dimensional array that is ruby 1.8.6 safe.

---
 .../associations/association_collection.rb         |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/associations/association_collection.rb b/activerecord/lib/active_record/associations/association_collection.rb
index 2855073..6384526 100644
--- a/activerecord/lib/active_record/associations/association_collection.rb
+++ b/activerecord/lib/active_record/associations/association_collection.rb
@@ -382,7 +382,7 @@ module ActiveRecord
             return  send("find_by_#{rest}", *args) ||
                     method_missing("create_by_#{rest}", *args)
           when /^create_by_(.*)$/
-            return create Hash[$1.split('_and_').zip(args)]
+            return create $1.split('_and_').zip(args).inject({}) { |h,kv| k,v=kv ; h[k] = v ; h }
           end
 
           if @target.respond_to?(method) || (!@reflection.klass.respond_to?(method) && Class.respond_to?(method))
-- 
1.6.4.2

