From ecafd3d0296d197055bf78c119017bb8d1eba949 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Juan=20Germ=C3=A1n=20Casta=C3=B1eda=20Echevarr=C3=ADa?= <juanger@gmail.com>
Date: Thu, 18 Dec 2008 21:33:43 -0600
Subject: [PATCH] Changed Rack::Logger to Rack::LogTailer in server command

---
 railties/lib/commands/server.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/commands/server.rb b/railties/lib/commands/server.rb
index 7057fcc..895f7ab 100644
--- a/railties/lib/commands/server.rb
+++ b/railties/lib/commands/server.rb
@@ -84,7 +84,7 @@ else
 end
 
 app = Rack::Builder.new {
-  use Rails::Rack::Logger
+  use Rails::Rack::LogTailer
   use Rails::Rack::Static
   use Rails::Rack::Debugger if options[:debugger]
   run inner_app
-- 
1.5.6.1

