Index: activerecord/lib/active_record/base.rb
===================================================================
--- a/rails/activerecord/lib/active_record/base.rb
+++ b/rails/activerecord/lib/active_record/base.rb
@@ -689,5 +689,4 @@
       def exists?(id_or_conditions = {})
         find_initial(
-          :select => "#{quoted_table_name}.#{primary_key}",
           :conditions => expand_id_conditions(id_or_conditions)) ? true : false
       end
