diff --git a/actionpack/lib/action_controller/routing/generation/polymorphic_routes.rb b/actionpack/lib/action_controller/routing/generation/polymorphic_routes.rb
index d9b614c..ca39c0a 100644
--- a/actionpack/lib/action_controller/routing/generation/polymorphic_routes.rb
+++ b/actionpack/lib/action_controller/routing/generation/polymorphic_routes.rb
@@ -171,7 +171,7 @@ module ActionController
 
         if record.is_a?(Symbol) || record.is_a?(String)
           route << "#{record}_"
-        else
+        elsif !record.nil?
           route << "#{RecordIdentifier.__send__("plural_class_name", record)}"
           route = route.singularize if inflection == :singular
           route << "_"
diff --git a/actionpack/test/activerecord/polymorphic_routes_test.rb b/actionpack/test/activerecord/polymorphic_routes_test.rb
index b9f5be2..fb57be4 100644
--- a/actionpack/test/activerecord/polymorphic_routes_test.rb
+++ b/actionpack/test/activerecord/polymorphic_routes_test.rb
@@ -180,6 +180,12 @@ class PolymorphicRoutesTest < ActionController::TestCase
     end
   end
   
+  def test_nesting_with_array_containing_only_singleton_resources
+    with_test_routes do
+      assert_equal "http://example.com/account/preference", polymorphic_url([:account, :preference])
+    end
+  end
+  
   def test_nesting_with_array_containing_singleton_resource_and_format
     with_test_routes do
       @project.save
@@ -344,6 +350,9 @@ class PolymorphicRoutesTest < ActionController::TestCase
           taxes.resource :bid
         end
         map.resources :series
+        map.resource :account do |account|
+          account.resource :preference
+        end
       end
 
       ActionController::Routing::Routes.install_helpers(self.class)
