diff --git a/activerecord/lib/active_record/transactions.rb b/activerecord/lib/active_record/transactions.rb
index 0b6e52c..92a374d 100644
--- a/activerecord/lib/active_record/transactions.rb
+++ b/activerecord/lib/active_record/transactions.rb
@@ -135,7 +135,8 @@ module ActiveRecord
     #     end
     #   end
     #   
-    #   User.find(:all)  # => empty
+    #   User.find(:all)  # => Returns Kotori and Nemu, because AcriveRecord::Rollback
+    #   do not re-raise
     #
     # It is also possible to requires a sub-transaction by passing
     # <tt>:requires_new => true</tt>.  If anything goes wrong, the
