From 25c1ff1afabf4a74a94c8ab3605aad69e14d1d86 Mon Sep 17 00:00:00 2001
From: Ryan Bates <ryan@railscasts.com>
Date: Wed, 21 May 2008 16:12:54 -0700
Subject: [PATCH] adding Rails.version

---
 railties/lib/initializer.rb |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index dfd4304..bf3579e 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -42,6 +42,10 @@ module Rails
     def cache
       RAILS_CACHE
     end
+  
+    def version
+      VERSION::STRING
+    end
 
     def public_path
       @@public_path ||= self.root ? File.join(self.root, "public") : "public"
-- 
1.5.5.1

