--- lib/active_record/connection_adapters/abstract/schema_statements.rb-orig	2009-07-12 15:47:09.000000000 +0200
+++ lib/active_record/connection_adapters/abstract/schema_statements.rb	2009-07-12 15:48:39.000000000 +0200
@@ -321,7 +321,7 @@
             schema_migrations_table.column :version, :string, :null => false
           end
           add_index sm_table, :version, :unique => true,
-            :name => 'unique_schema_migrations'
+            :name => Base.table_name_prefix + 'unique_schema_migrations' + Base.table_name_suffix
 
           # Backwards-compatibility: if we find schema_info, assume we've
           # migrated up to that point:
