diff --git a/actionpack/lib/action_controller/base.rb b/actionpack/lib/action_controller/base.rb
index e221141..65aebd6 100644
--- a/actionpack/lib/action_controller/base.rb
+++ b/actionpack/lib/action_controller/base.rb
@@ -260,7 +260,7 @@ module ActionController #:nodoc:
     # Prepends all the URL-generating helpers from AssetHelper. This makes it possible to easily move javascripts, stylesheets,
     # and images to a dedicated asset server away from the main web server. Example:
     #   ActionController::Base.asset_host = "http://assets.example.com"
-    @@asset_host = ""
+    @@asset_host = nil
     cattr_accessor :asset_host
 
     # All requests are considered local by default, so everyone will be exposed to detailed debugging screens on errors.
diff --git a/actionpack/lib/action_view/helpers/asset_tag_helper.rb b/actionpack/lib/action_view/helpers/asset_tag_helper.rb
index 58f8cca..b1724be 100644
--- a/actionpack/lib/action_view/helpers/asset_tag_helper.rb
+++ b/actionpack/lib/action_view/helpers/asset_tag_helper.rb
@@ -539,6 +539,8 @@ module ActionView
             else
               (host =~ /%d/) ? host % (source.hash % 4) : host
             end
+          else
+            respond_to?(:root_url) ? root_url.sub(/\/$/, '') : nil
           end
         end
 
diff --git a/actionpack/test/template/asset_tag_helper_test.rb b/actionpack/test/template/asset_tag_helper_test.rb
index 5e2fc20..3ccd866 100644
--- a/actionpack/test/template/asset_tag_helper_test.rb
+++ b/actionpack/test/template/asset_tag_helper_test.rb
@@ -654,6 +654,11 @@ class AssetTagHelperNonVhostTest < ActionView::TestCase
     ActionController::Base.relative_url_root = nil
   end
 
+  def test_should_compute_absolute_path_when_root_url_is_routed
+    stubs(:root_url => 'http://www.example.com/')
+    assert_dom_equal(%(http://www.example.com/collaboration/hieraki/images/xml.png), image_path("xml.png"))
+  end
+
   def test_should_compute_proper_path
     assert_dom_equal(%(<link href="http://www.example.com/collaboration/hieraki" rel="alternate" title="RSS" type="application/rss+xml" />), auto_discovery_link_tag)
     assert_dom_equal(%(/collaboration/hieraki/javascripts/xmlhr.js), javascript_path("xmlhr"))
