From 6ef87e4855a109d0b7dd39891fa38c233bc1aa6f Mon Sep 17 00:00:00 2001
From: Sebastian Martinez <sebastian@wyeworks.com>
Date: Tue, 23 Mar 2010 11:40:04 -0300
Subject: [PATCH] updated Rakefile with new app generator path

---
 railties/Rakefile |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/railties/Rakefile b/railties/Rakefile
index fe049d5..6368456 100644
--- a/railties/Rakefile
+++ b/railties/Rakefile
@@ -36,7 +36,7 @@ end
 
 desc "Updates application README to the latest version Railties README"
 task :update_readme do
-  readme = "lib/generators/rails/app/templates/README"
+  readme = "lib/rails/generators/rails/app/templates/README"
   rm readme
   cp "./README", readme
 end
@@ -48,7 +48,7 @@ task :generate_guides do
 end
 
 task :update_prototype_ujs do
-  system "curl http://github.com/rails/prototype-ujs/raw/master/src/rails.js > lib/generators/rails/app/templates/public/javascripts/rails.js"
+  system "curl http://github.com/rails/prototype-ujs/raw/master/src/rails.js > lib/rails/generators/rails/app/templates/public/javascripts/rails.js"
 end
 
 # Generate documentation ------------------------------------------------------------------
@@ -62,8 +62,8 @@ Rake::RDocTask.new { |rdoc|
   rdoc.rdoc_files.include('README', 'CHANGELOG')
   rdoc.rdoc_files.include('lib/*.rb')
   rdoc.rdoc_files.include('lib/rails/*.rb')
-  rdoc.rdoc_files.include('lib/generators/*.rb')
-  rdoc.rdoc_files.include('lib/commands/**/*.rb')
+  rdoc.rdoc_files.include('lib/rails/generators/*.rb')
+  rdoc.rdoc_files.include('lib/rails/commands/**/*.rb')
 }
 
 # Generate GEM ----------------------------------------------------------------------------
-- 
1.6.3.3

