diff --git a/railties/lib/commands/dbconsole.rb b/railties/lib/commands/dbconsole.rb
index 8002264..d793e01 100644
--- a/railties/lib/commands/dbconsole.rb
+++ b/railties/lib/commands/dbconsole.rb
@@ -68,14 +68,14 @@ when "postgresql"
   exec(find_cmd('psql'), config["database"])
 
 when "sqlite"
-  exec(find_cmd('sqlite'), config["database"])
+  exec(find_cmd('sqlite'), config['database'] || config['dbfile'])
 
 when "sqlite3"
   args = []
 
   args << "-#{options['mode']}" if options['mode']
   args << "-header" if options['header']
-  args << config['database']
+  args << config['database'] || config['dbfile']
 
   exec(find_cmd('sqlite3'), *args)
 else
diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index 9588fab..4e8bacd 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -401,7 +401,8 @@ def drop_database(config)
   when 'mysql'
     ActiveRecord::Base.establish_connection(config)
     ActiveRecord::Base.connection.drop_database config['database']
-  when /^sqlite/
+  when 'sqlite', 'sqlite3'
+    dbfile = config['database'] || config['dbfile']
     FileUtils.rm(File.join(RAILS_ROOT, config['database']))
   when 'postgresql'
     ActiveRecord::Base.establish_connection(config.merge('database' => 'postgres', 'schema_search_path' => 'public'))
