diff --git a/activerecord/lib/active_record/validations.rb b/activerecord/lib/active_record/validations.rb
index 1d12ea8..a9a5bc3 100755
--- a/activerecord/lib/active_record/validations.rb
+++ b/activerecord/lib/active_record/validations.rb
@@ -943,6 +943,11 @@ module ActiveRecord
     def errors
       @errors ||= Errors.new(self)
     end
+    
+    # Returns true if object has errors otherwise false
+    def has_errors?
+      !errors.empty?
+    end
 
     protected
       # Overwrite this method for validation checks on all saves and use Errors.add(field, msg) for invalid attributes.
