From d1e49ea29a967c8957cea98b5964036d84265d16 Mon Sep 17 00:00:00 2001
From: Aaron Patterson <aaron.patterson@gmail.com>
Date: Mon, 7 Jun 2010 22:50:29 -0400
Subject: [PATCH] adding oracle stuff to Gemfile so that oracle tests may be run more easily. [#4791 state:resolved]

---
 Gemfile |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/Gemfile b/Gemfile
index e91c569..86dfbe5 100644
--- a/Gemfile
+++ b/Gemfile
@@ -35,6 +35,12 @@ if mri || RUBY_ENGINE == "rbx"
     gem "pg", ">= 0.9.0"
     gem "mysql", ">= 2.8.1"
   end
+
+  group :oracle do
+    gem "ruby-oci8"
+    gem "activerecord-oracle_enhanced-adapter", :git => 'http://github.com/rsim/oracle-enhanced.git', :branch => 'rails3'
+  end
+
 elsif RUBY_ENGINE == "jruby"
   gem "activerecord-jdbcsqlite3-adapter"
 
-- 
1.7.0.6

