diff --git i/activesupport/lib/active_support/json/decoding.rb w/activesupport/lib/active_support/json/decoding.rb
index fdb219d..9da4048 100644
--- i/activesupport/lib/active_support/json/decoding.rb
+++ w/activesupport/lib/active_support/json/decoding.rb
@@ -16,7 +16,7 @@ module ActiveSupport
       
       protected
         # matches YAML-formatted dates
-        DATE_REGEX = /^\d{4}-\d{2}-\d{2}|\d{4}-\d{1,2}-\d{1,2}[ \t]+\d{1,2}:\d{2}:\d{2}(\.[0-9]*)?(([ \t]*)Z|[-+]\d{2}?(:\d{2})?)?$/
+        DATE_REGEX = /^(?:\d{4}-\d{2}-\d{2}|\d{4}-\d{1,2}-\d{1,2}[ \t]+\d{1,2}:\d{2}:\d{2}(\.[0-9]*)?(([ \t]*)Z|[-+]\d{2}?(:\d{2})?)?)$/
 
         # Ensure that ":" and "," are always followed by a space
         def convert_json_to_yaml(json) #:nodoc:
diff --git i/activesupport/test/json/decoding_test.rb w/activesupport/test/json/decoding_test.rb
index 19ae3a0..558b03b 100644
--- i/activesupport/test/json/decoding_test.rb
+++ w/activesupport/test/json/decoding_test.rb
@@ -15,7 +15,8 @@ class TestJSONDecoding < Test::Unit::TestCase
     # no time zone
     %({a: "2007-01-01 01:12:34"})              => {'a' => "2007-01-01 01:12:34"}, 
     # needs to be *exact*
-    %({a: " 2007-01-01 01:12:34 Z "})          => {'a' => " 2007-01-01 01:12:34 Z "}, 
+    %({a: " 2007-01-01 01:12:34 Z "})          => {'a' => " 2007-01-01 01:12:34 Z "},
+    %({a: "2007-01-01 : it's your birthday"})  => {'a' => "2007-01-01 : it's your birthday"},
     %([])    => [],
     %({})    => {},
     %(1)     => 1,
