From 9bccb35e00db8cb3eb3917f5a1079ee614fe3f3d Mon Sep 17 00:00:00 2001
From: Simon Effenberg <savar@schuldeigen.de>
Date: Wed, 31 Mar 2010 01:24:44 +0200
Subject: [PATCH] remove_index now uses quote_table_name() [#4300 state:resolved]

---
 .../abstract/schema_statements.rb                  |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
index 86ba7d7..6d4ab50 100644
--- a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
+++ b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
@@ -291,7 +291,7 @@ module ActiveRecord
       # Remove the index named by_branch_party in the accounts table.
       #   remove_index :accounts, :name => :by_branch_party
       def remove_index(table_name, options = {})
-        execute "DROP INDEX #{quote_column_name(index_name(table_name, options))} ON #{table_name}"
+        execute "DROP INDEX #{quote_column_name(index_name(table_name, options))} ON #{quote_table_name(table_name)}"
       end
 
       def index_name(table_name, options) #:nodoc:
-- 
1.7.0

