From d309e694af3c41de4497e905dcfb0a8c489b5b48 Mon Sep 17 00:00:00 2001
From: lakshmanan <lucky.developer@gmail.com>
Date: Wed, 30 Mar 2011 16:05:59 +0530
Subject: [PATCH] action-pack README typo fix

---
 actionpack/README.rdoc |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/actionpack/README.rdoc b/actionpack/README.rdoc
index 3661d27..bfccee9 100644
--- a/actionpack/README.rdoc
+++ b/actionpack/README.rdoc
@@ -34,7 +34,7 @@ A short rundown of some of the major features:
 * Actions grouped in controller as methods instead of separate command objects
   and can therefore share helper methods
 
-    CustomersController < ActionController::Base
+    class CustomersController < ActionController::Base
       def show
         @customer = find_customer
       end
-- 
1.7.4.1

