diff --git a/actionpack/test/template/form_options_helper_test.rb b/actionpack/test/template/form_options_helper_test.rb
index 48a26de..b882e9d 100644
--- a/actionpack/test/template/form_options_helper_test.rb
+++ b/actionpack/test/template/form_options_helper_test.rb
@@ -365,6 +365,51 @@ class FormOptionsHelperTest < ActionView::TestCase
     )
   end
 
+  def test_collection_select_under_fields_for_with_collection
+    @posts = [
+      Post.new("<Abe> went home", "<Abe>", "To a little house", "shh!"),
+      Post.new("Babe went home", "Babe", "To a little house", "shh!"),
+      Post.new("Cabe went home", "Cabe", "To a little house", "shh!")
+    ]
+
+    @post = Post.new
+    @post.author_name = "Babe"
+    def @post.to_param; 1; end
+
+    _erbout = ''
+
+    fields_for 'post[]', @post do |f|
+      _erbout.concat f.collection_select(:author_name, @posts, :author_name, :author_name)
+    end
+
+    assert_dom_equal(
+      "<select id=\"post_1_author_name\" name=\"post[1][author_name]\"><option value=\"&lt;Abe&gt;\">&lt;Abe&gt;</option>\n<option value=\"Babe\" selected=\"selected\">Babe</option>\n<option value=\"Cabe\">Cabe</option></select>",
+      _erbout
+    )
+  end
+
+  def test_collection_select_under_fields_for_with_collection_and_index
+    @posts = [
+      Post.new("<Abe> went home", "<Abe>", "To a little house", "shh!"),
+      Post.new("Babe went home", "Babe", "To a little house", "shh!"),
+      Post.new("Cabe went home", "Cabe", "To a little house", "shh!")
+    ]
+
+    @post = Post.new
+    @post.author_name = "Babe"
+
+    _erbout = ''
+
+    fields_for :post, @post, :index => 1 do |f|
+      _erbout.concat f.collection_select(:author_name, @posts, :author_name, :author_name)
+    end
+
+    assert_dom_equal(
+      "<select id=\"post_1_author_name\" name=\"post[1][author_name]\"><option value=\"&lt;Abe&gt;\">&lt;Abe&gt;</option>\n<option value=\"Babe\" selected=\"selected\">Babe</option>\n<option value=\"Cabe\">Cabe</option></select>",
+      _erbout
+    )
+  end
+
   def test_collection_select_with_blank_and_style
     @posts = [
       Post.new("<Abe> went home", "<Abe>", "To a little house", "shh!"),
