--- dirty-old.rb	Mon Aug 25 14:27:45 2008
+++ dirty.rb	Mon Aug 25 14:29:17 2008
@@ -150,9 +150,8 @@
             # If an old value of 0 is set to '' we want this to get changed to nil as otherwise it'll
             # be typecast back to 0 (''.to_i => 0)
             value = nil if value.blank?
-          else
-            value = column.type_cast(value)
           end
+          value = column.type_cast(value)
         end
 
         old != value
