From aefddcbbf0c54aaa73076e9ac601ff1e5608dbe9 Mon Sep 17 00:00:00 2001
From: Steven Soroka <ssoroka78@gmail.com>
Date: Fri, 16 May 2008 17:41:59 -0500
Subject: [PATCH] SchemaStatements#dump_schema_information joins inserts with a single \n, but is later split on \n\n, and fails when trying to execute all the inserts as a single sql statement. [Closes #201]

---
 .../abstract/schema_statements.rb                  |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
index 1594be4..e7741de 100644
--- a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
+++ b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
@@ -302,7 +302,7 @@ module ActiveRecord
       def dump_schema_information #:nodoc:
         sm_table = ActiveRecord::Migrator.schema_migrations_table_name
         migrated = select_values("SELECT version FROM #{sm_table}")
-        migrated.map { |v| "INSERT INTO #{sm_table} (version) VALUES ('#{v}');" }.join("\n")
+        migrated.map { |v| "INSERT INTO #{sm_table} (version) VALUES ('#{v}');" }.join("\n\n")
       end
 
       # Should not be called normally, but this operation is non-destructive.
-- 
1.5.5

