=== modified file 'vendor/rails/activerecord/test/cases/autosave_association_test.rb'
--- vendor/rails/activerecord/test/cases/autosave_association_test.rb	2009-06-18 13:18:59 +0000
+++ vendor/rails/activerecord/test/cases/autosave_association_test.rb	2009-06-18 13:20:52 +0000
@@ -39,6 +39,17 @@
     f.save
   end
 
+  def test_has_one_with_primary_and_foreign_key_set_to_id
+    f = Firm.new(:name => 'Firm with account_using_primary_key', :firm_id => 2)
+    a = Account.create(:credit_limit => 20, :firm_id => f.firm_id)
+
+    assoc = f.account_using_primary_key(true)
+    f.save
+
+    assert_equal assoc.firm_id, f.firm_id
+    assert_not_equal assoc.firm_id, f.id
+  end
+
   private
 
   def base

