--- sqlite_adapter.rb	2008-04-22 13:29:35.000000000 -0500
+++ /tmp/sqlite_adapter.rb	2008-05-14 12:49:08.000000000 -0500
@@ -216,7 +216,7 @@
       def add_column(table_name, column_name, type, options = {}) #:nodoc:
         super(table_name, column_name, type, options)
         # See last paragraph on http://www.sqlite.org/lang_altertable.html
-        execute "VACUUM"
+        execute "VACUUM" unless ActiveRecord::Base.connection.instance_variable_get(:@connection).transaction_active?
       end
 
       def remove_column(table_name, column_name) #:nodoc:
