diff --git a/ci/ci_setup_notes.txt b/ci/ci_setup_notes.txt
index 63f1851..780277c 100644
--- a/ci/ci_setup_notes.txt
+++ b/ci/ci_setup_notes.txt
@@ -47,7 +47,7 @@ $ sudo aptitude update
 * If you did not configure no-root-gem installation via ~/.gemrc as shown above, then allow no-password sudo for gem installation:
 $ sudo visudo
 # add this line to bottom:
-ci      ALL=NOPASSWD: /usr/local/bin/geminstaller, /usr/local/bin/ruby, /usr/local/bin/gem
+ci      ALL=(ALL) NOPASSWD: ALL
 
 * Start ccrb via init script and check for default homepage at port 3333
 
diff --git a/ci/cruise_config.rb b/ci/cruise_config.rb
index 7985e3c..2247a1b 100644
--- a/ci/cruise_config.rb
+++ b/ci/cruise_config.rb
@@ -1,5 +1,5 @@
 Project.configure do |project|
- project.build_command = 'ruby ci/ci_build.rb'
+ project.build_command = 'sudo update_rubygems && ruby ci/ci_build.rb'
  project.email_notifier.emails = ['thewoolleyman@gmail.com']
 # project.email_notifier.emails = ['thewoolleyman@gmail.com','michael@koziarski.com', 'david@loudthinking.com', 'jeremy@bitsweat.net', 'josh@joshpeek.com', 'pratiknaik@gmail.com', 'wycats@gmail.com']
  project.email_notifier.from = 'thewoolleyman+railsci@gmail.com'
diff --git a/ci/geminstaller.yml b/ci/geminstaller.yml
index 387e370..f1ef9b5 100644
--- a/ci/geminstaller.yml
+++ b/ci/geminstaller.yml
@@ -21,3 +21,5 @@ gems:
   version: >= 2.2.3
 - name: sqlite3-ruby
   version: >= 1.2.2
+- name: rubygems-update
+  version: >= 1.3.3
