Index: vendor/rails/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
===================================================================
--- vendor/rails/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb    (revision 8612)
+++ vendor/rails/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb    (working copy)
@@ -383,7 +383,8 @@

       def tables(name = nil) #:nodoc:
         tables = []
-        execute("SHOW TABLES", name).each { |field| tables << field[0] }
+        execute("SHOW TABLES", name).each { |field| tables << field[0] } unless @tables
+        @tables ||= tables
         tables
       end
