From 4681c0abd0074e33ede26b8033c9ea0bbc052411 Mon Sep 17 00:00:00 2001
From: Murat Demirten <mdemirten@yh.com.tr>
Date: Mon, 8 Dec 2008 22:43:46 +0200
Subject: [PATCH] Association_Proxy method_missing fix

---
 .../associations/association_proxy.rb              |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

diff --git a/activerecord/lib/active_record/associations/association_proxy.rb b/activerecord/lib/active_record/associations/association_proxy.rb
index 59f1d3b..3d12dda 100644
--- a/activerecord/lib/active_record/associations/association_proxy.rb
+++ b/activerecord/lib/active_record/associations/association_proxy.rb
@@ -207,11 +207,6 @@ module ActiveRecord
         # Forwards any missing method call to the \target.
         def method_missing(method, *args)
           if load_target
-            unless @target.respond_to?(method)
-              message = "undefined method `#{method.to_s}' for \"#{@target}\":#{@target.class.to_s}"
-              raise NoMethodError, message
-            end
-
             if block_given?
               @target.send(method, *args)  { |*block_args| yield(*block_args) }
             else
-- 
1.5.6

