diff --git a/actionpack/test/template/number_helper_test.rb b/actionpack/test/template/number_helper_test.rb
index a21a1a6..14e81fc 100644
--- a/actionpack/test/template/number_helper_test.rb
+++ b/actionpack/test/template/number_helper_test.rb
@@ -188,6 +188,7 @@ class NumberHelperTest < ActionView::TestCase
     assert_equal '10 KB',     number_to_human_size(kilobytes(10.000), :precision => 4)
     assert_equal '1 TB', number_to_human_size(1234567890123, :precision => 1)
     assert_equal '500 MB', number_to_human_size(524288000, :precision=>3)
+    assert_equal '10 MB', number_to_human_size(9961472, :precision=>0)
     assert_equal '40 KB', number_to_human_size(41010, :precision => 1)
     assert_equal '40 KB', number_to_human_size(41100, :precision => 2)
     assert_equal '1.0 KB',   number_to_human_size(kilobytes(1.0123), :precision => 2, :strip_insignificant_zeros => false)
