From 6732160d077c241a96cf3be83e08a4a0a491fb81 Mon Sep 17 00:00:00 2001
From: Kamal Fariz Mahyuddin <kamal.fariz@gmail.com>
Date: Mon, 10 Aug 2009 14:12:26 +0800
Subject: [PATCH] find_cmd should return the full path of the db command

---
 railties/lib/commands/dbconsole.rb |   12 ++++++++----
 1 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/railties/lib/commands/dbconsole.rb b/railties/lib/commands/dbconsole.rb
index 8002264..e6f11a4 100644
--- a/railties/lib/commands/dbconsole.rb
+++ b/railties/lib/commands/dbconsole.rb
@@ -33,11 +33,15 @@ end
 def find_cmd(*commands)
   dirs_on_path = ENV['PATH'].to_s.split(File::PATH_SEPARATOR)
   commands += commands.map{|cmd| "#{cmd}.exe"} if RUBY_PLATFORM =~ /win32/
-  commands.detect do |cmd|
-    dirs_on_path.detect do |path|
-      File.executable? File.join(path, cmd)
+
+  full_path_command = nil
+  found = commands.detect do |cmd|
+    dir = dirs_on_path.detect do |path|
+      full_path_command = File.join(path, cmd)
+      File.executable? full_path_command
     end
-  end || abort("Couldn't find database client: #{commands.join(', ')}. Check your $PATH and try again.")
+  end
+  found ? full_path_command : abort("Couldn't find database client: #{commands.join(', ')}. Check your $PATH and try again.")
 end
 
 case config["adapter"]
-- 
1.6.4

