Index: actionpack/test/controller/mime_responds_test.rb
===================================================================
--- actionpack/test/controller/mime_responds_test.rb	(revision 646e5202e001a7c12247ce66dc85b4edde50f25c)
+++ actionpack/test/controller/mime_responds_test.rb	(revision )
@@ -61,6 +61,12 @@
     end
   end
 
+  def just_atom
+    respond_to do |type|
+      type.atom  { render :text => "ATOM" }
+    end
+  end
+
   def using_defaults
     respond_to do |type|
       type.html
@@ -331,6 +337,13 @@
     assert_equal 'HTML', @response.body
   end
 
+  def test_atom_from_browser
+    @request.accept = "text/xml,application/xml,application/xhtml+xml,text/html;q=0.9,text/plain;q=0.8,image/png,*/*;q=0.5"
+    get :just_atom
+    assert_equal 200, @response.code
+    assert_equal 'ATOM', @response.body
+  end
+
   def test_handle_any
     @request.accept = "*/*"
     get :handle_any
