diff -Naur /Library/Ruby/Gems/1.8/gems/activerecord-2.1.0/lib/active_record/connection_adapters/abstract/schema_statements.rb activerecord-2.1.0/lib/active_record/connection_adapters/abstract/schema_statements.rb
--- /Library/Ruby/Gems/1.8/gems/activerecord-2.1.0/lib/active_record/connection_adapters/abstract/schema_statements.rb	2008-06-05 12:24:45.000000000 -0700
+++ activerecord-2.1.0/lib/active_record/connection_adapters/abstract/schema_statements.rb	2008-06-05 12:21:27.000000000 -0700
@@ -307,7 +307,7 @@
 
       # Should not be called normally, but this operation is non-destructive.
       # The migrations module handles this automatically.
-      def initialize_schema_migrations_table
+      def initialize_schema_migrations_table(migrations_path)
         sm_table = ActiveRecord::Migrator.schema_migrations_table_name
 
         unless tables.detect { |t| t == sm_table }
@@ -324,19 +324,18 @@
           if tables.detect { |t| t == si_table }
 
             old_version = select_value("SELECT version FROM #{quote_table_name(si_table)}").to_i
-            assume_migrated_upto_version(old_version)
+            assume_migrated_upto_version(old_version, migrations_path)
             drop_table(si_table)
           end
         end
       end
 
-      def assume_migrated_upto_version(version)
+      def assume_migrated_upto_version(version, migrations_path)
         sm_table = quote_table_name(ActiveRecord::Migrator.schema_migrations_table_name)
         migrated = select_values("SELECT version FROM #{sm_table}").map(&:to_i)
-        versions = Dir['db/migrate/[0-9]*_*.rb'].map do |filename|
+        versions = Dir["#{migrations_path}/[0-9]*_*.rb"].map do |filename|
           filename.split('/').last.split('_').first.to_i
         end
-
         execute "INSERT INTO #{sm_table} (version) VALUES ('#{version}')" unless migrated.include?(version.to_i)
         (versions - migrated).select { |v| v < version.to_i }.each do |v|
           execute "INSERT INTO #{sm_table} (version) VALUES ('#{v}')"
diff -Naur /Library/Ruby/Gems/1.8/gems/activerecord-2.1.0/lib/active_record/migration.rb activerecord-2.1.0/lib/active_record/migration.rb
--- /Library/Ruby/Gems/1.8/gems/activerecord-2.1.0/lib/active_record/migration.rb	2008-06-05 12:24:45.000000000 -0700
+++ activerecord-2.1.0/lib/active_record/migration.rb	2008-06-05 12:21:27.000000000 -0700
@@ -384,7 +384,7 @@
 
     def initialize(direction, migrations_path, target_version = nil)
       raise StandardError.new("This database does not yet support migrations") unless Base.connection.supports_migrations?
-      Base.connection.initialize_schema_migrations_table
+      Base.connection.initialize_schema_migrations_table(migrations_path)
       @direction, @migrations_path, @target_version = direction, migrations_path, target_version      
     end
 
