diff --git a/activeresource/lib/active_resource/connection.rb b/activeresource/lib/active_resource/connection.rb
index 0c4ea43..7b6b3e3 100644
--- a/activeresource/lib/active_resource/connection.rb
+++ b/activeresource/lib/active_resource/connection.rb
@@ -64,7 +64,7 @@ module ActiveResource
   # services.
   class Connection
     attr_reader :site, :user, :password, :timeout
-    attr_accessor :format
+    attr_accessor :format, :verify_mode
 
     class << self
       def requests
@@ -74,11 +74,12 @@ module ActiveResource
 
     # The +site+ parameter is required and will set the +site+
     # attribute to the URI for the remote resource service.
-    def initialize(site, format = ActiveResource::Formats[:xml])
+    def initialize(site, format = ActiveResource::Formats[:xml], verify_mode = :none)
       raise ArgumentError, 'Missing site URI' unless site
       @user = @password = nil
       self.site = site
       self.format = format
+      self.verify_mode = verify_mode
     end
 
     # Set URI for remote service.
@@ -181,7 +182,13 @@ module ActiveResource
       def http
         http             = Net::HTTP.new(@site.host, @site.port)
         http.use_ssl     = @site.is_a?(URI::HTTPS)
-        http.verify_mode = OpenSSL::SSL::VERIFY_NONE if http.use_ssl
+        if http.use_ssl
+          http.verify_mode = if self.verify_mode == :peer
+            OpenSSL::SSL::VERIFY_PEER
+          else
+            OpenSSL::SSL::VERIFY_NONE
+          end
+        end
         http.read_timeout = @timeout if @timeout # If timeout is not set, the default Net::HTTP timeout (60s) is used.
         http
       end
