--- activerecord-2.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb.old	2009-05-21 11:39:00.000000000 +0400
+++ activerecord-2.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb	2009-05-21 13:23:32.000000000 +0400
@@ -657,7 +657,7 @@
       def columns(table_name, name = nil)
         # Limit, precision, and scale are all handled by the superclass.
         column_definitions(table_name).collect do |name, type, default, notnull|
-          PostgreSQLColumn.new(name, default, type, notnull == 'f')
+          ActiveRecord::ConnectionAdapters::PostgreSQLColumn.new(name, default, type, notnull == 'f')
         end
       end
 
