commit 5df48f5345e3febe75d7742b817ac09510343f03
Author: Alexey I. Froloff <raorn@altlinux.org>
Date:   Fri Aug 29 16:39:16 2008 +0400

    Make mongrel server usable

diff --git a/railties/lib/commands/servers/mongrel.rb b/railties/lib/commands/servers/mongrel.rb
index 7bb110f..bc562d2 100644
--- a/railties/lib/commands/servers/mongrel.rb
+++ b/railties/lib/commands/servers/mongrel.rb
@@ -43,7 +43,8 @@ parameters = [
 ]
 
 if OPTIONS[:detach]
-  `mongrel_rails #{parameters.join(" ")} -d`
+  parameters << "-d"
+  exec "mongrel_rails", *parameters
 else
   ENV["RAILS_ENV"] = OPTIONS[:environment]
   RAILS_ENV.replace(OPTIONS[:environment]) if defined?(RAILS_ENV)
@@ -61,9 +62,13 @@ else
 
   begin
     silence_warnings { ARGV = parameters }
-    load("mongrel_rails")
+    begin
+      load("mongrel_rails")
+    rescue LoadError
+      load(File.join(Config::CONFIG["bindir"], "mongrel_rails"))
+    end
   ensure
     tail_thread.kill if tail_thread
     puts 'Exiting'
   end
-end
\ No newline at end of file
+end
