--- association_proxy.old	2009-04-06 19:29:38.000000000 -0300
+++ association_proxy.rb	2009-04-06 19:31:13.000000000 -0300
@@ -178,7 +178,7 @@
         def set_belongs_to_association_for(record)
           if @reflection.options[:as]
             record["#{@reflection.options[:as]}_id"]   = @owner.id unless @owner.new_record?
-            record["#{@reflection.options[:as]}_type"] = @owner.class.base_class.name.to_s
+            record["#{@reflection.options[:as]}_type"] = @owner.class.name.to_s
           else
             unless @owner.new_record?
               primary_key = @reflection.options[:primary_key] || :id
