From c7e428c34d490d22e255636398a02e4fb49f9023 Mon Sep 17 00:00:00 2001
From: Steven Hancock <stevenh512@gmail.com>
Date: Tue, 20 Jul 2010 21:54:36 -0700
Subject: [PATCH] Bump bundler dependency to 1.0.0.beta.7

---
 rails.gemspec |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/rails.gemspec b/rails.gemspec
index e3f8119..9320466 100644
--- a/rails.gemspec
+++ b/rails.gemspec
@@ -25,5 +25,5 @@ Gem::Specification.new do |s|
   s.add_dependency('activeresource', version)
   s.add_dependency('actionmailer',   version)
   s.add_dependency('railties',       version)
-  s.add_dependency('bundler',        '>= 1.0.0.beta.5')
+  s.add_dependency('bundler',        '>= 1.0.0.beta.7')
 end
-- 
1.7.1.1

