From c01c2c1113f56f94df743bc9c1ccdb7213e086bc Mon Sep 17 00:00:00 2001
From: Sam Granieri <sam@samgranieri.com>
Date: Wed, 16 Apr 2008 08:45:57 -0500
Subject: [PATCH] modifying the cleanlog.sh to clean up after the activeresource debug log

---
 cleanlogs.sh |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/cleanlogs.sh b/cleanlogs.sh
index a197441..a4e6baf 100755
--- a/cleanlogs.sh
+++ b/cleanlogs.sh
@@ -1 +1 @@
-rm activerecord/debug.log activerecord/test/debug.log actionpack/debug.log
+rm activerecord/debug.log activerecord/test/debug.log actionpack/debug.log activeresource/test/debug.log
-- 
1.5.5

