From 29bda20244057a259ee309aa98a330ad9af0f2e7 Mon Sep 17 00:00:00 2001
From: Tekin Suleyman <tekin@tekin.co.uk>
Date: Thu, 29 May 2008 18:40:30 +0100
Subject: [PATCH] Fixed typo that was stopping plugins from installing when installing over SSH (e.g. ./script/plugin install ssh://[user@]host.xz/path/to/repo.git/)

---
 railties/lib/commands/plugin.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/commands/plugin.rb b/railties/lib/commands/plugin.rb
index d12d002..105819c 100644
--- a/railties/lib/commands/plugin.rb
+++ b/railties/lib/commands/plugin.rb
@@ -163,7 +163,7 @@ class Plugin
   end
   
   def git_url?
-    @uri =~ /^git:\/\// || @url =~ /\.git$/
+    @uri =~ /^git:\/\// || @uri =~ /\.git$/
   end
   
   def installed?
-- 
1.5.5.1

