From f2eb0ec0ab1e5ee3ccf2cadf920430c540c2569b Mon Sep 17 00:00:00 2001
From: Miles Georgi <azimux@gmail.com>
Date: Sat, 19 Jul 2008 16:04:35 -0700
Subject: [PATCH] changed regular expression in RecursiveHTTPFetcher#ls to match svn: or svn+ssh:

---
 railties/lib/commands/plugin.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/commands/plugin.rb b/railties/lib/commands/plugin.rb
index 0256090..980244a 100644
--- a/railties/lib/commands/plugin.rb
+++ b/railties/lib/commands/plugin.rb
@@ -907,7 +907,7 @@ class RecursiveHTTPFetcher
 
   def ls
     @urls_to_fetch.collect do |url|
-      if url =~ /^svn:\/\/.*/
+      if url =~ /^svn(\+ssh)?:\/\/.*/
         `svn ls #{url}`.split("\n").map {|entry| "/#{entry}"} rescue nil
       else
         open(url) do |stream|
-- 
1.5.4.5

