diff --git a/railties/lib/commands/dbconsole.rb b/railties/lib/commands/dbconsole.rb
index 6ff895a..06848d3 100644
--- a/railties/lib/commands/dbconsole.rb
+++ b/railties/lib/commands/dbconsole.rb
@@ -41,7 +41,7 @@ when "mysql"
 
   if config['password'] && include_password
     args << "--password=#{config['password']}"
-  elsif config['password'] && !config['password'].empty?
+  elsif config['password'] && !config['password'].to_s.empty?
     args << "-p"
   end
 
