From 72872101fb20ce741b9df6139364877b4a403094 Mon Sep 17 00:00:00 2001
From: Darragh Curran <darragh@peelmeagrape.net>
Date: Mon, 14 Jul 2008 13:44:57 +0100
Subject: [PATCH] add warning that we're not going to lad observers or initializers because rails couldn't load a gem dependancy

---
 railties/lib/initializer.rb |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index 18bcf69..3bd4911 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -271,6 +271,7 @@ module Rails
             puts " - #{gem.name}"
           end
           puts %{Run "rake gems:install" to install them.}
+          puts %{We're not going to load initializers or observers until you fix this!}
         end
       else
         @gems_dependencies_loaded = true
-- 
1.5.5.1

