From 5398c5754f436759c276aa3877c7351f2c2709a3 Mon Sep 17 00:00:00 2001
From: Miles Georgi <azimux@gmail.com>
Date: Thu, 7 Aug 2008 03:33:37 -0700
Subject: [PATCH] postgresql does not seem to like single quoted role names for the OWNER clause

---
 .../connection_adapters/postgresql_adapter.rb      |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
index 6a20f41..13f918f 100644
--- a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
@@ -534,7 +534,7 @@ module ActiveRecord
         option_string = options.symbolize_keys.sum do |key, value|
           case key
           when :owner
-            " OWNER = '#{value}'"
+            " OWNER = #{value}"
           when :template
             " TEMPLATE = #{value}"
           when :encoding
-- 
1.5.6.1.1071.g76fb

