--- asset_tag_helper.rb-orig	2010-06-08 20:00:22.857105431 -0700
+++ asset_tag_helper.rb	2010-06-08 20:00:53.737607424 -0700
@@ -745,8 +745,8 @@
           source = rewrite_asset_path(source, config.asset_path)
 
           has_request = controller.respond_to?(:request)
-          if has_request && include_host && source !~ %r{^#{controller.config.relative_url_root}/}
-            source = "#{controller.config.relative_url_root}#{source}"
+          if has_request && include_host && source !~ %r{^#{ENV["RAILS_RELATIVE_URL_ROOT"]}/}
+            source = "#{ENV["RAILS_RELATIVE_URL_ROOT"]}#{source}"
           end
           source = rewrite_host_and_protocol(source, has_request) if include_host
 
