From d3a1461e6ae20b9f14ac2cfe305cf227dbcccec4 Mon Sep 17 00:00:00 2001
From: Micah Wedemeyer <mwedemeyer@micah.(none)>
Date: Sun, 25 May 2008 17:57:24 -0400
Subject: [PATCH] Updated MysqlAdapter comments to list the sslca option.

---
 .../connection_adapters/mysql_adapter.rb           |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
index f94b0f2..7728685 100755
--- a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
@@ -153,6 +153,7 @@ module ActiveRecord
     # * <tt>:password</tt> - Defaults to nothing.
     # * <tt>:database</tt> - The name of the database. No default, must be provided.
     # * <tt>:encoding</tt> - (Optional) Sets the client encoding by executing "SET NAMES <encoding>" after connection.
+    # * <tt>:sslca</tt> - Necessary to use MySQL with an SSL connection.
     # * <tt>:sslkey</tt> - Necessary to use MySQL with an SSL connection.
     # * <tt>:sslcert</tt> - Necessary to use MySQL with an SSL connection.
     # * <tt>:sslcapath</tt> - Necessary to use MySQL with an SSL connection.
-- 
1.5.2.5

