--- base.rb
+++ (clipboard)
@@ -338,7 +338,7 @@
         if block_given?
           begin
             @_proc_for_layout = block
-            concat(render(options.merge(:partial => partial_layout)))
+            render(options.merge(:partial => partial_layout))
           ensure
             @_proc_for_layout = nil
           end
@@ -346,12 +346,12 @@
           begin
             original_content_for_layout = @content_for_layout if defined?(@content_for_layout)
             @content_for_layout = render(options)
-
+           
             if (options[:inline] || options[:file] || options[:text])
               @cached_content_for_layout = @content_for_layout
               render(:file => partial_layout, :locals => local_assigns)
             else
-              render(options.merge(:partial => partial_layout))
+              render(options.merge(:partial => partial_layout).except(:collection))
             end
           ensure
             @content_for_layout = original_content_for_layout
@@ -359,4 +359,4 @@
         end
       end
   end
-end
\ No newline at end of file
+end
