From 78cd34a3e54144d7613834c5f61447509f95438e Mon Sep 17 00:00:00 2001
From: Gaspard Bucher <gaspard@teti.ch>
Date: Tue, 10 Feb 2009 12:58:32 +0100
Subject: [PATCH] Fixes a typo in initializer.rb producing error: undefined local variable or method `view_path'.

---
 railties/lib/initializer.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index 11aa8a5..3914006 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -371,7 +371,7 @@ Run `rake gems:install` to install the missing gems.
       if configuration.frameworks.include?(:action_view)
         if configuration.cache_classes
           ActionController::Base.view_paths = configuration.view_path if configuration.frameworks.include?(:action_controller)
-          ActionMailer::Base.template_root = view_path if configuration.frameworks.include?(:action_mailer)
+          ActionMailer::Base.template_root = configuration.view_path if configuration.frameworks.include?(:action_mailer)
         end
       end
     end
-- 
1.6.0

