From 015099dea3592c629b05f3d3154f4d721bf1113d Mon Sep 17 00:00:00 2001
From: Alex Eagle <eagle.alex@gmail.com>
Date: Mon, 12 May 2008 12:24:17 +0400
Subject: [PATCH] =?utf-8?q?Fix=20'PGError:=20ERROR:=20schema=20=E2=80=9Cmy=5Fschema=E2=80=9D=20does=20not=20exist=20:=20SET=20search=5Fpath=20TO=20my=5Fschema'=20on=20db:test:purge?=
MIME-Version: 1.0
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: 8bit

---
 .../connection_adapters/postgresql_adapter.rb      |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
index 2ec2d80..fccdca1 100644
--- a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
@@ -856,7 +856,7 @@ module ActiveRecord
             end
           end
           self.client_min_messages = @config[:min_messages] if @config[:min_messages]
-          self.schema_search_path = @config[:schema_search_path] || @config[:schema_order]
+          self.schema_search_path = @config[:schema_search_path] || @config[:schema_order] unless @config[:database] == ‘template1’
         end
 
         # Returns the current ID of a table's sequence.
-- 
1.5.4.3

