diff --git a/ci/ci_build.rb b/ci/ci_build.rb
index 7b9cdce..eb9539c 100755
--- a/ci/ci_build.rb
+++ b/ci/ci_build.rb
@@ -23,6 +23,7 @@ cd "#{root_dir}/activesupport" do
   build_results[:activesupport] = system 'rake'
 end
 
+rm_f "#{root_dir}/activerecord/debug.log"
 cd "#{root_dir}/activerecord" do
   puts
   puts "[CruiseControl] Building ActiveRecord with MySQL"
@@ -59,6 +60,7 @@ cd "#{root_dir}/activemodel" do
   build_results[:activemodel] = system 'rake'
 end
 
+rm_f "#{root_dir}/activeresource/debug.log"
 cd "#{root_dir}/activeresource" do
   puts
   puts "[CruiseControl] Building ActiveResource"
diff --git a/ci/ci_setup_notes.txt b/ci/ci_setup_notes.txt
index 86df33c..baf7cdc 100644
--- a/ci/ci_setup_notes.txt
+++ b/ci/ci_setup_notes.txt
@@ -54,10 +54,14 @@ ci      ALL=NOPASSWD: /usr/local/bin/geminstaller, /usr/local/bin/ruby, /usr/loc
 * Install/setup nginx:
 $ sudo aptitude install nginx
 $ sudo vi /etc/nginx/sites-available/default
+# Change server_name entry to match server name
+
 # comment two lines and add one to proxy to ccrb:
 #               root   /var/www/nginx-default;
 #               index  index.html index.htm;
                 proxy_pass      http://127.0.0.1:3333;
+                
+# also comment default locations for /doc and /images
 $ sudo /etc/init.d/nginx start
 
 * Add project to cruise (It will still fail until everything is set up):
@@ -101,6 +105,13 @@ $ sudo aptitude install sqlite sqlite3 libsqlite-dev libsqlite3-dev
 $ sudo aptitude install postgresql postgresql-server-dev-8.3
 $ sudo su - postgres -c 'createuser -s ci'
 
+* Install fcgi libraries
+$ sudo apt-get install libfcgi-dev
+
+* Install memcached and start for first time (should start on reboot automatically)
+$ sudo aptitude install memcached
+$ sudo /etc/init.d/memcached start
+
 * Install and run GemInstaller to get all dependency gems
 $ sudo gem install geminstaller
 $ cd ~/.cruise/projects/rails/work
diff --git a/ci/cruise_config.rb b/ci/cruise_config.rb
index ec1c82a..7985e3c 100644
--- a/ci/cruise_config.rb
+++ b/ci/cruise_config.rb
@@ -1,5 +1,6 @@
 Project.configure do |project|
  project.build_command = 'ruby ci/ci_build.rb'
- project.email_notifier.emails = ['thewoolleyman@gmail.com','michael@koziarski.com']
+ project.email_notifier.emails = ['thewoolleyman@gmail.com']
+# project.email_notifier.emails = ['thewoolleyman@gmail.com','michael@koziarski.com', 'david@loudthinking.com', 'jeremy@bitsweat.net', 'josh@joshpeek.com', 'pratiknaik@gmail.com', 'wycats@gmail.com']
  project.email_notifier.from = 'thewoolleyman+railsci@gmail.com'
 end
diff --git a/ci/geminstaller.yml b/ci/geminstaller.yml
index 3a1862c..d0f5787 100644
--- a/ci/geminstaller.yml
+++ b/ci/geminstaller.yml
@@ -2,13 +2,19 @@
 gems:
 - name: geminstaller
   version: >= 0.4.3
+- name: fcgi
+  version: >= 0.8.7
+- name: memcache-client
+  version: >= 1.5.0
 - name: mocha
-  version: >= 0.9.0
+  version: >= 0.9.4
 - name: mysql
   #version: >= 2.7
   version: = 2.7
 - name: postgres
   version: >= 0.7.9.2008.01.28
+- name: rack
+  version: '~> 0.4.0'
 - name: rake
   version: >= 0.8.1
 - name: sqlite-ruby
