Index: lib/active_record/base.rb
===================================================================
--- lib/active_record/base.rb   (revision 79)
+++ lib/active_record/base.rb   (working copy)
@@ -1514,8 +1514,11 @@
           scope = scope(:find) if :auto == scope
           scoped_order = scope[:order] if scope
           if order
-            sql << " ORDER BY #{order}"
-            sql << ", #{scoped_order}" if scoped_order
+            if scoped_order
+              sql << " ORDER BY #{scoped_order}, #{order}"
+            else
+              sql << " ORDER BY #{order}"
+            end
           else
             sql << " ORDER BY #{scoped_order}" if scoped_order
           end

