diff --git a/actionmailer/test/quoting_test.rb b/actionmailer/test/quoting_test.rb
index 13a859a..d2a7a1f 100644
--- a/actionmailer/test/quoting_test.rb
+++ b/actionmailer/test/quoting_test.rb
@@ -58,6 +58,7 @@ class QuotingTest < Test::Unit::TestCase
 
     unquoted = TMail::Unquoter.unquote_and_convert_to(result, nil)
     assert_equal unquoted, original
+    assert_no_match %r/=[0-9a-f]{2}/, result
   end
 
 
diff --git a/actionmailer/test/test_helper_test.rb b/actionmailer/test/test_helper_test.rb
index f8913e5..30f000b 100644
--- a/actionmailer/test/test_helper_test.rb
+++ b/actionmailer/test/test_helper_test.rb
@@ -36,7 +36,7 @@ class TestHelperMailerTest < ActionMailer::TestCase
   end
 
   def test_encode
-    assert_equal "=?utf-8?Q?=0aasdf=0a?=", encode("\nasdf\n")
+    assert_equal "=?utf-8?Q?=0Aasdf=0A?=", encode("\nasdf\n")
   end
 
   def test_assert_emails
