--- /home/esben/linova/linova-web-frontend/vendor/rails/activerecord/lib/active_record/validations.rb.orig	2009-06-04 16:16:33.000000000 +0200
+++ /home/esben/linova/linova-web-frontend/vendor/rails/activerecord/lib/active_record/validations.rb	2009-06-04 16:16:51.000000000 +0200
@@ -947,7 +947,15 @@
         end
 
         validates_each(attr_names,configuration) do |record, attr_name, value|
-          raw_value = record.send("#{attr_name}_before_type_cast") || value
+          raw_value = record.send("#{attr_name}_before_type_cast") 
+          raw_value =
+          case raw_value
+            when 'Infinity' then 1.0/0.0
+            when '-Infinity' then -1.0/0.0
+            when 'NaN' then 0.0/0.0
+            when nil then value
+            else raw_value
+          end
 
           next if configuration[:allow_nil] and raw_value.nil?
 
