From eca681c59eb08f72202c134e83d3b7566f3ee2c1 Mon Sep 17 00:00:00 2001
From: Nathaniel Bibler <git@nathanielbibler.com>
Date: Fri, 29 Aug 2008 23:36:16 -0400
Subject: [PATCH] Added optional rake doc:app TITLE environment parameter

---
 railties/lib/tasks/documentation.rake |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/railties/lib/tasks/documentation.rake b/railties/lib/tasks/documentation.rake
index f4927a0..4ef8386 100644
--- a/railties/lib/tasks/documentation.rake
+++ b/railties/lib/tasks/documentation.rake
@@ -1,9 +1,9 @@
 namespace :doc do
-  desc "Generate documentation for the application. Set custom template with TEMPLATE=/path/to/rdoc/template.rb"
+  desc "Generate documentation for the application. Set custom template with TEMPLATE=/path/to/rdoc/template.rb or title with TITLE=\"Custom Title\""
   Rake::RDocTask.new("app") { |rdoc|
     rdoc.rdoc_dir = 'doc/app'
     rdoc.template = ENV['template'] if ENV['template']
-    rdoc.title    = "Rails Application Documentation"
+    rdoc.title    = ENV['title'] || "Rails Application Documentation"
     rdoc.options << '--line-numbers' << '--inline-source'
     rdoc.options << '--charset' << 'utf-8'
     rdoc.rdoc_files.include('doc/README_FOR_APP')
-- 
1.5.5.1

