From 1a37897dd2fef8c171510fa063ec17b40f3e9444 Mon Sep 17 00:00:00 2001
From: Joe Van Dyk <joe@fixieconsulting.com>
Date: Sat, 26 Sep 2009 16:46:27 -0700
Subject: [PATCH 1/2] Mocha is required for most of the tests

---
 activerecord/Rakefile |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/activerecord/Rakefile b/activerecord/Rakefile
index 1d8062e..8c52e81 100644
--- a/activerecord/Rakefile
+++ b/activerecord/Rakefile
@@ -1,5 +1,6 @@
 require 'rubygems'
 require 'rake'
+require 'mocha'
 require 'rake/testtask'
 require 'rake/rdoctask'
 require 'rake/packagetask'
-- 
1.6.3.2


From ce8bc5639a94dbb2a40fda411d34234dcb130276 Mon Sep 17 00:00:00 2001
From: Joe Van Dyk <joe@fixieconsulting.com>
Date: Sat, 26 Sep 2009 16:52:59 -0700
Subject: [PATCH 2/2] Broken test for showing that having a column named 'field' breaks ActiveRecord

---
 activerecord/test/schema/schema.rb |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/activerecord/test/schema/schema.rb b/activerecord/test/schema/schema.rb
index 15e5e12..3d62143 100644
--- a/activerecord/test/schema/schema.rb
+++ b/activerecord/test/schema/schema.rb
@@ -33,6 +33,7 @@ ActiveRecord::Schema.define do
 
   create_table :authors, :force => true do |t|
     t.string :name, :null => false
+    t.string :field
     t.integer :author_address_id
     t.integer :author_address_extra_id
   end
-- 
1.6.3.2

