From f90197d64dc480e68054d45cbc6fb8d8d204c320 Mon Sep 17 00:00:00 2001
From: Nigel Ramsay <nigelramsay@nigel-ramsays-macbook.local>
Date: Wed, 3 Sep 2008 15:57:06 +1200
Subject: [PATCH] Inline help text was incorrectly telling user to uncomment line to use default local time. User should comment the line to use default local time.

---
 railties/environments/environment.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/environments/environment.rb b/railties/environments/environment.rb
index a85ade3..2140dfe 100644
--- a/railties/environments/environment.rb
+++ b/railties/environments/environment.rb
@@ -40,7 +40,7 @@ Rails::Initializer.run do |config|
 
   # Make Time.zone default to the specified zone, and make Active Record store time values
   # in the database in UTC, and return them converted to the specified local zone.
-  # Run "rake -D time" for a list of tasks for finding time zone names. Uncomment to use default local time.
+  # Run "rake -D time" for a list of tasks for finding time zone names. Comment line to use default local time.
   config.time_zone = 'UTC'
 
   # Your secret key for verifying cookie session data integrity.
-- 
1.5.6.4

