From c3f0fd38eb8b1bcaeddbea3da391f2fcf7273f99 Mon Sep 17 00:00:00 2001
From: Jaroslaw Zabiello <hipertracker@gmail.com>
Date: Mon, 24 May 2010 18:41:01 +0100
Subject: [PATCH] Enforced UTF-8 encoding for Ruby 1.9 and MySQL text fields

---
 .../connection_adapters/mysql_adapter.rb           |   24 ++++++++++++-------
 1 files changed, 15 insertions(+), 9 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
index ec25bbf..aa2bab6 100644
--- a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
@@ -5,7 +5,7 @@ require 'set'
 
 module MysqlCompat #:nodoc:
   # add all_hashes method to standard mysql-c bindings or pure ruby version
-  def self.define_all_hashes_method!
+  def self.define_all_hashes_method!(config)
     raise 'Mysql not loaded' unless defined?(::Mysql)
 
     target = defined?(Mysql::Result) ? Mysql::Result : MysqlRes
@@ -15,13 +15,19 @@ module MysqlCompat #:nodoc:
     # Ruby driver has a version string and returns null values in each_hash
     # C driver >= 2.7 returns null values in each_hash
     if Mysql.const_defined?(:VERSION) && (Mysql::VERSION.is_a?(String) || Mysql::VERSION >= 20700)
-      target.class_eval <<-'end_eval', __FILE__, __LINE__ + 1
-        def all_hashes                     # def all_hashes
-          rows = []                        #   rows = []
-          each_hash { |row| rows << row }  #   each_hash { |row| rows << row }
-          rows                             #   rows
-        end                                # end
-      end_eval
+      code = '
+        def all_hashes                     
+          rows = []                        
+          each_hash do |row| 
+             row.each_pair do |k,v| 
+               row[k] = v.force_encoding("UTF-8") if v.instance_of?(String) && v.encoding.name == "ASCII-8BIT"
+             end if ["", "utf8"].include?("%s") && RUBY_VERSION =~ /1.9/
+            rows << row 
+          end 
+          rows                             
+        end                                
+        ' % config['encoding']
+     target.class_eval code , __FILE__, __LINE__ + 1
 
     # adapters before 2.7 don't have a version constant
     # and don't return null values in each_hash
@@ -67,7 +73,7 @@ module ActiveRecord
         end
       end
 
-      MysqlCompat.define_all_hashes_method!
+      MysqlCompat.define_all_hashes_method! config
 
       mysql = Mysql.init
       mysql.ssl_set(config[:sslkey], config[:sslcert], config[:sslca], config[:sslcapath], config[:sslcipher]) if config[:sslca] || config[:sslkey]
-- 
1.7.1

