diff --git a/activerecord/lib/active_record/calculations.rb b/activerecord/lib/active_record/calculations.rb
index 65512d5..fe742d4 100644
--- a/activerecord/lib/active_record/calculations.rb
+++ b/activerecord/lib/active_record/calculations.rb
@@ -192,7 +192,14 @@ module ActiveRecord
 
           if merged_includes.any?
             join_dependency = ActiveRecord::Associations::ClassMethods::JoinDependency.new(self, merged_includes, joins)
-            sql << join_dependency.join_associations.collect{|join| join.association_join }.join
+
+            #  copy strategy from associations.rb construct_finder_sql_for_association_limiting...
+            #  only include :include joins from tables required by the conditions.
+            tables_from_conditions = conditions_tables(options)
+            distinct_join_associations = tables_from_conditions.uniq.map{|table|
+              join_dependency.joins_for_table_name(table)
+            }.flatten.compact.uniq
+            sql << distinct_join_associations.collect{|assoc| assoc.association_join}.join 
           end
 
           sql << joins unless joins.blank?
