From c2007f71b13526f0c6bdfe44e358973ce0120daf Mon Sep 17 00:00:00 2001
From: gmile <iamexile@gmail.com>
Date: Sat, 12 Feb 2011 14:27:46 +0200
Subject: [PATCH] Generate 'IS NULL' instead of IN (NULL)

---
 lib/arel/predications.rb |   12 +++++++++++-
 1 files changed, 11 insertions(+), 1 deletions(-)

diff --git a/lib/arel/predications.rb b/lib/arel/predications.rb
index 58f02a2..49741df 100644
--- a/lib/arel/predications.rb
+++ b/lib/arel/predications.rb
@@ -41,7 +41,17 @@ module Arel
           Nodes::Between.new(self, Nodes::And.new([other.begin, other.end]))
         end
       else
-        Nodes::In.new self, other
+        if other.include?(nil)
+          if other.size > 1
+            set  = Nodes::In.new self, other.compact
+            null = Nodes::Equality.new self, nil
+            Nodes::Or.new set, null
+          else
+            Nodes::Equality.new self, nil
+          end
+        else
+          Nodes::In.new self, other
+        end
       end
     end
 
-- 
1.7.1

