From e28bf25393589ca1e735e3d089ca873dfd6186de Mon Sep 17 00:00:00 2001
From: Henrik N <henrik@nyh.se>
Date: Mon, 12 May 2008 22:01:55 +0200
Subject: [PATCH] 'rake gem:install' should be 'rake gems:install' in environment.rb template.

---
 railties/environments/environment.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/environments/environment.rb b/railties/environments/environment.rb
index c33b80d..1e6b4b4 100644
--- a/railties/environments/environment.rb
+++ b/railties/environments/environment.rb
@@ -21,7 +21,7 @@ Rails::Initializer.run do |config|
   # config.frameworks -= [ :active_record, :active_resource, :action_mailer ]
 
   # Specify gems that this application depends on. 
-  # They can then be installed with rake gem:install on new installations.
+  # They can then be installed with rake gems:install on new installations.
   # config.gem "bj"
   # config.gem "hpricot", :version => '0.6', :source => "http://code.whytheluckystiff.net"
   # config.gem "aws-s3", :lib => "aws/s3"
-- 
1.5.4.4

