From 561d4ca64891eb94902a0e480bc1f39e4fd865e2 Mon Sep 17 00:00:00 2001
From: Tim Peters <tim.peters@adelaide.edu.au>
Date: Wed, 10 Dec 2008 16:50:06 +1030
Subject: [PATCH] use table prefix and suffix when creating index unique_schema_migrations

---
 .../abstract/schema_statements.rb                  |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
index c29c156..f12235b 100644
--- a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
+++ b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
@@ -321,7 +321,7 @@ module ActiveRecord
             schema_migrations_table.column :version, :string, :null => false
           end
           add_index sm_table, :version, :unique => true,
-            :name => 'unique_schema_migrations'
+            :name => Base.table_name_prefix + 'unique_schema_migrations' + Base.table_name_suffix
 
           # Backwards-compatibility: if we find schema_info, assume we've
           # migrated up to that point:
-- 
1.5.4.3

