=== modified file 'vendor/rails/actionpack/test/controller/render_test.rb'
--- vendor/rails/actionpack/test/controller/render_test.rb	2009-04-03 13:23:10 +0000
+++ vendor/rails/actionpack/test/controller/render_test.rb	2009-04-30 21:15:10 +0000
@@ -193,6 +193,10 @@
     render :inline =>  "<%= controller_name %>"
   end
 
+  def render_json_nil
+    render :json => nil
+  end
+
   def render_json_hello_world
     render :json => {:hello => 'world'}.to_json
   end
@@ -886,6 +890,12 @@
     assert_equal "The secret is in the sauce\n", @response.body
   end
 
+  def test_render_json_nil
+    get :render_json_nil
+    assert_equal 'null', @response.body
+    assert_equal 'application/json', @response.content_type
+  end
+
   def test_render_json
     get :render_json_hello_world
     assert_equal '{"hello": "world"}', @response.body

