diff --git a/actionpack/lib/action_dispatch/middleware/session/abstract_store.rb b/actionpack/lib/action_dispatch/middleware/session/abstract_store.rb
index 6c039cf..0c7dca5 100644
--- a/actionpack/lib/action_dispatch/middleware/session/abstract_store.rb
+++ b/actionpack/lib/action_dispatch/middleware/session/abstract_store.rb
@@ -34,6 +34,11 @@ module ActionDispatch
           super
         end
 
+        def delete(key)
+          load! unless @loaded
+          super
+        end
+
         def to_hash
           h = {}.replace(self)
           h.delete_if { |k,v| v.nil? }
