--- /usr/local/lib/ruby/gems/1.8/gems/actionpack-2.1.0/lib/action_controller/vendor/html-scanner/html/node.rb	2008-07-04 17:02:24.000000000 +0900
+++ node.rb	2008-09-02 14:32:19.000000000 +0900
@@ -166,15 +166,6 @@
               if scanner.scan(/\s*=\s*/)
                 if delim = scanner.scan(/['"]/)
-                  value = ""
-                  while text = scanner.scan(/[^#{delim}\\]+|./)
-                    case text
-                      when "\\" then
-                        value << text
-                        value << scanner.getch
-                      when delim
-                        break
-                      else value << text
-                    end
-                  end
+                  value = scanner.scan(/[^#{delim}]+/)
+                  scanner.getch
                 else
                   value = scanner.scan(/[^\s>\/]+/)
