From 231e79f40dbf11dfaa6b83ac81399237b1096578 Mon Sep 17 00:00:00 2001
From: John Aughey <jha@aughey.com>
Date: Wed, 11 Feb 2009 21:05:46 -0600
Subject: [PATCH] Saving the config in SQLite3Adapter for ThinkingSphinx

---
 .../connection_adapters/sqlite3_adapter.rb         |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb b/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb
index cc9c465..dc11d24 100644
--- a/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb
@@ -18,12 +18,16 @@ module ActiveRecord
 
       db.busy_timeout(config[:timeout]) unless config[:timeout].nil?
 
-      ConnectionAdapters::SQLite3Adapter.new(db, logger)
+      ConnectionAdapters::SQLite3Adapter.new(db, logger, config)
     end
   end
 
   module ConnectionAdapters #:nodoc:
     class SQLite3Adapter < SQLiteAdapter # :nodoc:
+      def initialize(connection, logger, config)
+        super(connection, logger)
+        @config = config
+      end
       def table_structure(table_name)
         returning structure = @connection.table_info(quote_table_name(table_name)) do
           raise(ActiveRecord::StatementInvalid, "Could not find table '#{table_name}'") if structure.empty?
-- 
1.6.0.5

