diff --git a/activerecord/test/cases/adapters/postgresql/schema_test.rb b/activerecord/test/cases/adapters/postgresql/schema_test.rb
index a5c3e69..100b08b 100644
--- a/activerecord/test/cases/adapters/postgresql/schema_test.rb
+++ b/activerecord/test/cases/adapters/postgresql/schema_test.rb
@@ -164,6 +164,23 @@ class SchemaTest < ActiveRecord::TestCase
     ActiveRecord::Base.connection.schema_search_path = "public"
   end
 
+  def object_does_not_update_when_saved_under_a_schema
+    t = Thing1.new
+    t.save
+    assert !t.changed?
+    assert_equal [], t.changed
+    t.name = "postgresql"
+    t.email = "rails@postgresql.org"
+    time = Time.now
+    t.moment = time
+    t.save
+    assert t.changed? # should work
+    assert_equal ["nil", "postgresql", "rails@postgresql.org", "#{time}"], t.changed # should be wrong!
+    assert_equal ["1", "postgresql", "rails@postgresql.org", "#{time}"], Thing1.first # works
+    assert_equal ["1", "postgresql", "rails@postgresql.org", "#{time}"], t.changed # should work!
+  end
+
+
   private
     def columns(table_name)
       @connection.send(:column_definitions, table_name).map do |name, type, default|
