From 7854ab1d9e7efab3c99b4f66da9301014f0ddf33 Mon Sep 17 00:00:00 2001
From: Bryan Larsen <bryan@larsen.st>
Date: Thu, 17 Dec 2009 17:27:38 -0600
Subject: [PATCH] hack fix for rails bug #3595

---
 railties/lib/tasks/databases.rake |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index ebd3a8f..bb7d4ac 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -360,7 +360,7 @@ namespace :db do
       case abcs["test"]["adapter"]
       when /mysql/
         ActiveRecord::Base.establish_connection(:test)
-        ActiveRecord::Base.connection.recreate_database(abcs["test"]["database"], abcs["test"])
+        ActiveRecord::Base.connection.recreate_database(abcs["test"]["database"])
       when /postgresql/
         ActiveRecord::Base.clear_active_connections!
         drop_database(abcs['test'])
-- 
1.6.3.3

