From cf02a2b8e83913e98477facbe7788db7e0b6a8e8 Mon Sep 17 00:00:00 2001
From: Matt Jones <al2o3cr@gmail.com>
Date: Wed, 8 Oct 2008 17:00:38 -0400
Subject: [PATCH] Fix console --sandbox

---
 railties/lib/console_sandbox.rb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/railties/lib/console_sandbox.rb b/railties/lib/console_sandbox.rb
index 5d57679..65a3d68 100644
--- a/railties/lib/console_sandbox.rb
+++ b/railties/lib/console_sandbox.rb
@@ -1,6 +1,6 @@
-ActiveRecord::Base.send :increment_open_transactions
+ActiveRecord::Base.connection.increment_open_transactions
 ActiveRecord::Base.connection.begin_db_transaction
 at_exit do
   ActiveRecord::Base.connection.rollback_db_transaction
-  ActiveRecord::Base.send :decrement_open_transactions
+  ActiveRecord::Base.connection.decrement_open_transactions
 end
-- 
1.5.3.1

