From 2206a2c1f3f2cb116e7edef4e2cb94ef6637d920 Mon Sep 17 00:00:00 2001
From: Andrew White <andyw@pixeltrix.co.uk>
Date: Mon, 29 Mar 2010 05:10:42 +0100
Subject: [PATCH] Make Railties RDoc task work again

---
 Rakefile          |    4 ++--
 railties/Rakefile |    8 ++------
 2 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/Rakefile b/Rakefile
index cae9b04..2fa1f75 100644
--- a/Rakefile
+++ b/Rakefile
@@ -72,8 +72,8 @@ Rake::RDocTask.new do |rdoc|
   rdoc.rdoc_files.include('railties/CHANGELOG')
   rdoc.rdoc_files.include('railties/MIT-LICENSE')
   rdoc.rdoc_files.include('railties/README')
-  rdoc.rdoc_files.include('railties/lib/{*.rb,commands/*.rb,rails/*.rb,generators/*.rb}')
-  rdoc.rdoc_files.exclude('railties/lib/vendor/*')
+  rdoc.rdoc_files.include('railties/lib/**/*.rb')
+  rdoc.rdoc_files.exclude('railties/lib/rails/generators/**/templates/*')
 
   rdoc.rdoc_files.include('activerecord/README')
   rdoc.rdoc_files.include('activerecord/CHANGELOG')
diff --git a/railties/Rakefile b/railties/Rakefile
index 6368456..d88036f 100644
--- a/railties/Rakefile
+++ b/railties/Rakefile
@@ -1,5 +1,3 @@
-require File.expand_path('../../load_paths', __FILE__)
-
 require 'rake'
 require 'rake/testtask'
 require 'rake/rdoctask'
@@ -60,10 +58,8 @@ Rake::RDocTask.new { |rdoc|
   rdoc.options << '--charset' << 'utf-8'
   rdoc.template = ENV['template'] ? "#{ENV['template']}.rb" : '../doc/template/horo'
   rdoc.rdoc_files.include('README', 'CHANGELOG')
-  rdoc.rdoc_files.include('lib/*.rb')
-  rdoc.rdoc_files.include('lib/rails/*.rb')
-  rdoc.rdoc_files.include('lib/rails/generators/*.rb')
-  rdoc.rdoc_files.include('lib/rails/commands/**/*.rb')
+  rdoc.rdoc_files.include('lib/**/*.rb')
+  rdoc.rdoc_files.exclude('lib/rails/generators/**/templates/*')
 }
 
 # Generate GEM ----------------------------------------------------------------------------
-- 
1.6.4.4

