From a1cd9162c787ffc9c8f927e46b2b986e30cb6ca5 Mon Sep 17 00:00:00 2001
From: Felipe Kaufmann <felipekaufmann@me.com>
Date: Tue, 2 Jun 2009 19:31:59 +0200
Subject: [PATCH] modified db:sessions:clear to use configured sessions table name

---
 railties/lib/tasks/databases.rake |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index 0e25673..9d81efa 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -437,7 +437,7 @@ def drop_database(config)
 end
 
 def session_table_name
-  ActiveRecord::Base.pluralize_table_names ? :sessions : :session
+  ActiveRecord::SessionStore::Session.table_name
 end
 
 def set_firebird_env(config)
-- 
1.6.0.1

