diff --git a/actionpack/lib/action_controller/session/mem_cache_store.rb b/actionpack/lib/action_controller/session/mem_cache_store.rb
index f745715..3e3587e 100644
--- a/actionpack/lib/action_controller/session/mem_cache_store.rb
+++ b/actionpack/lib/action_controller/session/mem_cache_store.rb
@@ -19,9 +19,6 @@ begin
           unless @pool.servers.any? { |s| s.alive? }
             raise "#{self} unable to find server during initialization."
           end
-          @mutex = Mutex.new
-
-          super
         end
 
         private
