Index: vendor/rails/railties/lib/tasks/databases.rake
===================================================================
--- vendor/rails/railties/lib/tasks/databases.rake	(revision 0)
+++ vendor/rails/railties/lib/tasks/databases.rake	(revision 1)
@@ -85,8 +85,12 @@
       ActiveRecord::Base.configurations.each_value do |config|
         # Skip entries that don't have a database key
         next unless config['database']
-        # Only connect to local databases
-        local_database?(config) { drop_database(config) }
+        begin
+          # Only connect to local databases
+          local_database?(config) { drop_database(config) }
+        rescue Exception => e
+          puts "Couldn't drop #{config['database']} : #{e.inspect}"
+        end
       end
     end
   end
