From e1008dcdcd7e946b5f41289a972096087bc616a7 Mon Sep 17 00:00:00 2001
From: Roger Pack <rogerpack@roger-packs-macbook-pro.local>
Date: Fri, 23 May 2008 15:47:38 -0600
Subject: [PATCH] Made it so that on rails code based error messages, the filename is shown in red, so that people can identify more easily which file they need to go and edit

---
 .../templates/rescues/template_error.erb           |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/actionpack/lib/action_controller/templates/rescues/template_error.erb b/actionpack/lib/action_controller/templates/rescues/template_error.erb
index eda64db..00fe233 100644
--- a/actionpack/lib/action_controller/templates/rescues/template_error.erb
+++ b/actionpack/lib/action_controller/templates/rescues/template_error.erb
@@ -4,7 +4,7 @@
 </h1>
 
 <p>
-  Showing <i><%=h @exception.file_name %></i> where line <b>#<%=h @exception.line_number %></b> raised:
+  Showing <i style="color: #500;"><%=h @exception.file_name %></i> where line <b>#<%=h @exception.line_number %></b> raised:
   <pre><code><%=h @exception.message %></code></pre>
 </p>
 
-- 
1.5.5.1

