diff --git a/activerecord/lib/active_record/railties/databases.rake b/activerecord/lib/active_record/railties/databases.rake
index 2c17c74..f00a122 100644
--- a/activerecord/lib/active_record/railties/databases.rake
+++ b/activerecord/lib/active_record/railties/databases.rake
@@ -35,6 +35,28 @@ namespace :db do
     create_database(ActiveRecord::Base.configurations[Rails.env])
   end
 
+  class MySqlExceptionHelper
+    def initialize(excpt)
+      @excpt = excpt
+    end
+    
+    def access_denied?
+      case @excpt.class.name
+        when 'Mysql::Error' then (@excpt.errno == Mysql::Error::ER_ACCESS_DENIED_ERROR)   
+        when 'Mysql2::Error' then (@excpt.error_number == 1045)
+        else false
+      end
+    end
+    
+    def error
+      case @excpt.class.name
+        when 'Mysql::Error' then @excpt.error
+        when 'Mysql2::Error' then @excpt.to_s
+        else "Unknown mysql error"
+      end
+    end 
+  end  
+
   def create_database(config)
     begin
       if config['adapter'] =~ /sqlite/
@@ -65,8 +87,9 @@ namespace :db do
           ActiveRecord::Base.establish_connection(config.merge('database' => nil))
           ActiveRecord::Base.connection.create_database(config['database'], creation_options)
           ActiveRecord::Base.establish_connection(config)
-        rescue Mysql::Error => sqlerr
-          if sqlerr.errno == Mysql::Error::ER_ACCESS_DENIED_ERROR
+        rescue Exception => excpt
+          sqlerr = MySqlExceptionHelper.new(excpt)
+          if sqlerr.access_denied?
             print "#{sqlerr.error}. \nPlease provide the root password for your mysql installation\n>"
             root_password = $stdin.gets.strip
             grant_statement = "GRANT ALL PRIVILEGES ON #{config['database']}.* " \
