From d8d2f47cdc15bd63ff3c1977945fc0116084e86a Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Daniel=20Rodr=C3=ADguez=20Troiti=C3=B1o?= <drodrigueztroitino@yahoo.es>
Date: Mon, 4 Aug 2008 12:31:05 +0200
Subject: [PATCH] Added file name information for errors and exceptions in script/runnner

---
 railties/lib/commands/runner.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/commands/runner.rb b/railties/lib/commands/runner.rb
index 926bc26..14159c3 100644
--- a/railties/lib/commands/runner.rb
+++ b/railties/lib/commands/runner.rb
@@ -42,7 +42,7 @@ if code_or_file.nil?
   $stderr.puts "Run '#{$0} -h' for help."
   exit 1
 elsif File.exist?(code_or_file)
-  eval(File.read(code_or_file))
+  eval(File.read(code_or_file), nil, code_or_file)
 else
   eval(code_or_file)
 end
-- 
1.5.6.2

