From d9662f60e6712334cb85d67c0f4f82f2e7e89247 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Tarmo=20T=C3=A4nav?= <tarmo@itech.ee>
Date: Tue, 15 Jul 2008 05:24:24 +0300
Subject: [PATCH] Raise ActiveRecordError if sqlite rename_column is attempted on a non-existing column

---
 .../connection_adapters/sqlite_adapter.rb          |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb b/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
index f4d387c..399922a 100644
--- a/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/sqlite_adapter.rb
@@ -260,6 +260,9 @@ module ActiveRecord
       end
 
       def rename_column(table_name, column_name, new_column_name) #:nodoc:
+        unless columns(table_name).detect{|c| c.name == column_name.to_s }
+          raise ActiveRecord::ActiveRecordError, "Column '#{column_name}' on table '#{table_name}', does not exist"
+        end
         alter_table(table_name, :rename => {column_name.to_s => new_column_name.to_s})
       end
 
-- 
1.5.5.4

