From 483963c3b1bd5f84b9241159809c4dbe06fc9b78 Mon Sep 17 00:00:00 2001
From: Marius Mathiesen <marius.mathiesen@gmail.com>
Date: Thu, 30 Apr 2009 14:54:14 +0200
Subject: [PATCH] Backporting Racks handling of multipart on Ruby 1.9

---
 .../vendor/rack-1.0/rack/utils.rb                  |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/actionpack/lib/action_controller/vendor/rack-1.0/rack/utils.rb b/actionpack/lib/action_controller/vendor/rack-1.0/rack/utils.rb
index 0a61bce..8c8838b 100644
--- a/actionpack/lib/action_controller/vendor/rack-1.0/rack/utils.rb
+++ b/actionpack/lib/action_controller/vendor/rack-1.0/rack/utils.rb
@@ -1,3 +1,4 @@
+# coding: binary
 require 'set'
 require 'tempfile'
 
@@ -302,7 +303,7 @@ module Rack
           content_length = env['CONTENT_LENGTH'].to_i
           input = env['rack.input']
 
-          boundary_size = boundary.size + EOL.size
+          boundary_size = Utils.bytesize(boundary) + EOL.size
           bufsize = 16384
 
           content_length -= boundary_size
-- 
1.6.0.2

