diff --git a/activemodel/lib/active_model/errors.rb b/activemodel/lib/active_model/errors.rb
index 1431228..e2277fc 100644
--- a/activemodel/lib/active_model/errors.rb
+++ b/activemodel/lib/active_model/errors.rb
@@ -170,6 +170,17 @@ module ActiveModel
       self
     end
 
+    # Represents an error message. Contains the model instance (+base+), the +attribute+ and the +options+
+    # passed to validators.
+    class ErrorMessage < String
+      attr_reader :full_message
+      
+      def initialize(message, full_message)
+        super(message || full_message)
+        @full_message = full_message
+      end
+    end
+
     # Adds +message+ to the error messages on +attribute+, which will be returned on a call to
     # <tt>on(attribute)</tt> for the same attribute. More than one error can be added to the same
     # +attribute+ in which case an array will be returned on a call to <tt>on(attribute)</tt>.
@@ -179,13 +190,17 @@ module ActiveModel
     # If +message+ is a proc, it will be called, allowing for things like <tt>Time.now</tt> to be used within an error.
     def add(attribute, message = nil, options = {})
       message ||= :invalid
+      full_message = options[:full_message]
 
       if message.is_a?(Symbol)
-        message = generate_message(attribute, message, options.except(*CALLBACKS_OPTIONS))
+        options = options.except(*CALLBACKS_OPTIONS)
+        options[:message] = options.delete(:full_message) if full_message
+        message = generate_message(attribute, message, options)
       elsif message.is_a?(Proc)
         message = message.call
       end
 
+      message = ErrorMessage.new(message, full_message) if message && full_message
       self[attribute] << message
     end
 
@@ -245,7 +260,8 @@ module ActiveModel
           options = { :default => "%{attribute} %{message}", :attribute => attr_name }
 
           messages.each do |m|
-            full_messages << I18n.t(:"errors.format", options.merge(:message => m))
+            full_messages << ((m.respond_to?(:full_message) && m.full_message) ?
+              m : I18n.t(:"errors.format", options.merge(:message => m)))
           end
         end
       end
