diff --git a/activesupport/lib/active_support/cache/strategy/local_cache.rb b/activesupport/lib/active_support/cache/strategy/local_cache.rb
index d83e259..ea59375 100644
--- a/activesupport/lib/active_support/cache/strategy/local_cache.rb
+++ b/activesupport/lib/active_support/cache/strategy/local_cache.rb
@@ -38,7 +38,7 @@ module ActiveSupport
           elsif value.nil?
             value = super
             local_cache.write(key, value || NULL) if local_cache
-            value
+            value.duplicable? ? value.dup : value
           else
             # forcing the value to be immutable
             value.duplicable? ? value.dup : value
diff --git a/activesupport/test/caching_test.rb b/activesupport/test/caching_test.rb
index 4e212f1..194e179 100644
--- a/activesupport/test/caching_test.rb
+++ b/activesupport/test/caching_test.rb
@@ -173,6 +173,15 @@ uses_memcached 'memcached backed store' do
       end
     end
 
+    def test_stored_objects_should_not_be_frozen
+      @cache.with_local_cache do
+        @cache.write('foo', 'bar')
+      end
+      @cache.with_local_cache do
+        assert !@cache.read('foo').frozen?
+      end
+    end
+
     def test_write_should_return_true_on_success
       @cache.with_local_cache do
         result = @cache.write('foo', 'bar')
