diff --git a/activerecord/lib/active_record/validations.rb b/activerecord/lib/active_record/validations.rb
index 9220eae..7f34f6b 100644
--- a/activerecord/lib/active_record/validations.rb
+++ b/activerecord/lib/active_record/validations.rb
@@ -706,6 +706,7 @@ module ActiveRecord
         configuration.update(attr_names.extract_options!)
 
         validates_each(attr_names,configuration) do |record, attr_name, value|
+          next unless record.send("#{attr_name}_changed?".to_sym)
           # The check for an existing value should be run from a class that
           # isn't abstract. This means working down from the current class
           # (self), to the first non-abstract class. Since classes don't know
