From 52112b4a810a14ea63f84ec05e9632f6f62008c8 Mon Sep 17 00:00:00 2001
From: Sebastian Martinez <sebastian@wyeworks.com>
Date: Thu, 18 Mar 2010 13:25:12 -0300
Subject: [PATCH] Fixed require line

---
 activesupport/test/ts_isolated.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activesupport/test/ts_isolated.rb b/activesupport/test/ts_isolated.rb
index cbab61a..fc1bbd6 100644
--- a/activesupport/test/ts_isolated.rb
+++ b/activesupport/test/ts_isolated.rb
@@ -1,6 +1,6 @@
 require 'test/unit'
 require 'rbconfig'
-require 'active_support/core_ext/kernel/reporting'
+require 'lib/active_support/core_ext/kernel/reporting'
 
 class TestIsolated < Test::Unit::TestCase
   ruby = File.join(*RbConfig::CONFIG.values_at('bindir', 'RUBY_INSTALL_NAME'))
-- 
1.6.3.3

