--- a/activerecord/lib/active_record/autosave_association.rb
+++ b/activerecord/lib/active_record/autosave_association.rb
@@ -251,6 +251,7 @@ module ActiveRecord
           unless association.marked_for_destruction?
             association.errors.each_error do |attribute, error|
               error = error.dup
+              error.base = self
               error.attribute = "#{reflection.name}_#{attribute}"
               errors.add(error) unless errors.on(error.attribute)
             end
