diff --git a/actionpack/test/template/tag_helper_test.rb b/actionpack/test/template/tag_helper_test.rb
index 2941dfe..f3eb979 100644
--- a/actionpack/test/template/tag_helper_test.rb
+++ b/actionpack/test/template/tag_helper_test.rb
@@ -49,6 +49,11 @@ class TagHelperTest < ActionView::TestCase
     assert_equal content_tag("a", "Create", :href => "create"),
                  content_tag("a", "href" => "create") { "Create" }    
   end
+
+  def test_content_tag_nested_in_content_tag
+    assert_equal content_tag("p", content_tag("b", "Hello")),
+                 content_tag("p") { content_tag("b", "Hello") }
+  end
   
   def test_cdata_section
     assert_equal "<![CDATA[<hello world>]]>", cdata_section("<hello world>")
