From f9d0c880b682d532cd3a241f8d62d9c8f0422791 Mon Sep 17 00:00:00 2001
From: Damian Janowski <damian.janowski@gmail.com>
Date: Thu, 17 Jul 2008 14:17:07 -0300
Subject: [PATCH] Default loading mechanism should be to not reload.

---
 railties/lib/initializer.rb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index b8b071d..2145cc1 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -842,11 +842,11 @@ Run `rake gems:install` to install the missing gems.
       end
 
       def default_dependency_mechanism
-        :load
+        :require
       end
 
       def default_cache_classes
-        false
+        true
       end
 
       def default_whiny_nils
-- 
1.5.5.1

