diff --git a/activesupport/lib/active_support/inflector.rb b/activesupport/lib/active_support/inflector.rb
index a4fd619..fcbfbaa 100644
--- a/activesupport/lib/active_support/inflector.rb
+++ b/activesupport/lib/active_support/inflector.rb
@@ -276,10 +276,12 @@ module Inflector
 # NameError is raised when the name is not in CamelCase or the constant is
 # unknown.
 def constantize(camel_cased_word)
-    unless /\A(?:::)?([A-Z]\w*(?:::[A-Z]\w*)*)\z/ =~ camel_cased_word
-      raise NameError, "#{camel_cased_word.inspect} is not a valid constant name!"
+    list = camel_cased_word.split("::")
+    list.shift if list.first.blank?
+    obj = Object
+    list.each do |x|
+      obj = obj.const_defined?(x) ? obj.const_get(x) : obj.const_missing(x)
   end
-
-    Object.module_eval("::#{$1}", __FILE__, __LINE__)
+    obj
 end
 # Turns a number into an ordinal string used to denote the position in an 
