Index: rails/actionpack/lib/action_view/helpers/prototype_helper.rb
===================================================================
--- rails/actionpack/lib/action_view/helpers/prototype_helper.rb	(revision 3947)
+++ rails/actionpack/lib/action_view/helpers/prototype_helper.rb	(working copy)
@@ -106,7 +106,7 @@
     # on the page in an Ajax response. 
     module PrototypeHelper
       unless const_defined? :CALLBACKS
-        CALLBACKS    = Set.new([ :uninitialized, :loading, :loaded,
+        CALLBACKS    = Set.new([ :create, :uninitialized, :loading, :loaded,
                          :interactive, :complete, :failure, :success ] +
                          (100..599).to_a)
         AJAX_OPTIONS = Set.new([ :before, :after, :condition, :url,

