From f14f1fad7e68501004845e8479b9f9c16ca5e83c Mon Sep 17 00:00:00 2001
From: Marcelo Giorgi <marklazz.uy@gmail.com>
Date: Fri, 15 Oct 2010 16:00:55 -0200
Subject: [PATCH] Test to exercise Callback#reset_callbacks for descendants [#5372 state: resolved]

---
 activesupport/test/callbacks_test.rb |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/activesupport/test/callbacks_test.rb b/activesupport/test/callbacks_test.rb
index 51b28b6..a20b20d 100644
--- a/activesupport/test/callbacks_test.rb
+++ b/activesupport/test/callbacks_test.rb
@@ -201,6 +201,9 @@ module CallbacksTest
   end
 
   class CleanPerson < ConditionalPerson
+
+    class ::CleanPersonChild < self; end
+
     reset_callbacks :save
   end
 
@@ -383,6 +386,14 @@ module CallbacksTest
     end
   end
 
+  class ResetDescendantsCallbackTest < Test::Unit::TestCase
+    def test_reset_save_child_person
+      person = CleanPersonChild.new
+      person.save
+      assert_equal [], person.history
+    end
+  end
+
   class CallbackTerminator
     include ActiveSupport::Callbacks
 
-- 
1.7.0.4

