diff --git a/railties/lib/rails_generator/generators/applications/app/template_runner.rb b/railties/lib/rails_generator/generators/applications/app/template_runner.rb
old mode 100644
new mode 100755
index 0e24d11..bb383b6
--- a/railties/lib/rails_generator/generators/applications/app/template_runner.rb
+++ b/railties/lib/rails_generator/generators/applications/app/template_runner.rb
@@ -4,6 +4,7 @@ require File.dirname(__FILE__) + '/scm/svn'
 
 require 'open-uri'
 require 'fileutils'
+require 'rake'
 
 module Rails
   class TemplateRunner
@@ -247,8 +248,8 @@ module Rails
 
     # Executes a ruby script (taking into account WIN32 platform quirks)
     def run_ruby_script(command, log_action = true)
-      ruby_command = RUBY_PLATFORM=~ /win32/ ? 'ruby ' : ''
-      run("#{ruby_command}#{command}", log_action)
+      ruby_command = FileUtils::RUBY
+      run("#{ruby_command} #{command}", log_action)
     end
 
     # Runs the supplied rake task
