commit c7d2a50309b4a74ffc3564c4b50b9ee86ff2e21e
Author: Jacob Lewallen <jlewallen@gmail.com>
Date:   Wed Jul 7 10:55:40 2010 +0900

    Set destroyed=true in opt locking's destroy

diff --git a/activerecord/lib/active_record/locking/optimistic.rb b/activerecord/lib/active_record/locking/optimistic.rb
index ceb0902..b6f87a5 100644
--- a/activerecord/lib/active_record/locking/optimistic.rb
+++ b/activerecord/lib/active_record/locking/optimistic.rb
@@ -124,6 +124,7 @@ module ActiveRecord
             end
           end
 
+          @destroyed = true
           freeze
         end
 
diff --git a/activerecord/test/cases/locking_test.rb b/activerecord/test/cases/locking_test.rb
index 66874cd..e712696 100644
--- a/activerecord/test/cases/locking_test.rb
+++ b/activerecord/test/cases/locking_test.rb
@@ -53,7 +53,8 @@ class OptimisticLockingTest < ActiveRecord::TestCase
     assert_raises(ActiveRecord::StaleObjectError) { p2.destroy }
 
     assert p1.destroy
-    assert_equal true, p1.frozen?
+    assert p1.frozen?
+    assert p1.destroyed?
     assert_raises(ActiveRecord::RecordNotFound) { Person.find(1) }
   end
 
