From 1c83dc9d2ba6d68ebe017f95c8104fa194649e5d Mon Sep 17 00:00:00 2001
From: Hongli Lai (Phusion) <hongli@phusion.nl>
Date: Mon, 6 Jul 2009 14:14:08 +0200
Subject: [PATCH] Mocha >= 0.9.7 is required, otherwise mocking doesn't work.

---
 actionpack/test/abstract_unit.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/actionpack/test/abstract_unit.rb b/actionpack/test/abstract_unit.rb
index cdeee93..0b67e56 100644
--- a/actionpack/test/abstract_unit.rb
+++ b/actionpack/test/abstract_unit.rb
@@ -8,7 +8,7 @@ require 'yaml'
 require 'stringio'
 require 'test/unit'
 
-gem 'mocha', '>= 0.9.5'
+gem 'mocha', '>= 0.9.7'
 require 'mocha'
 
 begin
-- 
1.6.0.5

