diff --git a/activerecord/test/cases/relation_scoping_test.rb b/activerecord/test/cases/relation_scoping_test.rb
index 1bdf313..a9151fc 100644
--- a/activerecord/test/cases/relation_scoping_test.rb
+++ b/activerecord/test/cases/relation_scoping_test.rb
@@ -1,4 +1,5 @@
 require "cases/helper"
+require 'models/car'
 require 'models/post'
 require 'models/author'
 require 'models/developer'
@@ -481,6 +482,16 @@ class DefaultScopingTest < ActiveRecord::TestCase
     assert_equal 'Aaron', aaron.name
   end
 
+  def test_where_attribute_cast_range
+    car = Car.where(:wheels_count => 0..2).new
+    assert_nil car.wheels_count
+  end
+
+  def test_where_attribute_cast_array
+    car = Car.where(:wheels_count => [0, 1, 2]).new
+    assert_nil car.wheels_count
+  end
+
   def test_scope_composed_by_limit_and_then_offset_is_equal_to_scope_composed_by_offset_and_then_limit
     posts_limit_offset = Post.limit(3).offset(2)
     posts_offset_limit = Post.offset(2).limit(3)

