diff --git a/railties/lib/rails/rack/log_tailer.rb b/railties/lib/rails/rack/log_tailer.rb
index a237cee..9c8f681 100644
--- a/railties/lib/rails/rack/log_tailer.rb
+++ b/railties/lib/rails/rack/log_tailer.rb
@@ -1,7 +1,11 @@
 module Rails
   module Rack
     class LogTailer
-      EnvironmentLog = "#{File.expand_path(Rails.root)}/log/#{Rails.env}.log"
+      EnvironmentLog = if Rails.configuration
+                         Rails.configuration.log_path
+                       else
+                         Rails::Configuration.new.default_log_path
+                       end  
 
       def initialize(app, log = nil)
         @app = app
