From e39421e007a77cbdf51cd62ab8f1c86f705eb0ff Mon Sep 17 00:00:00 2001
From: Kyle Hargraves <philodespotos@gmail.com>
Date: Fri, 16 May 2008 06:51:39 -0500
Subject: [PATCH] dbconsole should not require a username for postgresql

---
 railties/lib/commands/dbconsole.rb |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/railties/lib/commands/dbconsole.rb b/railties/lib/commands/dbconsole.rb
index 3e5010c..b81997a 100644
--- a/railties/lib/commands/dbconsole.rb
+++ b/railties/lib/commands/dbconsole.rb
@@ -40,10 +40,11 @@ when "mysql"
   exec(find_cmd('mysql5', 'mysql'), *args)
 
 when "postgresql"
+  ENV['PGUSER']     = config["username"] if config["username"]
   ENV['PGHOST']     = config["host"] if config["host"]
   ENV['PGPORT']     = config["port"].to_s if config["port"]
   ENV['PGPASSWORD'] = config["password"].to_s if config["password"]
-  exec(find_cmd('psql'), '-U', config["username"], config["database"])
+  exec(find_cmd('psql'), config["database"])
 
 when "sqlite"
   exec(find_cmd('sqlite'), config["database"])
-- 
1.5.4.5

