From 139aad415c25fae5a6a81fd0bdb886bb3d19acec Mon Sep 17 00:00:00 2001
From: Chris Cherry <ctcherry@gmail.com>
Date: Wed, 28 Jan 2009 00:52:17 -0800
Subject: [PATCH] script/server command can accept --path option to set app's root path

---
 railties/lib/commands/server.rb |   19 +++++++++++++++----
 1 files changed, 15 insertions(+), 4 deletions(-)

diff --git a/railties/lib/commands/server.rb b/railties/lib/commands/server.rb
index 43b1800..ebe34a4 100644
--- a/railties/lib/commands/server.rb
+++ b/railties/lib/commands/server.rb
@@ -17,7 +17,8 @@ options = {
   :environment => (ENV['RAILS_ENV'] || "development").dup,
   :config      => RAILS_ROOT + "/config.ru",
   :detach      => false,
-  :debugger    => false
+  :debugger    => false,
+  :path        => nil
 }
 
 ARGV.clone.options do |opts|
@@ -32,6 +33,7 @@ ARGV.clone.options do |opts|
   opts.on("-e", "--environment=name", String,
           "Specifies the environment to run this server under (test/development/production).",
           "Default: development") { |v| options[:environment] = v }
+  opts.on("-P", "--path=/path", String, "Runs Rails app mounted at a specific path.", "Default: /") { |v| options[:path] = v }
 
   opts.separator ""
 
@@ -50,7 +52,7 @@ unless server
 end
 
 puts "=> Booting #{ActiveSupport::Inflector.demodulize(server)}"
-puts "=> Rails #{Rails.version} application starting on http://#{options[:Host]}:#{options[:Port]}"
+puts "=> Rails #{Rails.version} application starting on http://#{options[:Host]}:#{options[:Port]}#{options[:path]}"
 
 %w(cache pids sessions sockets).each do |dir_to_make|
   FileUtils.mkdir_p(File.join(RAILS_ROOT, 'tmp', dir_to_make))
@@ -83,11 +85,20 @@ else
   inner_app = ActionController::Dispatcher.new
 end
 
+if options[:path].nil?
+  map_path = "/"
+else
+  ActionController::Base.relative_url_root = options[:path]
+  map_path = options[:path]
+end
+
 app = Rack::Builder.new {
   use Rails::Rack::LogTailer unless options[:detach]
-  use Rails::Rack::Static
   use Rails::Rack::Debugger if options[:debugger]
-  run inner_app
+  map map_path do
+    use Rails::Rack::Static 
+    run inner_app
+  end
 }.to_app
 
 puts "=> Call with -d to detach"
-- 
1.5.6.4

