diff --git a/activerecord/lib/active_record/log_subscriber.rb b/activerecord/lib/active_record/log_subscriber.rb
index 71065f9..f5c1189 100644
--- a/activerecord/lib/active_record/log_subscriber.rb
+++ b/activerecord/lib/active_record/log_subscriber.rb
@@ -6,6 +6,8 @@ module ActiveRecord
     end
 
     def sql(event)
+      return unless logger.debug?
+
       name = '%s (%.1fms)' % [event.payload[:name], event.duration]
       sql  = event.payload[:sql].squeeze(' ')
 
