diff --git a/vendor/rails/activeresource/lib/active_resource/connection.rb b/vendor/rails/activeresource/lib/active_resource/connection.rb
index 2bf83b1..fa9a171 100644
--- a/vendor/rails/activeresource/lib/active_resource/connection.rb
+++ b/vendor/rails/activeresource/lib/active_resource/connection.rb
@@ -108,7 +108,7 @@ module ActiveResource
         logger.info "#{method.to_s.upcase} #{site.scheme}://#{site.host}:#{site.port}#{path}" if logger
         result = nil
         time = Benchmark.realtime { result = http.send(method, path, *arguments) }
-        logger.info "--> #{result.code} #{result.message} (#{result.body ? result.body : 0}b %.2fs)" % time if logger
+        logger.info "--> %s %s (%sb %.2fs)" % [result.code, result.message, result.body ? result.body : 0, time] if logger
         handle_response(result)
       end
 
