From 1a6a9449c96aadb60a8838d1c960ab401accdaab Mon Sep 17 00:00:00 2001
From: Chris Moore <cdmwebs@gmail.com>
Date: Wed, 26 Nov 2008 03:30:59 -0500
Subject: [PATCH] create the db directory if it doesn't exist (rake db:create)

---
 railties/lib/tasks/databases.rake |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index 5cb27f1..a894ae9 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -29,6 +29,10 @@ namespace :db do
   def create_database(config)
     begin
       if config['adapter'] =~ /sqlite/
+        if !File.exists?("#{RAILS_ROOT}/db")
+          FileUtils.mkdir "#{RAILS_ROOT}/db"
+        end 
+      
         if File.exist?(config['database'])
           $stderr.puts "#{config['database']} already exists"
         else
-- 
1.6.0.2

