From 4703d1da355106a68025586c0113fa762eeadb00 Mon Sep 17 00:00:00 2001
From: sdsykes <sdsykes@gmail.com>
Date: Mon, 7 Sep 2009 13:22:19 +0300
Subject: [PATCH] Corrected instance_methods test for all hashes to work with ruby 1.9

---
 .../connection_adapters/mysql_adapter.rb           |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
index 4edb64c..1bb1c0b 100644
--- a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
@@ -8,7 +8,8 @@ module MysqlCompat #:nodoc:
     raise 'Mysql not loaded' unless defined?(::Mysql)
 
     target = defined?(Mysql::Result) ? Mysql::Result : MysqlRes
-    return if target.instance_methods.include?('all_hashes')
+    return if target.instance_methods.include?('all_hashes') ||
+              target.instance_methods.include?(:all_hashes)
 
     # Ruby driver has a version string and returns null values in each_hash
     # C driver >= 2.7 returns null values in each_hash
-- 
1.5.4.4

