From 9ca8291f618e8d75ca6de171564f5a0907140ad2 Mon Sep 17 00:00:00 2001
From: Rizwan Reza <rizwanreza@gmail.com>
Date: Mon, 17 May 2010 02:10:25 +0430
Subject: [PATCH] Uses random_element.

---
 .../application/initializers/frameworks_test.rb    |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/railties/test/application/initializers/frameworks_test.rb b/railties/test/application/initializers/frameworks_test.rb
index 8e57022..fadcc4c 100644
--- a/railties/test/application/initializers/frameworks_test.rb
+++ b/railties/test/application/initializers/frameworks_test.rb
@@ -47,7 +47,7 @@ module ApplicationTests
     test "if there's no config.active_support.bare, all of ActiveSupport is required" do
       use_frameworks []
       require "#{app_path}/config/environment"
-      assert_nothing_raised { [1,2,3].rand }
+      assert_nothing_raised { [1,2,3].random_element }
     end
 
     test "config.active_support.bare does not require all of ActiveSupport" do
@@ -57,7 +57,7 @@ module ApplicationTests
 
       Dir.chdir("#{app_path}/app") do
         require "#{app_path}/config/environment"
-        assert_raises(NoMethodError) { [1,2,3].rand }
+        assert_raises(NoMethodError) { [1,2,3].random_element }
       end
     end
 
-- 
1.6.5.2

