From c08fe47ca1319d00c06018bb31f19e814114ccfc Mon Sep 17 00:00:00 2001
From: George Ogata <george.ogata@gmail.com>
Date: Tue, 16 Sep 2008 14:46:50 +1000
Subject: [PATCH] Add descriptions to doc:plugins:<plugin> tasks.

---
 railties/lib/tasks/documentation.rake |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/railties/lib/tasks/documentation.rake b/railties/lib/tasks/documentation.rake
index 4ef8386..b20b58b 100644
--- a/railties/lib/tasks/documentation.rake
+++ b/railties/lib/tasks/documentation.rake
@@ -55,6 +55,7 @@ namespace :doc do
   namespace :plugins do
     # Define doc tasks for each plugin
     plugins.each do |plugin|
+      desc "Generate documentation for the #{plugin} plugin"
       task(plugin => :environment) do
         plugin_base   = "vendor/plugins/#{plugin}"
         options       = []
-- 
1.5.6.5

