From 044efe53f11eaf5301b34edce79f651776b95e7d Mon Sep 17 00:00:00 2001
From: Miles Georgi <azimux@gmail.com>
Date: Sun, 17 Aug 2008 23:45:25 -0700
Subject: [PATCH] fixes a pretty nasty transaction bug that can occur if you call change_column with invalid parameters

---
 .../connection_adapters/postgresql_adapter.rb      |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
index 3d52ecb..97e8c20 100644
--- a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
@@ -761,7 +761,8 @@ module ActiveRecord
 
         begin
           execute "ALTER TABLE #{quoted_table_name} ALTER COLUMN #{quote_column_name(column_name)} TYPE #{type_to_sql(type, options[:limit], options[:precision], options[:scale])}"
-        rescue ActiveRecord::StatementInvalid
+        rescue ActiveRecord::StatementInvalid => e
+          raise e if postgresql_version > 80000
           # This is PostgreSQL 7.x, so we have to use a more arcane way of doing it.
           begin
             begin_db_transaction
-- 
1.5.6.1.1071.g76fb

