From ed2bf8b00016b23a355dba00374584807210bbf7 Mon Sep 17 00:00:00 2001
From: Tim Connor <tim@wasabi.local>
Date: Fri, 17 Sep 2010 15:04:27 +1200
Subject: [PATCH] Raise errors when index creation fails

---
 .../abstract/schema_statements.rb                  |    6 ++----
 1 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
index 0245e63..63d7647 100644
--- a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
+++ b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
@@ -340,12 +340,10 @@ module ActiveRecord
         end
 
         if index_name.length > index_name_length
-          @logger.warn("Index name '#{index_name}' on table '#{table_name}' is too long; the limit is #{index_name_length} characters. Skipping.")
-          return
+          raise ArgumentError, "Index name '#{index_name}' on table '#{table_name}' is too long; the limit is #{index_name_length} characters"
         end
         if index_name_exists?(table_name, index_name, false)
-          @logger.warn("Index name '#{index_name}' on table '#{table_name}' already exists. Skipping.")
-          return
+          raise ArgumentError, "Index name '#{index_name}' on table '#{table_name}' already exists"
         end
         quoted_column_names = quoted_columns_for_index(column_names, options).join(", ")
 
-- 
1.6.5.7

