From c451c45df5b2314316cfaa42d1600009bf24f57a Mon Sep 17 00:00:00 2001
From: Ken Collins <ken@metaskills.net>
Date: Sun, 8 Aug 2010 21:51:02 -0400
Subject: [PATCH] A few schema changes for the SQL Server adapter.

---
 activerecord/test/schema/schema.rb |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/activerecord/test/schema/schema.rb b/activerecord/test/schema/schema.rb
index fc3810f..e58cbe8 100644
--- a/activerecord/test/schema/schema.rb
+++ b/activerecord/test/schema/schema.rb
@@ -224,7 +224,7 @@ ActiveRecord::Schema.define do
   end
 
   create_table :items, :force => true do |t|
-    t.column :name, :integer
+    t.column :name, :string
   end
 
   create_table :inept_wizards, :force => true do |t|
@@ -334,8 +334,8 @@ ActiveRecord::Schema.define do
     t.decimal :my_house_population, :precision => 2, :scale => 0
     t.decimal :decimal_number_with_default, :precision => 3, :scale => 2, :default => 2.78
     t.float   :temperature
-    # Oracle supports precision up to 38
-    if current_adapter?(:OracleAdapter)
+    # Oracle/SQLServer supports precision up to 38
+    if current_adapter?(:OracleAdapter,:SQLServerAdapter)
       t.decimal :atoms_in_universe, :precision => 38, :scale => 0
     else
       t.decimal :atoms_in_universe, :precision => 55, :scale => 0
-- 
1.7.1

