diff --git a/actionpack/lib/action_view/helpers/prototype_helper.rb b/actionpack/lib/action_view/helpers/prototype_helper.rb
index 908728c..9a3b84b 100644
--- a/actionpack/lib/action_view/helpers/prototype_helper.rb
+++ b/actionpack/lib/action_view/helpers/prototype_helper.rb
@@ -1020,7 +1020,7 @@ module ActionView
           js_options['parameters'] = options[:with]
         end
         
-        if protect_against_forgery? && !options[:form]
+        if protect_against_forgery? && !options[:form] && js_options['method'] != "'get'"
           if js_options['parameters']
             js_options['parameters'] << " + '&"
           else
