From c692523029860216c55e143d2b5adb262aa5afdf Mon Sep 17 00:00:00 2001
From: Andrew White <andyw@pixeltrix.co.uk>
Date: Mon, 29 Mar 2010 04:09:39 +0100
Subject: [PATCH] Ignore generated railties RDocs

---
 .gitignore |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/.gitignore b/.gitignore
index 9d68f48..16bffc1 100644
--- a/.gitignore
+++ b/.gitignore
@@ -15,7 +15,6 @@ activesupport/test/fixtures/isolation_test
 railties/test/500.html
 railties/test/fixtures/tmp
 railties/test/initializer/root/log
-railties/doc/guides/html/images
-railties/doc/guides/html/stylesheets
+railties/doc
 railties/guides/output
 railties/tmp
-- 
1.6.4.4

