From a8cb373d7ccedb2ca28ea6388f0f331fd05e67ff Mon Sep 17 00:00:00 2001
From: miloops <miloops@gmail.com>
Date: Thu, 11 Sep 2008 12:06:36 -0300
Subject: [PATCH] Add --debugger option to script/console.

---
 railties/lib/commands/console.rb |   13 +++++++++++++
 1 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/railties/lib/commands/console.rb b/railties/lib/commands/console.rb
index 2b9d92f..63df834 100644
--- a/railties/lib/commands/console.rb
+++ b/railties/lib/commands/console.rb
@@ -1,11 +1,13 @@
 irb = RUBY_PLATFORM =~ /(:?mswin|mingw)/ ? 'irb.bat' : 'irb'
 
 require 'optparse'
+
 options = { :sandbox => false, :irb => irb }
 OptionParser.new do |opt|
   opt.banner = "Usage: console [environment] [options]"
   opt.on('-s', '--sandbox', 'Rollback database modifications on exit.') { |v| options[:sandbox] = v }
   opt.on("--irb=[#{irb}]", 'Invoke a different irb.') { |v| options[:irb] = v }
+  opt.on("--debugger", 'Enable ruby-debugging for the console.') { |v| options[:debugger] = v }
   opt.parse!(ARGV)
 end
 
@@ -15,6 +17,17 @@ libs << " -r console_app"
 libs << " -r console_sandbox" if options[:sandbox]
 libs << " -r console_with_helpers"
 
+if options[:debugger]
+  begin
+    require 'ruby-debug'
+    libs << " -r ruby-debug"
+    puts "=> Debugger enabled"
+  rescue Exception
+    puts "You need to install ruby-debug to run the console in debugging mode. With gems, use 'gem install ruby-debug'"
+    exit
+  end
+end
+
 ENV['RAILS_ENV'] = case ARGV.first
   when "p"; "production"
   when "d"; "development"
-- 
1.5.5.1

