From b87e3d69d66f9e3d7e1645fe2842cc3173a426c2 Mon Sep 17 00:00:00 2001
From: Quin Hoxie <qhoxie@gmail.com>
Date: Sat, 13 Feb 2010 21:38:32 -0800
Subject: [PATCH 1/2] Use identity equivalence for tag comparison

---
 src/rails.js |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/rails.js b/src/rails.js
index faccea6..ebf6626 100644
--- a/src/rails.js
+++ b/src/rails.js
@@ -2,7 +2,7 @@ document.observe("dom:loaded", function() {
   function handleRemote(element) {
     var method, url, params;
 
-    if (element.tagName.toLowerCase() == 'form') {
+    if (element.tagName.toLowerCase() === 'form') {
       method = element.readAttribute('method') || 'post';
       url    = element.readAttribute('action');
       params = element.serialize(true);
@@ -72,7 +72,7 @@ document.observe("dom:loaded", function() {
   $(document.body).observe("ajax:complete", function(event) {
     var element = event.findElement();
 
-    if (element.tagName.toLowerCase() == 'form') {
+    if (element.tagName.toLowerCase() === 'form') {
       var inputs = element.select("input[type=submit][disabled=true][data-disable-with]");
       inputs.each(function(input) {
         input.value = input.readAttribute('data-original-value');
-- 
1.6.3.3


From 7851749722346d7c892e561a54006533868ca155 Mon Sep 17 00:00:00 2001
From: Quin Hoxie <qhoxie@gmail.com>
Date: Sat, 13 Feb 2010 21:56:28 -0800
Subject: [PATCH 2/2] Observe ajax:after for disable-with

---
 src/rails.js |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/rails.js b/src/rails.js
index ebf6626..12fb777 100644
--- a/src/rails.js
+++ b/src/rails.js
@@ -69,7 +69,7 @@ document.observe("dom:loaded", function() {
     }
   });
 
-  $(document.body).observe("ajax:complete", function(event) {
+  $(document.body).observe("ajax:after", function(event) {
     var element = event.findElement();
 
     if (element.tagName.toLowerCase() === 'form') {
-- 
1.6.3.3

