diff --git a/activerecord/test/cases/validations_test.rb b/activerecord/test/cases/validations_test.rb
index 97ac22e..54aa893 100755
--- a/activerecord/test/cases/validations_test.rb
+++ b/activerecord/test/cases/validations_test.rb
@@ -5,6 +5,7 @@ require 'models/reply'
 require 'models/person'
 require 'models/developer'
 require 'models/warehouse_thing'
+require 'models/guid'
 
 # The following methods in Topic are used in test_conditional_validation_*
 class Topic
@@ -459,6 +460,13 @@ class ValidationsTest < ActiveRecord::TestCase
     assert !t3.errors.on(:parent_id)
     assert_not_equal "has already been taken", t3.errors.on(:title)
   end
+  
+  def test_validate_uniqueness_with_columns_which_are_sql_keywords 
+ 	  Guid.validates_uniqueness_of :key 
+ 	  g = Guid.new 
+ 	  g.key = "foo" 
+ 	  assert_nothing_raised { !g.valid? } 
+ 	end
 
   def test_validate_uniqueness_with_non_standard_table_names
     i1 = WarehouseThing.create(:value => 1000)
diff --git a/activerecord/test/schema/schema.rb b/activerecord/test/schema/schema.rb
index 856f2fd..c772320 100644
--- a/activerecord/test/schema/schema.rb
+++ b/activerecord/test/schema/schema.rb
@@ -203,8 +203,6 @@ ActiveRecord::Schema.define do
       t.string   :type
     end
 
-
-
     ### These tables are created last as the order is significant
 
     # fk_test_has_fk should be before fk_test_has_pk
@@ -424,4 +422,8 @@ ActiveRecord::Schema.define do
     t.integer :sponsorable_id
     t.string :sponsorable_type
   end
+  
+  create_table :guids, :force => true do |t| 
+    t.string :key 
+  end
 end
