From 5801bac6c46ad00f91a90a49e04c717bf28dc4cc Mon Sep 17 00:00:00 2001
From: Aaron Patterson <aaron.patterson@gmail.com>
Date: Fri, 25 Jun 2010 16:32:09 -0700
Subject: [PATCH] refactoring the postgres adapter index method to avoid inject and use values_at. [#4976 state:resolved]

---
 .../connection_adapters/postgresql_adapter.rb      |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
index 26d8815..2fe2ae7 100644
--- a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
@@ -615,15 +615,15 @@ module ActiveRecord
           indkey = row[2].split(" ")
           oid = row[3]
 
-          columns = query(<<-SQL, "Columns for index #{row[0]} on #{table_name}").inject({}) {|attlist, r| attlist[r[1]] = r[0]; attlist}
-          SELECT a.attname, a.attnum
+          columns = Hash[query(<<-SQL, "Columns for index #{row[0]} on #{table_name}")]
+          SELECT a.attnum, a.attname
           FROM pg_attribute a
           WHERE a.attrelid = #{oid}
           AND a.attnum IN (#{indkey.join(",")})
           SQL
 
-          column_names = indkey.map {|attnum| columns[attnum] }
-          column_names.compact.empty? ? nil : IndexDefinition.new(table_name, index_name, unique, column_names)
+          column_names = columns.values_at(*indkey).compact
+          column_names.empty? ? nil : IndexDefinition.new(table_name, index_name, unique, column_names)
         end.compact
       end
 
-- 
1.7.0.6

