diff --git a/railties/builtin/rails_info/rails/info.rb b/railties/builtin/rails_info/rails/info.rb
index 8c858d2..0d66bb1 100644
--- a/railties/builtin/rails_info/rails/info.rb
+++ b/railties/builtin/rails_info/rails/info.rb
@@ -55,7 +55,7 @@ module Rails
       alias inspect to_s
 
       def to_html
-        returning table = '<table>' do
+        '<table>'.tap do |table|
           properties.each do |(name, value)|
             table << %(<tr><td class="name">#{CGI.escapeHTML(name.to_s)}</td>)
             table << %(<td class="value">#{CGI.escapeHTML(value.to_s)}</td></tr>)
diff --git a/railties/lib/rails/plugin.rb b/railties/lib/rails/plugin.rb
index 1dc4d14..5291257 100644
--- a/railties/lib/rails/plugin.rb
+++ b/railties/lib/rails/plugin.rb
@@ -35,7 +35,7 @@ module Rails
     def load_paths
       report_nonexistant_or_empty_plugin! unless valid?
       
-      returning [] do |load_paths|
+      [].tap do |load_paths|
         load_paths << lib_path  if has_lib_directory?
         load_paths << app_paths if has_app_directory?
       end.flatten
diff --git a/railties/lib/rails_generator/generators/components/migration/migration_generator.rb b/railties/lib/rails_generator/generators/components/migration/migration_generator.rb
index acf41e0..9a40c5d 100644
--- a/railties/lib/rails_generator/generators/components/migration/migration_generator.rb
+++ b/railties/lib/rails_generator/generators/components/migration/migration_generator.rb
@@ -8,7 +8,7 @@ class MigrationGenerator < Rails::Generator::NamedBase
   
   private  
     def get_local_assigns
-      returning(assigns = {}) do
+      {}.tap do |assigns|
         if class_name.underscore =~ /^(add|remove)_.*_(?:to|from)_(.*)/
           assigns[:migration_action] = $1
           assigns[:table_name]       = $2.pluralize
