From cbe3e79f02076231a83eab18c988dd29d072faad Mon Sep 17 00:00:00 2001
From: Aliaksey Kandratsenka <alk@tut.by>
Date: Fri, 28 Nov 2008 17:09:46 +0200
Subject: [PATCH] establish mysql connection before dropping database

---
 railties/lib/tasks/databases.rake |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index 5cb27f1..8d1422c 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -393,6 +393,7 @@ end
 def drop_database(config)
   case config['adapter']
   when 'mysql'
+    ActiveRecord::Base.establish_connection(config)
     ActiveRecord::Base.connection.drop_database config['database']
   when /^sqlite/
     FileUtils.rm(File.join(RAILS_ROOT, config['database']))
-- 
1.5.6.5

