From dca4897ace0e9f437646c7d7651734386a9d20e2 Mon Sep 17 00:00:00 2001
From: STLRUBY group (Mike Gaffney, Craig Buchek, Jeff Barczewski, Kent Jordan) <stlruby@googlegroups.com>
Date: Mon, 12 Apr 2010 19:38:28 -0500
Subject: [PATCH 1/2] Remove RAILS_ROOT from ci/site_config.rb file.

---
 ci/site_config.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/ci/site_config.rb b/ci/site_config.rb
index 47b7e92..395ffea 100644
--- a/ci/site_config.rb
+++ b/ci/site_config.rb
@@ -69,4 +69,4 @@ BuildReaper.number_of_builds_to_keep = 100
 
 # any files that you'd like to override in cruise, keep in ~/.cruise, and copy over when this file is loaded like this
 site_css = CRUISE_DATA_ROOT + "/site.css"
-FileUtils.cp site_css, RAILS_ROOT + "/public/stylesheets/site.css" if File.exists? site_css
+FileUtils.cp site_css, Rails.root.to_s + "/public/stylesheets/site.css" if File.exists? site_css
-- 
1.7.0.4

