From c79d58becb0e8ff039566c58d3b10625e1e028b4 Mon Sep 17 00:00:00 2001
From: ender672 <tle@holymonkey.com>
Date: Sat, 17 Oct 2009 14:57:35 -0700
Subject: [PATCH] Fix rename of active_support.

---
 lib/arel.rb |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/lib/arel.rb b/lib/arel.rb
index fcca607..97a5a61 100644
--- a/lib/arel.rb
+++ b/lib/arel.rb
@@ -1,7 +1,11 @@
 $LOAD_PATH.unshift(File.dirname(__FILE__))
 
 require 'rubygems'
-require 'activesupport'
+begin
+  require 'activesupport'
+rescue LoadError
+  require 'active_support'
+end
 require 'activerecord'
 require 'active_record/connection_adapters/abstract/quoting'
 
-- 
1.6.3.3

