--- sybase_adapter.rb.orig	2009-04-08 16:34:28.000000000 +0800
+++ sybase_adapter.rb	2009-04-08 16:34:51.000000000 +0800
@@ -17,8 +17,6 @@
 #              from columns(); added missing scale parameter in normalize_type().
 
 require 'active_record/connection_adapters/abstract_adapter'
-
-begin
 require 'sybsql'
 
 module ActiveRecord
@@ -658,7 +656,3 @@
     end
   end
 end
-
-rescue LoadError => cannot_require_sybase
-  # Couldn't load sybase adapter
-end
