diff --git a/rails/actionpack/lib/action_controller/url_rewriter.rb b/rails/actionpack/lib/action_controller/url_rewriter.rb
index e4c2a29..fa197ce 100644
--- a/rails/actionpack/lib/action_controller/url_rewriter.rb
+++ b/rails/actionpack/lib/action_controller/url_rewriter.rb
@@ -92,6 +92,12 @@ module ActionController
   #     end
   #   end
   module UrlWriter
+	@@default_proto = 'http'
+	def self.default_proto() @@default_proto; end
+	def self.default_proto=(proto)
+		@@default_proto = proto
+	end
+
     def self.included(base) #:nodoc:
       ActionController::Routing::Routes.install_helpers(base)
       base.mattr_accessor :default_url_options
@@ -129,7 +135,7 @@ module ActionController
       url = ''
 
       unless options.delete(:only_path)
-        url << (options.delete(:protocol) || 'http')
+        url << (options.delete(:protocol) || @@default_proto)
         url << '://' unless url.match("://")
 
         raise "Missing host to link to! Please provide :host parameter or set default_url_options[:host]" unless options[:host]
