diff --git a/actionpack/test/controller/base_test.rb b/actionpack/test/controller/base_test.rb
index 5ec59ac..dde174d 100644
--- a/actionpack/test/controller/base_test.rb
+++ b/actionpack/test/controller/base_test.rb
@@ -86,6 +86,15 @@ end
 class RecordIdentifierController < ActionController::Base
 end
 
+class ConfigActionController < ActionController::Base
+  def index
+    render :nothing => true
+  end
+  
+  def config
+  end
+end
+
 class ControllerClassTests < ActiveSupport::TestCase
 
   def test_controller_path
@@ -186,6 +195,12 @@ class PerformActionTest < ActionController::TestCase
     assert_raise(ActionController::UnknownAction) { get :hidden_action }
     assert_raise(ActionController::UnknownAction) { get :another_hidden_action }
   end
+  
+  def test_config_action_error
+    use_controller ConfigActionController
+    get :index
+  end
+  
 end
 
 class UrlOptionsTest < ActionController::TestCase
