From c7f8682b26b9e63a7f31ff34856cd0efc188fffc Mon Sep 17 00:00:00 2001
From: Daniel Hofstetter <daniel.hofstetter@42dh.com>
Date: Sat, 26 Sep 2009 17:37:49 +0200
Subject: [PATCH] Add missing else-clause to drop_databases

---
 railties/lib/tasks/databases.rake |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index 0c83311..f777bc7 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -416,6 +416,8 @@ def drop_database(config)
     when 'postgresql'
       ActiveRecord::Base.establish_connection(config.merge('database' => 'postgres', 'schema_search_path' => 'public'))
       ActiveRecord::Base.connection.drop_database config['database']
+    else
+      raise "Task not supported by '#{config["adapter"]}'"
     end
   rescue Exception => e
     puts "Couldn't drop #{config['database']} : #{e.inspect}"
-- 
1.6.3.3

