From a2e5cd22ac0c5f0dd3748360bed48cf486ac7c51 Mon Sep 17 00:00:00 2001
From: Diego Carrion <dc.rec1@gmail.com>
Date: Tue, 13 Apr 2010 18:23:41 -0300
Subject: [PATCH] fixed typo in RUNNING_UNIT_TESTS

---
 activerecord/RUNNING_UNIT_TESTS |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/RUNNING_UNIT_TESTS b/activerecord/RUNNING_UNIT_TESTS
index 39fc867..8559d72 100644
--- a/activerecord/RUNNING_UNIT_TESTS
+++ b/activerecord/RUNNING_UNIT_TESTS
@@ -14,7 +14,7 @@ connection.rb otherwise (on Postgres, at least) tests for default values will fa
 
 The easiest way to run the unit tests is through Rake. The default task runs
 the entire test suite for all the adapters. You can also run the suite on just
-one adapter by using the tasks test_mysql, test_sqlite, test_postgresql or any
+one adapter by using the tasks test_mysql, test_sqlite3, test_postgresql or any
 of the other test_ tasks. For more information, checkout the full array of rake
 tasks with "rake -T"
 
-- 
1.6.5.2

