From 3d056e426ae82c9b7c18682508b900a75b3c97ac Mon Sep 17 00:00:00 2001
From: Neeraj Singh <neerajdotname@gmail.com>
Date: Tue, 22 Jun 2010 14:00:21 -0400
Subject: [PATCH] should be able to run ActiveModel validations_test

---
 activemodel/test/cases/helper.rb |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/activemodel/test/cases/helper.rb b/activemodel/test/cases/helper.rb
index a81584b..a32f114 100644
--- a/activemodel/test/cases/helper.rb
+++ b/activemodel/test/cases/helper.rb
@@ -6,6 +6,7 @@ $:.unshift(lib) unless $:.include?('lib') || $:.include?(lib)
 require 'config'
 require 'active_model'
 require 'active_support/core_ext/string/access'
+require 'active_support/core_ext/hash/except'
 
 # Show backtraces for deprecated behavior for quicker cleanup.
 ActiveSupport::Deprecation.debug = true
-- 
1.6.5.2

