diff --git a/railties/lib/rails/gem_dependency.rb b/railties/lib/rails/gem_dependency.rb
index 471e03f..8b33cc7 100644
--- a/railties/lib/rails/gem_dependency.rb
+++ b/railties/lib/rails/gem_dependency.rb
@@ -35,6 +35,10 @@ module Rails
         args << @requirement.to_s if @requirement
         gem *args
       else
+        @spec ||= Gem::Specification.new do |s|
+          s.name    = @name
+          s.version = @version
+        end
         $LOAD_PATH.unshift File.join(unpacked_paths.first, 'lib')
         ext = File.join(unpacked_paths.first, 'ext')
         $LOAD_PATH.unshift(ext) if File.exist?(ext)
