diff --git a/activeresource/lib/active_resource/base.rb b/activeresource/lib/active_resource/base.rb
index 463ee9f..7dfdb96 100644
--- a/activeresource/lib/active_resource/base.rb
+++ b/activeresource/lib/active_resource/base.rb
@@ -27,6 +27,10 @@ module ActiveResource
   #     self.element_name = "person"
   #   end
   #  
+  # Basic Authentication parameters may also be included.
+  #   class Person < ActiveResource::Base
+  #     self.site = "http://username:password@api.people.com:3000/"
+  #   end
   # 
   # == Lifecycle methods
   #
