From 52ac30cd4b4b7a19bec320937d66d47b2dd1c1e7 Mon Sep 17 00:00:00 2001
From: Santiago Pastorino <santiago@wyeworks.com>
Date: Mon, 1 Feb 2010 03:52:34 -0200
Subject: [PATCH] action_view/safe_buffer is no longer valid

---
 actionpack/lib/action_view.rb |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/actionpack/lib/action_view.rb b/actionpack/lib/action_view.rb
index 20a2a63..325338d 100644
--- a/actionpack/lib/action_view.rb
+++ b/actionpack/lib/action_view.rb
@@ -35,7 +35,6 @@ module ActionView
     autoload :Context
     autoload :Template
     autoload :Helpers
-    autoload :SafeBuffer
 
     autoload_under "render" do
       autoload :Partials
-- 
1.6.5

