From 3ed99f85ef87166050ba2122cf48fd475f35b3df Mon Sep 17 00:00:00 2001
From: Carlos Paramio <carlosparamio@gmail.com>
Date: Tue, 4 Nov 2008 20:32:52 +0100
Subject: [PATCH] Change usage of defined? to check the rubygems constant existance by a rescue block on boot.rb for Ruby 1.9 compatibility

---
 railties/environments/boot.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/environments/boot.rb b/railties/environments/boot.rb
index 6a30b54..e2aff13 100644
--- a/railties/environments/boot.rb
+++ b/railties/environments/boot.rb
@@ -67,7 +67,7 @@ module Rails
 
     class << self
       def rubygems_version
-        Gem::RubyGemsVersion if defined? Gem::RubyGemsVersion
+        Gem::RubyGemsVersion rescue nil
       end
 
       def gem_version
-- 
1.6.0.2

