From 5ee6426611a1f0d9c3e33a147464ab0fd04a3582 Mon Sep 17 00:00:00 2001
From: Nick Sieger <nick@nicksieger.com>
Date: Tue, 6 Jan 2009 15:00:08 -0600
Subject: [PATCH] Use current user for postgresql

---
 .../connections/jdbc_jdbcpostgresql/connection.rb  |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/activerecord/test/connections/jdbc_jdbcpostgresql/connection.rb b/activerecord/test/connections/jdbc_jdbcpostgresql/connection.rb
index 9da3f70..0685da4 100644
--- a/activerecord/test/connections/jdbc_jdbcpostgresql/connection.rb
+++ b/activerecord/test/connections/jdbc_jdbcpostgresql/connection.rb
@@ -11,12 +11,12 @@ ActiveRecord::Base.logger = Logger.new("debug.log")
 ActiveRecord::Base.configurations = {
   'arunit' => {
     :adapter  => 'jdbcpostgresql',
-    :username => 'rails',
+    :username => ENV['USER'] || 'rails',
     :database => 'activerecord_unittest'
   },
   'arunit2' => {
     :adapter  => 'jdbcpostgresql',
-    :username => 'rails',
+    :username => ENV['USER'] || 'rails',
     :database => 'activerecord_unittest2'
   }
 }
-- 
1.5.6

