From f37a92bb00ae4e5f63ec4e9dafe1dad98eb9ddec Mon Sep 17 00:00:00 2001
From: Quin Hoxie <qhoxie@gmail.com>
Date: Thu, 28 Jan 2010 01:52:41 -0800
Subject: [PATCH] Proper, consistent equality operators for Prototype driver.

---
 .../public/javascripts/prototype.rails.js          |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/generators/rails/app/templates/public/javascripts/prototype.rails.js b/railties/lib/generators/rails/app/templates/public/javascripts/prototype.rails.js
index e4bc82a..cd0f9f4 100644
--- a/railties/lib/generators/rails/app/templates/public/javascripts/prototype.rails.js
+++ b/railties/lib/generators/rails/app/templates/public/javascripts/prototype.rails.js
@@ -9,7 +9,7 @@ Event.observe(document, 'dom:loaded', function() {
 
     if (el.readAttribute('data-submit')) {
       var submit_el = $(el.readAttribute('data-submit'));
-      if(submit_el !== undefined && submit_el.tagName.toUpperCase() == 'FORM'){
+      if(submit_el !== undefined && submit_el.tagName.toUpperCase() === 'FORM'){
         data = submit_el.serialize();
       }
     } else if (el.readAttribute('data-with')) {
-- 
1.6.3.3

