From 594e3dbe39eb710940147aba79ad2566066b0f0d Mon Sep 17 00:00:00 2001
From: Taylor Weibley (themcgruff) <tweibley@engineyard.com>
Date: Fri, 6 Aug 2010 21:48:04 -0700
Subject: [PATCH] Use stat instead of ping since it is less expensive on the server side.

---
 .../connection_adapters/mysql_adapter.rb           |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
index ba0051d..4c52a40 100644
--- a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
@@ -233,13 +233,13 @@ module ActiveRecord
       # CONNECTION MANAGEMENT ====================================
 
       def active?
-        if @connection.respond_to?(:stat)
-          @connection.stat
+        if @connection.respond_to?(:ping)
+          @connection.ping
         else
           @connection.query 'select 1'
         end
 
-        # mysql-ruby doesn't raise an exception when stat fails.
+        # mysql-ruby doesn't raise an exception when ping fails.
         if @connection.respond_to?(:errno)
           @connection.errno.zero?
         else
-- 
1.7.2

