diff --git a/actionpack/lib/action_view/helpers/form_helper.rb b/actionpack/lib/action_view/helpers/form_helper.rb
index 0e77a7e..3591b7d 100644
--- a/actionpack/lib/action_view/helpers/form_helper.rb
+++ b/actionpack/lib/action_view/helpers/form_helper.rb
@@ -469,7 +469,7 @@ module ActionView
       def to_label_tag(text = nil, options = {})
         name_and_id = options.dup
         add_default_name_and_id(name_and_id)
-        options["for"] = name_and_id["id"]
+        options.stringify_keys!.reverse_merge!("for" => name_and_id["id"])
         content = (text.blank? ? nil : text.to_s) || method_name.humanize
         content_tag("label", content, options)
       end
diff --git a/actionpack/test/template/form_helper_test.rb b/actionpack/test/template/form_helper_test.rb
index c48d5df..9ac7269 100644
--- a/actionpack/test/template/form_helper_test.rb
+++ b/actionpack/test/template/form_helper_test.rb
@@ -77,6 +77,14 @@ class FormHelperTest < ActionView::TestCase
     assert_dom_equal('<label for="post_title">Title</label>', label(:post, :title))
   end
 
+  def test_label_with_for_attribute_as_symbol
+    assert_dom_equal('<label for="my_for">Title</label>', label(:post, :title, nil, :for => "my_for"))
+  end
+
+  def test_label_with_for_attribute_as_string
+    assert_dom_equal('<label for="my_for">Title</label>', label(:post, :title, nil, "for" => "my_for"))
+  end
+
   def test_text_field
     assert_dom_equal(
       '<input id="post_title" name="post[title]" size="30" type="text" value="Hello World" />', text_field("post", "title")
