From 89ffc9d6d0c9de418b70ba79ad6dce99ad09a961 Mon Sep 17 00:00:00 2001
From: Vincent Charles <vjcharles@gmail.com>
Date: Mon, 23 Mar 2009 19:05:31 -0700
Subject: [PATCH] Updated for rails 2.3.2 and now returns tokens of any length

---
 lib/token_generator.rb |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/token_generator.rb b/lib/token_generator.rb
index 8af994f..52bbe32 100644
--- a/lib/token_generator.rb
+++ b/lib/token_generator.rb
@@ -1,9 +1,9 @@
 module TokenGenerator
   def generate_token(size = 12, &validity)
-    constant = "#{self.class.name}#{id}"
-
+    bytes = ((size+1)/2).to_i
+    
     begin
-      token = CGI::Session.generate_unique_id(constant).first(size)
+      token = ActiveSupport::SecureRandom.hex(bytes).first(size)
     end while !validity.call(token) if block_given?
 
     token
-- 
1.6.2

