From 6fb08902cc07ff3ac435df9624dbd10f715e0ed9 Mon Sep 17 00:00:00 2001
From: Rodrigo Rosenfeld Rosas <rr_rosas@yahoo.com.br>
Date: Wed, 19 Aug 2009 16:39:30 -0300
Subject: [PATCH] Fix spelling in Rails 'generator' generator.

---
 .../rails/generator/generator_generator.rb         |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/generators/rails/generator/generator_generator.rb b/railties/lib/generators/rails/generator/generator_generator.rb
index 2fc97b2..5b5d188 100644
--- a/railties/lib/generators/rails/generator/generator_generator.rb
+++ b/railties/lib/generators/rails/generator/generator_generator.rb
@@ -6,7 +6,7 @@ module Rails
       class_option :namespace, :type => :boolean, :default => true,
                                :desc => "Namespace generator under lib/generators/name"
 
-      def craete_generator_files
+      def create_generator_files
         directory '.', generator_dir
       end
 
-- 
1.6.0.4

