From 2ea6a64775927cbdcc02a420b1c08cee01c2802c Mon Sep 17 00:00:00 2001
From: Andrew White <andyw@pixeltrix.co.uk>
Date: Mon, 29 Mar 2010 00:10:18 +0100
Subject: [PATCH] Add ActionDispatch to RDoc task

---
 Rakefile |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/Rakefile b/Rakefile
index 66a65fb..cae9b04 100644
--- a/Rakefile
+++ b/Rakefile
@@ -88,6 +88,7 @@ Rake::RDocTask.new do |rdoc|
   rdoc.rdoc_files.include('actionpack/README')
   rdoc.rdoc_files.include('actionpack/CHANGELOG')
   rdoc.rdoc_files.include('actionpack/lib/action_controller/**/*.rb')
+  rdoc.rdoc_files.include('actionpack/lib/action_dispatch/**/*.rb')
   rdoc.rdoc_files.include('actionpack/lib/action_view/**/*.rb')
   rdoc.rdoc_files.exclude('actionpack/lib/action_controller/vendor/*')
 
-- 
1.6.4.4

