From ab621b6753441905d266aa13c7295c34d45183aa Mon Sep 17 00:00:00 2001
From: Luke Amdor <luke@absoluterubbish.net>
Date: Mon, 27 Apr 2009 11:30:08 -0500
Subject: [PATCH] Removed reference in documentation of form_authenticity_token to deprecated :secret option of protect_from_forgery

---
 .../base/request_forgery_protection.rb             |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/actionpack/lib/action_controller/base/request_forgery_protection.rb b/actionpack/lib/action_controller/base/request_forgery_protection.rb
index 3067122..df91dc1 100644
--- a/actionpack/lib/action_controller/base/request_forgery_protection.rb
+++ b/actionpack/lib/action_controller/base/request_forgery_protection.rb
@@ -96,8 +96,7 @@ module ActionController #:nodoc:
         !request.content_type.nil? && request.content_type.verify_request?
       end
     
-      # Sets the token value for the current session.  Pass a <tt>:secret</tt> option
-      # in +protect_from_forgery+ to add a custom salt to the hash.
+      # Sets the token value for the current session.
       def form_authenticity_token
         session[:_csrf_token] ||= ActiveSupport::SecureRandom.base64(32)
       end
-- 
1.6.2.38.ga54a

