From 9c55b6fc7863a558185fe95f478bb3a9f3d369e7 Mon Sep 17 00:00:00 2001
From: Graeme Mathieson <mathie@woss.name>
Date: Wed, 20 May 2009 06:41:24 +0100
Subject: [PATCH 3/3] It's the frozen gems who's specs want refreshing.

---
 railties/lib/tasks/gems.rake |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/tasks/gems.rake b/railties/lib/tasks/gems.rake
index efadb1d..372ff67 100644
--- a/railties/lib/tasks/gems.rake
+++ b/railties/lib/tasks/gems.rake
@@ -51,7 +51,7 @@ namespace :gems do
 
   desc "Regenerate gem specifications in correct format."
   task :refresh_specs => :base do
-    current_gems.each &:refresh
+    frozen_gems.each &:refresh
   end
 end
 
-- 
1.6.2.1

