--- base.rb	2010-07-01 22:56:09.000000000 +0200
+++ /home/jreidinger/base.rb	2010-07-02 15:17:49.000000000 +0200
@@ -1030,10 +1030,10 @@
             when Array
               resource = find_or_create_resource_for_collection(key)
               value.map do |attrs|
-                if attrs.is_a?(String) || attrs.is_a?(Numeric)
-                  attrs.duplicable? ? attrs.dup : attrs
-                else
+                if attrs.is_a?(Hash)
                   resource.new(attrs)
+                else
+                  attrs.duplicable? ? attrs.dup : attrs
                 end
               end
             when Hash
