From 714dca739d699cb7c3306818a331d25d050341da Mon Sep 17 00:00:00 2001
From: Paulo Schneider <paulo.schneider@gmail.com>
Date: Mon, 6 Apr 2009 21:38:24 +0100
Subject: [PATCH] Fixing typo on the routes.rb comments

---
 railties/configs/routes.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/configs/routes.rb b/railties/configs/routes.rb
index 4f3d9d2..ea14ce1 100644
--- a/railties/configs/routes.rb
+++ b/railties/configs/routes.rb
@@ -37,7 +37,7 @@ ActionController::Routing::Routes.draw do |map|
 
   # Install the default routes as the lowest priority.
   # Note: These default routes make all actions in every controller accessible via GET requests. You should
-  # consider removing the them or commenting them out if you're using named routes and resources.
+  # consider removing or commenting them out if you're using named routes and resources.
   map.connect ':controller/:action/:id'
   map.connect ':controller/:action/:id.:format'
 end
-- 
1.5.5.1

