diff --git a/actionmailer/lib/action_mailer/railtie.rb b/actionmailer/lib/action_mailer/railtie.rb
index d67a5b6..537fa57 100644
--- a/actionmailer/lib/action_mailer/railtie.rb
+++ b/actionmailer/lib/action_mailer/railtie.rb
@@ -19,7 +19,7 @@ module ActionMailer
       options.stylesheets_dir ||= paths["public/stylesheets"].first
 
       # make sure readers methods get compiled
-      options.asset_path           ||= nil
+      options.asset_base_path      ||= nil
       options.asset_host           ||= nil
 
       ActiveSupport.on_load(:action_mailer) do
diff --git a/actionpack/lib/abstract_controller/asset_paths.rb b/actionpack/lib/abstract_controller/asset_paths.rb
index 9ca2fb7..928d139 100644
--- a/actionpack/lib/abstract_controller/asset_paths.rb
+++ b/actionpack/lib/abstract_controller/asset_paths.rb
@@ -3,7 +3,7 @@ module AbstractController
     extend ActiveSupport::Concern
 
     included do
-      config_accessor :asset_host, :asset_path, :assets_dir, :javascripts_dir, :stylesheets_dir
+      config_accessor :asset_host, :asset_base_path, :assets_dir, :javascripts_dir, :stylesheets_dir
     end
   end
 end
\ No newline at end of file
diff --git a/actionpack/lib/action_controller/railtie.rb b/actionpack/lib/action_controller/railtie.rb
index c5a661f..fb65054 100644
--- a/actionpack/lib/action_controller/railtie.rb
+++ b/actionpack/lib/action_controller/railtie.rb
@@ -27,7 +27,7 @@ module ActionController
       options.page_cache_directory ||= paths["public"].first
 
       # make sure readers methods get compiled
-      options.asset_path           ||= nil
+      options.asset_base_path      ||= nil
       options.asset_host           ||= nil
 
       ActiveSupport.on_load(:action_controller) do
diff --git a/actionpack/lib/action_view/helpers/asset_tag_helper.rb b/actionpack/lib/action_view/helpers/asset_tag_helper.rb
index c1dfbe5..af2771f 100644
--- a/actionpack/lib/action_view/helpers/asset_tag_helper.rb
+++ b/actionpack/lib/action_view/helpers/asset_tag_helper.rb
@@ -740,7 +740,7 @@ module ActionView
           if controller.respond_to?(:env) && controller.env["action_dispatch.asset_path"]
             source = rewrite_asset_path(source, controller.env["action_dispatch.asset_path"])
           end
-          source = rewrite_asset_path(source, config.asset_path)
+          source = rewrite_asset_path(source, config.asset_base_path)
 
           has_request = controller.respond_to?(:request)
           source = rewrite_relative_url_root(source, controller.config.relative_url_root) if has_request && include_host
diff --git a/actionpack/test/template/asset_tag_helper_test.rb b/actionpack/test/template/asset_tag_helper_test.rb
index 3abcdfb..54ca9d5 100644
--- a/actionpack/test/template/asset_tag_helper_test.rb
+++ b/actionpack/test/template/asset_tag_helper_test.rb
@@ -381,14 +381,14 @@ class AssetTagHelperTest < ActionView::TestCase
   end
 
   def test_string_asset_id
-    @controller.config.asset_path = "/assets.v12345%s"
+    @controller.config.asset_base_path = "/assets.v12345%s"
 
     expected_path = "/assets.v12345/images/rails.png"
     assert_equal %(<img alt="Rails" src="#{expected_path}" />), image_tag("rails.png")
   end
 
   def test_env_asset_path
-    @controller.config.asset_path = "/assets%s"
+    @controller.config.asset_base_path = "/assets%s"
     def @controller.env; @_env ||= {} end
     @controller.env["action_dispatch.asset_path"] = "/omg%s"
 
@@ -397,7 +397,7 @@ class AssetTagHelperTest < ActionView::TestCase
   end
 
   def test_proc_asset_id
-    @controller.config.asset_path = Proc.new do |asset_path|
+    @controller.config.asset_base_path = Proc.new do |asset_path|
       "/assets.v12345#{asset_path}"
     end
 
@@ -406,7 +406,7 @@ class AssetTagHelperTest < ActionView::TestCase
   end
 
   def test_env_proc_asset_path
-    @controller.config.asset_path = Proc.new do |asset_path|
+    @controller.config.asset_base_path = Proc.new do |asset_path|
       "/assets.v12345#{asset_path}"
     end
 
diff --git a/railties/lib/rails/application/configuration.rb b/railties/lib/rails/application/configuration.rb
index 3505388..a827bcc 100644
--- a/railties/lib/rails/application/configuration.rb
+++ b/railties/lib/rails/application/configuration.rb
@@ -29,8 +29,8 @@ module Rails
       end
 
       def asset_path=(value)
-        action_mailer.asset_path = value if respond_to?(:action_mailer) && action_mailer
-        action_controller.asset_path = value if respond_to?(:action_controller) && action_controller
+        action_mailer.asset_base_path = value if respond_to?(:action_mailer) && action_mailer
+        action_controller.asset_base_path = value if respond_to?(:action_controller) && action_controller
         super(value)
       end
 
