From c7714b70a44b4f53f6bddd351e384f2cf8eccc7f Mon Sep 17 00:00:00 2001
From: Ken Collins <ken@metaskills.net>
Date: Sun, 8 Aug 2010 17:37:11 -0400
Subject: [PATCH] Use same precision for atoms_in_universe as Oracle for SQL Server schema.

---
 activerecord/test/schema/schema.rb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/activerecord/test/schema/schema.rb b/activerecord/test/schema/schema.rb
index fc3810f..05ca143 100644
--- a/activerecord/test/schema/schema.rb
+++ b/activerecord/test/schema/schema.rb
@@ -334,8 +334,8 @@ ActiveRecord::Schema.define do
     t.decimal :my_house_population, :precision => 2, :scale => 0
     t.decimal :decimal_number_with_default, :precision => 3, :scale => 2, :default => 2.78
     t.float   :temperature
-    # Oracle supports precision up to 38
-    if current_adapter?(:OracleAdapter)
+    # Oracle/SQLServer supports precision up to 38
+    if current_adapter?(:OracleAdapter,:SQLServerAdapter)
       t.decimal :atoms_in_universe, :precision => 38, :scale => 0
     else
       t.decimal :atoms_in_universe, :precision => 55, :scale => 0
-- 
1.7.1

