diff --git a/railties/lib/rails/gem_dependency.rb b/railties/lib/rails/gem_dependency.rb
index 471e03f..d1e1f1f 100644
--- a/railties/lib/rails/gem_dependency.rb
+++ b/railties/lib/rails/gem_dependency.rb
@@ -86,7 +86,7 @@ module Rails
     def unpack_to(directory)
       FileUtils.mkdir_p directory
       Dir.chdir directory do
-        Gem::GemRunner.new.run(unpack_command)
+        Gem::GemRunner.new.run(unpack_command.map {|i| i.gsub(/\A["']|["']\Z/, '')})
       end
 
       # copy the gem's specification into GEMDIR/.specification so that
