From d5ffa5c2131107e710142e48ccf8458dfaa52d1d Mon Sep 17 00:00:00 2001
From: Justin Baker <bakermoto@gmail.com>
Date: Mon, 31 May 2010 15:44:30 -0400
Subject: [PATCH] added to global

---
 src/rails.js |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/rails.js b/src/rails.js
index c6752d2..3d142b7 100644
--- a/src/rails.js
+++ b/src/rails.js
@@ -104,4 +104,8 @@ document.on("dom:loaded", function() {
       input.disabled = false;
     });
   });
+  window.Prototype.Rails = {
+    handleRemote: handleRemote,
+    handleMethod: handleMethod
+  };
 });
-- 
1.7.0.4

