From f96613c03d9e80ad175c371d4d52eeb452b3af3f Mon Sep 17 00:00:00 2001
From: Sam Granieri <sam@samgranieri.com>
Date: Tue, 22 Apr 2008 16:35:21 -0500
Subject: [PATCH] preventing generated documentation from being picked up by git

---
 .gitignore |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/.gitignore b/.gitignore
index 2bf1ef8..d19b982 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1 +1,7 @@
-debug.log
\ No newline at end of file
+debug.log
+activeresource/doc
+activerecord/doc
+actionpack/doc
+actionmailer/doc
+activesupport/doc
+railties/doc
-- 
1.5.5.1

