From c71b4637d77b57d56574b3c23dbdb28ccfa1aac9 Mon Sep 17 00:00:00 2001
From: Anil Wadghule <anildigital@gmail.com>
Date: Wed, 6 Oct 2010 23:15:48 +0530
Subject: [PATCH] Add jruby-openssl gem in Gemfile necessary for JRuby

---
 Gemfile |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/Gemfile b/Gemfile
index 7fe1c3a..be91531 100644
--- a/Gemfile
+++ b/Gemfile
@@ -55,6 +55,8 @@ platforms :jruby do
 
   gem "activerecord-jdbcsqlite3-adapter"
 
+  gem "jruby-openssl"
+
   group :db do
     gem "activerecord-jdbcmysql-adapter"
     gem "activerecord-jdbcpostgresql-adapter"
-- 
1.7.3.1

