Index: lib/active_record/autosave_association.rb
===================================================================
--- lib/active_record/autosave_association.rb	(revision 1545)
+++ lib/active_record/autosave_association.rb	(working copy)
@@ -157,8 +157,8 @@
       # Adds a validate and save callback for the association as specified by
       # the +reflection+.
       def add_autosave_association_callbacks(reflection)
-        save_method = "autosave_associated_records_for_#{reflection.name}"
-        validation_method = "validate_associated_records_for_#{reflection.name}"
+        save_method = :"autosave_associated_records_for_#{reflection.name}"
+        validation_method = :"validate_associated_records_for_#{reflection.name}"
         validate validation_method
 
         case reflection.macro
