diff --git a/activerecord/test/cases/relations_test.rb b/activerecord/test/cases/relations_test.rb
index 5018b16..64e43a4 100644
--- a/activerecord/test/cases/relations_test.rb
+++ b/activerecord/test/cases/relations_test.rb
@@ -473,10 +473,19 @@ class RelationTest < ActiveRecord::TestCase
     relation = Author.unscoped
     relation = relation.where(:name => david.name)
     relation = relation.where(:name => 'Santiago')
-    relation = relation.where(:id => david.id)
+    #relation = relation.where(:id => david.id)
     assert_equal [david], relation.all
   end
 
+  def test_dynamic_finder_after_where_for_same_column
+    david = authors(:david)
+    mary = authors(:mary)
+    relation = Author.unscoped
+    relation = relation.where(:id => [david.id, mary.id])
+    assert_equal mary, relation.find_by_id(mary.id)
+  end
+
+
   def test_find_all_with_multiple_ors
     david = authors(:david)
     relation = [
