diff --git a/activesupport/lib/active_support/message_verifier.rb b/activesupport/lib/active_support/message_verifier.rb
index aae5a34..349b775 100644
--- a/activesupport/lib/active_support/message_verifier.rb
+++ b/activesupport/lib/active_support/message_verifier.rb
@@ -43,7 +43,8 @@ module ActiveSupport
         if a.length == b.length
           result = 0
           for i in 0..(a.length - 1)
-            result |= a[i] ^ b[i]
+            # We don't want strings, which is what Ruby 1.9 will give us
+            result |= a[i].is_a?(String) ? (a[i].ord ^ b[i].ord) : (a[i] ^ b[i])
           end
           result == 0
         else
