From ca0452c289ec7fe659a8f8d29d51e4f59fe40513 Mon Sep 17 00:00:00 2001
From: Anil Wadghule <anildigital@gmail.com>
Date: Mon, 12 Apr 2010 23:45:24 +0530
Subject: [PATCH] Replace deprecated RAILS_ROOT to Rails.env [#4373 state:resolved]

---
 ci/site_config.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/ci/site_config.rb b/ci/site_config.rb
index 47b7e92..ac7f658 100644
--- a/ci/site_config.rb
+++ b/ci/site_config.rb
@@ -69,4 +69,4 @@ BuildReaper.number_of_builds_to_keep = 100
 
 # any files that you'd like to override in cruise, keep in ~/.cruise, and copy over when this file is loaded like this
 site_css = CRUISE_DATA_ROOT + "/site.css"
-FileUtils.cp site_css, RAILS_ROOT + "/public/stylesheets/site.css" if File.exists? site_css
+FileUtils.cp site_css, Rails.root + "/public/stylesheets/site.css" if File.exists? site_css
-- 
1.7.0.4

