diff --git a/activerecord/lib/active_record/base.rb b/activerecord/lib/active_record/base.rb
index 47dc2d4..7add623 100644
--- a/activerecord/lib/active_record/base.rb
+++ b/activerecord/lib/active_record/base.rb
@@ -1855,7 +1855,7 @@ MSG
       end
 
       def object_from_yaml(string)
-        return string unless string.is_a?(String) && string =~ /^---/
+        return string unless string.is_a?(String) && string =~ /\A---/
         YAML::load(string) rescue string
       end
 
diff --git a/activerecord/test/cases/base_test.rb b/activerecord/test/cases/base_test.rb
index a58d5de..9504ca5 100644
--- a/activerecord/test/cases/base_test.rb
+++ b/activerecord/test/cases/base_test.rb
@@ -977,6 +977,12 @@ class BasicsTest < ActiveRecord::TestCase
     assert_equal(myobj, topic.content)
   end
 
+  def test_serialized_string_attribute_with_markdown_content_that_resembles_yaml
+    myobj = "![Rails](http://rubyonrails.org/images/rails.png)\n\n-----\n\n"
+    topic = Topic.create("content" => myobj).reload
+    assert_equal(myobj, topic.content)
+  end
+
   def test_nil_serialized_attribute_with_class_constraint
     topic = Topic.new
     assert_nil topic.content
