From 9edfdfb3be56d99a710fbfc61f22da22007f73b9 Mon Sep 17 00:00:00 2001
From: Ryan Bigg <radarlistener@gmail.com>
Date: Thu, 28 Jan 2010 16:48:09 +1000
Subject: [PATCH] Remove require to 'rails/engine', as 'rails/plugin' already requires this.

---
 railties/lib/rails/application.rb |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/railties/lib/rails/application.rb b/railties/lib/rails/application.rb
index 9e41210..6d8f7df 100644
--- a/railties/lib/rails/application.rb
+++ b/railties/lib/rails/application.rb
@@ -1,7 +1,6 @@
 require 'fileutils'
 require 'rails/railties_path'
 require 'rails/plugin'
-require 'rails/engine'
 
 module Rails
   class Application < Engine
-- 
1.6.4.1

