From 347f2c9f1c62c4622377923cfb829c08575df276 Mon Sep 17 00:00:00 2001
From: Subba Rao Pasupuleti <subbarao.pasupuleti@gmail.com>
Date: Fri, 6 Aug 2010 09:04:13 -0400
Subject: [PATCH] Add model require

---
 .../serializeration/json_serialization_test.rb     |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/activemodel/test/cases/serializeration/json_serialization_test.rb b/activemodel/test/cases/serializeration/json_serialization_test.rb
index 1ac991a..381124c 100644
--- a/activemodel/test/cases/serializeration/json_serialization_test.rb
+++ b/activemodel/test/cases/serializeration/json_serialization_test.rb
@@ -1,5 +1,6 @@
 require 'cases/helper'
 require 'models/contact'
+require 'models/automobile'
 require 'active_support/core_ext/object/instance_variables'
 
 class Contact
-- 
1.7.0.4

