From c19fbaeb3d76701384eba954a1edd1aaa4370ce4 Mon Sep 17 00:00:00 2001
From: Elliot Winkler <elliot.winkler@gmail.com>
Date: Wed, 15 Oct 2008 18:31:38 -0500
Subject: [PATCH] Added test for ARes::Connection#build_request_headers

---
 activeresource/test/connection_test.rb |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/activeresource/test/connection_test.rb b/activeresource/test/connection_test.rb
index 84bcf69..64f97f0 100644
--- a/activeresource/test/connection_test.rb
+++ b/activeresource/test/connection_test.rb
@@ -184,6 +184,11 @@ class ConnectionTest < Test::Unit::TestCase
       assert_nothing_raised(Mocha::ExpectationError) { @conn.get(path, {'Accept' => 'application/xhtml+xml'}) }
     end
   end
+  
+  def test_build_request_headers
+    headers = @conn.send(:build_request_headers, {})
+    assert(!headers.include?(nil))
+  end
 
   protected
     def assert_response_raises(klass, code)
-- 
1.5.5

