diff --git a/actionpack/test/controller/mime_responds_test.rb b/actionpack/test/controller/mime_responds_test.rb
index edd7162..598ee01 100644
--- a/actionpack/test/controller/mime_responds_test.rb
+++ b/actionpack/test/controller/mime_responds_test.rb
@@ -204,6 +204,12 @@ class MimeControllerTest < ActionController::TestCase
     assert_equal 'XML', @response.body
   end
 
+  def test_406_with_unspecified_format
+    @request.accept = "*/*"
+    get :just_xml
+    assert_response 406
+  end
+
   def test_js_or_html
     @request.accept = "text/javascript, text/html"
     get :js_or_html
@@ -471,6 +477,7 @@ class MimeControllerTest < ActionController::TestCase
     @request.accept = "text/iphone"
     assert_raise(ActionView::MissingTemplate) { get :iphone_with_html_response_type_without_layout }
   end
+
 end
 
 class AbstractPostController < ActionController::Base
