From 9aa3b272ad7e9f0d3b790bb37af972317ae7beaa Mon Sep 17 00:00:00 2001
From: Brian Lopez <seniorlopez@gmail.com>
Date: Tue, 4 May 2010 13:46:52 -0700
Subject: [PATCH] add support for mysql2 adapter to dbconsole

---
 railties/lib/commands/dbconsole.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/commands/dbconsole.rb b/railties/lib/commands/dbconsole.rb
index e6f11a4..31ffedd 100644
--- a/railties/lib/commands/dbconsole.rb
+++ b/railties/lib/commands/dbconsole.rb
@@ -45,7 +45,7 @@ def find_cmd(*commands)
 end
 
 case config["adapter"]
-when "mysql"
+when "mysql", "mysql2"
   args = {
     'host'      => '--host',
     'port'      => '--port',
-- 
1.7.1

