--- inflections.rb.orig	2008-11-03 23:15:37.000000000 -0800
+++ ./vendor/rails/activesupport/lib/active_support/inflections.rb	2008-11-03 23:20:25.000000000 -0800
@@ -17,6 +17,7 @@
     inflect.plural(/([m|l])ouse$/i, '\1ice')
     inflect.plural(/^(ox)$/i, '\1en')
     inflect.plural(/(quiz)$/i, '\1zes')
+    inflect.plural(/(cact|fung)us$/i, '\1i')
 
     inflect.singular(/s$/i, '')
     inflect.singular(/(n)ews$/i, '\1ews')
@@ -48,7 +49,7 @@
     inflect.irregular('child', 'children')
     inflect.irregular('sex', 'sexes')
     inflect.irregular('move', 'moves')
-    inflect.irregular('cow', 'kine')
+    inflect.irregular('die', 'dice')
 
     inflect.uncountable(%w(equipment information rice money species series fish sheep))
   end
