From 8174731a17c9beb998000dedc2334fdaba74ae49 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jakub=20Ku=C5=BAma?= <qoobaa@gmail.com>
Date: Mon, 2 Aug 2010 12:07:03 +0200
Subject: [PATCH] added timestamps to Pet and Toy models

---
 activerecord/test/schema/schema.rb |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/activerecord/test/schema/schema.rb b/activerecord/test/schema/schema.rb
index f3fd37c..a0e620c 100644
--- a/activerecord/test/schema/schema.rb
+++ b/activerecord/test/schema/schema.rb
@@ -398,6 +398,7 @@ ActiveRecord::Schema.define do
   create_table :pets, :primary_key => :pet_id ,:force => true do |t|
     t.string :name
     t.integer :owner_id, :integer
+    t.timestamps
   end
 
   create_table :pirates, :force => true do |t|
@@ -530,6 +531,7 @@ ActiveRecord::Schema.define do
   create_table :toys, :primary_key => :toy_id ,:force => true do |t|
     t.string :name
     t.integer :pet_id, :integer
+    t.timestamps
   end
 
   create_table :traffic_lights, :force => true do |t|
-- 
1.7.0.4

