From ec3108863e6fa7613fb251f1dd916c9a3e668665 Mon Sep 17 00:00:00 2001
From: David Lowenfels <david@internautdesign.com>
Date: Tue, 8 Jul 2008 18:34:20 -0700
Subject: [PATCH] requiring rubygems version 1.1.1

---
 railties/environments/boot.rb |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/railties/environments/boot.rb b/railties/environments/boot.rb
index cd21fb9..4bb0e06 100644
--- a/railties/environments/boot.rb
+++ b/railties/environments/boot.rb
@@ -82,17 +82,17 @@ module Rails
 
       def load_rubygems
         require 'rubygems'
-
-        unless rubygems_version >= '0.9.4'
-          $stderr.puts %(Rails requires RubyGems >= 0.9.4 (you have #{rubygems_version}). Please `gem update --system` and try again.)
+        min_version = '1.1.1'
+        unless rubygems_version >= min_version
+          $stderr.puts %Q(Rails requires RubyGems >= #{min_version} (you have #{rubygems_version}). Please `gem update --system` and try again.)
           exit 1
         end
 
       rescue LoadError
-        $stderr.puts %(Rails requires RubyGems >= 0.9.4. Please install RubyGems and try again: http://rubygems.rubyforge.org)
+        $stderr.puts %Q(Rails requires RubyGems >= #{min_version}. Please install RubyGems and try again: http://rubygems.rubyforge.org)
         exit 1
       end
-
+      
       def parse_gem_version(text)
         $1 if text =~ /^[^#]*RAILS_GEM_VERSION\s*=\s*["']([!~<>=]*\s*[\d.]+)["']/
       end
-- 
1.5.3.6

