From 8e60023eed8a95d9c0b2cf02838faa1cb9d6facb Mon Sep 17 00:00:00 2001
From: Ivan Torres <mexpolk@gmail.com>
Date: Tue, 13 Jul 2010 13:45:15 -0500
Subject: [PATCH] Adding .rvmc to .gitignore

---
 .gitignore |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/.gitignore b/.gitignore
index 2dfdf96..b593c9c 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,6 +1,7 @@
 *.gem
 pkg
 .bundle
+.rvmrc
 Gemfile.lock
 debug.log
 doc/rdoc
-- 
1.7.1

