diff --git a/.gitignore b/.gitignore
index 16bffc1..9f4a585 100644
--- a/.gitignore
+++ b/.gitignore
@@ -18,3 +18,4 @@ railties/test/initializer/root/log
 railties/doc
 railties/guides/output
 railties/tmp
+*.swp
diff --git a/railties/lib/rails/generators/rails/app/templates/public/javascripts/rails.js b/railties/lib/rails/generators/rails/app/templates/public/javascripts/rails.js
index c5fa02a..9a9ef9f 100644
--- a/railties/lib/rails/generators/rails/app/templates/public/javascripts/rails.js
+++ b/railties/lib/rails/generators/rails/app/templates/public/javascripts/rails.js
@@ -59,7 +59,11 @@ document.observe("dom:loaded", function() {
   }
 
   $(document.body).observe("click", function(event) {
-    var message = event.findElement().readAttribute('data-confirm');
+    var el = event.findElement();
+    if (el.tagName.toLowerCase() !== 'input' && el.tagName.toLowerCase() !== 'a') {
+      el = event.findElement('a[data-confirm]');
+    }
+    var message = el.readAttribute('data-confirm');
     if (message && !confirm(message)) {
       event.stop();
       return false;
@@ -115,4 +119,4 @@ document.observe("dom:loaded", function() {
       });
     }
   });
-});
\ No newline at end of file
+});
