From d3c1665b3d37914f5510c6e56dab969c885b766e Mon Sep 17 00:00:00 2001
From: Santiago Pastorino <santiago@wyeworks.com>
Date: Sun, 21 Mar 2010 23:36:29 -0300
Subject: [PATCH] Revert previous accidentally add of gem "system_timer"

---
 Gemfile |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/Gemfile b/Gemfile
index b40916b..a51c560 100644
--- a/Gemfile
+++ b/Gemfile
@@ -19,8 +19,6 @@ group :test do
   gem "mysql", ">= 2.8.1"
 end
 
-gem "system_timer"
-
 # AP
 gem "rack-test", "0.5.3", :require => 'rack/test'
 gem "RedCloth", ">= 4.2.2"
-- 
1.7.0

