From fcb32742d34c8c6e9d0c0058bd8ae9af6dc8d619 Mon Sep 17 00:00:00 2001
From: Ryan Bigg <radarlistener@gmail.com>
Date: Mon, 8 Feb 2010 09:50:39 +1000
Subject: [PATCH] Remove the final initializers since initializers += will return it anyway.

---
 railties/lib/rails/application.rb |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/railties/lib/rails/application.rb b/railties/lib/rails/application.rb
index 9384492..4306d39 100644
--- a/railties/lib/rails/application.rb
+++ b/railties/lib/rails/application.rb
@@ -100,7 +100,6 @@ module Rails
       railties.all { |r| initializers += r.initializers }
       initializers += super
       initializers += Finisher.initializers_for(self)
-      initializers
     end
 
   protected
-- 
1.6.4.1

