diff --git a/railties/lib/rails/generators/rails/app/templates/config.ru b/railties/lib/rails/generators/rails/app/templates/config.ru
index 30b00bf..0446265 100644
--- a/railties/lib/rails/generators/rails/app/templates/config.ru
+++ b/railties/lib/rails/generators/rails/app/templates/config.ru
@@ -1,2 +1,2 @@
-require File.dirname(__FILE__) + '/config/environment'
+require ::File.dirname(__FILE__) + '/config/environment'
 run ActionController::Dispatcher.new
