From 7a5cbed3ffc3ae02f24d0ff81e0926841a944cbc Mon Sep 17 00:00:00 2001
From: Noel Proffitt <noelp@ci.garden-grove.ca.us>
Date: Wed, 15 Sep 2010 14:53:15 -0700
Subject: [PATCH] fixed version check for disabling referential integrity [#5635 state:resolved]

---
 .../connection_adapters/postgresql_adapter.rb      |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
index 0a2bacd..42cdbfa 100644
--- a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
@@ -378,8 +378,9 @@ module ActiveRecord
       # REFERENTIAL INTEGRITY ====================================
 
       def supports_disable_referential_integrity?() #:nodoc:
-        version = query("SHOW server_version")[0][0].split('.')
-        version[0].to_i >= 8 && version[1].to_i >= 1
+        if postgres_version > 80100
+        #version = query("SHOW server_version")[0][0].split('.')
+        #version[0].to_i >= 8 && version[1].to_i >= 1
       rescue
         return false
       end
@@ -897,7 +898,7 @@ module ActiveRecord
             else
               # Mimic PGconn.server_version behavior
               begin
-                query('SELECT version()')[0][0] =~ /PostgreSQL (\d+)\.(\d+)\.(\d+)/
+                query('SELECT version()')[0][0] =~ /PostgreSQL (\d+)\.(\d+)\.?(\d+)?/
                 ($1.to_i * 10000) + ($2.to_i * 100) + $3.to_i
               rescue
                 0
-- 
1.5.4.3

