From 562b09b1885024637ce46ec99ab6d4064cd02ff8 Mon Sep 17 00:00:00 2001
From: Barry Sears <barry.sears@gmail.com>
Date: Wed, 29 Sep 2010 18:31:58 -0500
Subject: [PATCH] Make ActiveSupport load own version.

---
 activesupport/lib/active_support.rb |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/activesupport/lib/active_support.rb b/activesupport/lib/active_support.rb
index ba91e8b..d2a845d 100644
--- a/activesupport/lib/active_support.rb
+++ b/activesupport/lib/active_support.rb
@@ -43,6 +43,7 @@ module ActiveSupport
   autoload :FileUpdateChecker
   autoload :LogSubscriber
   autoload :Notifications
+  autoload :VERSION
 
   # TODO: Narrow this list down
   eager_autoload do
-- 
1.7.3

