From cdde718098c6246934209de5179920b0eb0b19f1 Mon Sep 17 00:00:00 2001
From: Micah Wedemeyer <mwedemeyer@micah.(none)>
Date: Sun, 25 May 2008 16:47:35 -0400
Subject: [PATCH] Updated MysqlAdapter to check for sslca option instead of sslkey when setting SSL connection options.

---
 .../connection_adapters/mysql_adapter.rb           |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
index f00a2c8..f94b0f2 100755
--- a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
@@ -83,7 +83,7 @@ module ActiveRecord
 
       require_mysql
       mysql = Mysql.init
-      mysql.ssl_set(config[:sslkey], config[:sslcert], config[:sslca], config[:sslcapath], config[:sslcipher]) if config[:sslkey]
+      mysql.ssl_set(config[:sslkey], config[:sslcert], config[:sslca], config[:sslcapath], config[:sslcipher]) if config[:sslca]
 
       ConnectionAdapters::MysqlAdapter.new(mysql, logger, [host, username, password, database, port, socket], config)
     end
-- 
1.5.2.5

