From d1cc5ccb827ef81f3399585d51d0778628f8c6df Mon Sep 17 00:00:00 2001
From: rohit <rohit.arondekar@gmail.com>
Date: Fri, 11 Jun 2010 08:04:38 +0530
Subject: [PATCH] Remove ruby-debug from Gemfile for JRuby, native gem linecache can't be built

---
 Gemfile |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/Gemfile b/Gemfile
index 19e532b..8c8ecf9 100644
--- a/Gemfile
+++ b/Gemfile
@@ -17,7 +17,6 @@ if mri || RUBY_ENGINE == "rbx"
   gem 'yajl-ruby'
   gem "nokogiri", ">= 1.4.0"
 elsif RUBY_ENGINE == "jruby"
-  gem "ruby-debug"
   gem "jruby-openssl"
 end
 
-- 
1.7.0.4

