From 77cfbfc27937c6e3d6d6fb053c205ab98c37b960 Mon Sep 17 00:00:00 2001
From: Rizwan Reza <rizwanreza@gmail.com>
Date: Sun, 16 May 2010 20:10:16 +0430
Subject: [PATCH] Take out stale tasks from Actionpack's Rakefile

---
 actionpack/Rakefile |   19 -------------------
 1 files changed, 0 insertions(+), 19 deletions(-)

diff --git a/actionpack/Rakefile b/actionpack/Rakefile
index b9ace86..f3bd7df 100644
--- a/actionpack/Rakefile
+++ b/actionpack/Rakefile
@@ -88,23 +88,4 @@ task :lines do
   end
 
   puts "Total: Lines #{total_lines}, LOC #{total_codelines}"
-end
-
-# Publishing ------------------------------------------------------
-
-task :update_scriptaculous do
-  for js in %w( controls dragdrop effects )
-    system("svn export --force http://dev.rubyonrails.org/svn/rails/spinoffs/scriptaculous/src/#{js}.js #{File.dirname(__FILE__)}/lib/action_view/helpers/javascripts/#{js}.js")
-  end
-end
-
-desc "Updates actionpack to the latest version of the javascript spinoffs"
-task :update_js => [ :update_scriptaculous ]
-
-# Publishing ------------------------------------------------------
-
-desc "Publish the API documentation"
-task :pdoc => [:rdoc] do
-  require 'rake/contrib/sshpublisher'
-  Rake::SshDirPublisher.new("wrath.rubyonrails.org", "public_html/ap", "doc").upload
 end
\ No newline at end of file
-- 
1.6.5.2

