diff --git a/actionpack/lib/action_view/helpers/form_helper.rb b/actionpack/lib/action_view/helpers/form_helper.rb
index 568687e..07ef6ba 100644
--- a/actionpack/lib/action_view/helpers/form_helper.rb
+++ b/actionpack/lib/action_view/helpers/form_helper.rb
@@ -908,6 +908,9 @@ module ActionView
           index = ""
         end
 
+        # propagate :builder for nested forms
+        args.last[:builder] ||= @options[:builder] if args.last.is_a?(Hash) && @options.has_key?(:builder)
+
         case record_or_name_or_array
         when String, Symbol
           if nested_attributes_association?(record_or_name_or_array)
