From 019203a0f304dc88e5da39e82b62816df3896881 Mon Sep 17 00:00:00 2001
From: Rizwan Reza <rizwanreza@gmail.com>
Date: Fri, 26 Mar 2010 13:09:38 +0430
Subject: [PATCH] Better MySQL Error message.

---
 .../connection_adapters/mysql_adapter.rb           |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
index 1f1df7e..b36e2cf 100644
--- a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
@@ -61,7 +61,7 @@ module ActiveRecord
         begin
           require_library_or_gem('mysql')
         rescue LoadError
-          $stderr.puts '!!! The bundled mysql.rb driver has been removed from Rails 2.2. Please install the mysql gem and try again: gem install mysql.'
+          $stderr.puts '!!! Please install the mysql gem and try again: gem install mysql.'
           raise
         end
       end
-- 
1.6.5.2

