From 0bcbf26d7a8691e41934ef954454208a1f79e0ac Mon Sep 17 00:00:00 2001
From: Dan Grigsby <dan@dan.local>
Date: Thu, 4 Mar 2010 12:58:27 -0600
Subject: [PATCH] fixed ujs links

---
 src/rails.js |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/rails.js b/src/rails.js
index f2a40a2..9d9ff94 100644
--- a/src/rails.js
+++ b/src/rails.js
@@ -86,7 +86,7 @@ jQuery(function ($) {
         var link = $(this),
             href = link.attr('href'),
             method = link.attr('data-method'),
-            form = $('<form method="post" action="'+href+'">'),
+            form = $('<form method="post" action="'+href+'" />'),
             metadata_input = '<input name="_method" value="'+method+'" type="hidden" />';
 
         if (csrf_param != null && csrf_token != null) {
-- 
1.6.6.1

