From 9ac1cfc001f1c0385d92ba43de9933e0245ba2b8 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Rapha=C3=ABl=20Valyi?= <rvalyi@localhost.(none)>
Date: Mon, 24 Nov 2008 00:44:47 +0100
Subject: [PATCH] code cleanup: unused 'final' variable removal

---
 .../routing/recognition_optimisation.rb            |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/actionpack/lib/action_controller/routing/recognition_optimisation.rb b/actionpack/lib/action_controller/routing/recognition_optimisation.rb
index 3b98b16..226b12c 100644
--- a/actionpack/lib/action_controller/routing/recognition_optimisation.rb
+++ b/actionpack/lib/action_controller/routing/recognition_optimisation.rb
@@ -98,7 +98,6 @@ module ActionController
           if Array === item
             i += 1
             start = (i == 1)
-            final = (i == list.size)
             tag, sub = item
             if tag == :dynamic
               body += padding + "#{start ? 'if' : 'elsif'} true\n"
-- 
1.5.6.3

