From 6a9c9854da7c79801c566d620fe501d119b75a98 Mon Sep 17 00:00:00 2001
From: Simon Jefford <simon.jefford@gmail.com>
Date: Thu, 15 Jul 2010 23:35:02 +0100
Subject: [PATCH] Added rack-mount gem to Gemfile

---
 Gemfile |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/Gemfile b/Gemfile
index e915135..4dbfa46 100644
--- a/Gemfile
+++ b/Gemfile
@@ -28,6 +28,9 @@ gem "memcache-client", ">= 1.8.5"
 # AM
 gem "text-format", "~> 1.0.0"
 
+# AP
+gem "rack-mount", "~> 0.6.9"
+
 # AR
 if mri || RUBY_ENGINE == "rbx"
   gem "sqlite3-ruby", "~> 1.3.1", :require => 'sqlite3'
-- 
1.7.1.1

