From 34cbc9668dc8c95241482061dcb2125d2e583705 Mon Sep 17 00:00:00 2001
From: Boba Fat <ivan@workisfun.ru>
Date: Mon, 23 Mar 2009 15:13:29 +0300
Subject: [PATCH] fixed .htaccess example indent

---
 railties/README |   80 +++++++++++++++++++++++++++---------------------------
 1 files changed, 40 insertions(+), 40 deletions(-)

diff --git a/railties/README b/railties/README
index 37ec8ea..db33f83 100644
--- a/railties/README
+++ b/railties/README
@@ -50,46 +50,46 @@ FCGI or proxy to a pack of Mongrels/Thin/Ebb servers.
 
 == Apache .htaccess example for FCGI/CGI
 
-# General Apache options
-AddHandler fastcgi-script .fcgi
-AddHandler cgi-script .cgi
-Options +FollowSymLinks +ExecCGI
-
-# If you don't want Rails to look in certain directories,
-# use the following rewrite rules so that Apache won't rewrite certain requests
-# 
-# Example:
-#   RewriteCond %{REQUEST_URI} ^/notrails.*
-#   RewriteRule .* - [L]
-
-# Redirect all requests not available on the filesystem to Rails
-# By default the cgi dispatcher is used which is very slow
-# 
-# For better performance replace the dispatcher with the fastcgi one
-#
-# Example:
-#   RewriteRule ^(.*)$ dispatch.fcgi [QSA,L]
-RewriteEngine On
-
-# If your Rails application is accessed via an Alias directive,
-# then you MUST also set the RewriteBase in this htaccess file.
-#
-# Example:
-#   Alias /myrailsapp /path/to/myrailsapp/public
-#   RewriteBase /myrailsapp
-
-RewriteRule ^$ index.html [QSA]
-RewriteRule ^([^.]+)$ $1.html [QSA]
-RewriteCond %{REQUEST_FILENAME} !-f
-RewriteRule ^(.*)$ dispatch.cgi [QSA,L]
-
-# In case Rails experiences terminal errors
-# Instead of displaying this message you can supply a file here which will be rendered instead
-# 
-# Example:
-#   ErrorDocument 500 /500.html
-
-ErrorDocument 500 "<h2>Application error</h2>Rails application failed to start properly"
+  # General Apache options
+  AddHandler fastcgi-script .fcgi
+  AddHandler cgi-script .cgi
+  Options +FollowSymLinks +ExecCGI
+
+  # If you don't want Rails to look in certain directories,
+  # use the following rewrite rules so that Apache won't rewrite certain requests
+  # 
+  # Example:
+  #   RewriteCond %{REQUEST_URI} ^/notrails.*
+  #   RewriteRule .* - [L]
+
+  # Redirect all requests not available on the filesystem to Rails
+  # By default the cgi dispatcher is used which is very slow
+  # 
+  # For better performance replace the dispatcher with the fastcgi one
+  #
+  # Example:
+  #   RewriteRule ^(.*)$ dispatch.fcgi [QSA,L]
+  RewriteEngine On
+
+  # If your Rails application is accessed via an Alias directive,
+  # then you MUST also set the RewriteBase in this htaccess file.
+  #
+  # Example:
+  #   Alias /myrailsapp /path/to/myrailsapp/public
+  #   RewriteBase /myrailsapp
+
+  RewriteRule ^$ index.html [QSA]
+  RewriteRule ^([^.]+)$ $1.html [QSA]
+  RewriteCond %{REQUEST_FILENAME} !-f
+  RewriteRule ^(.*)$ dispatch.cgi [QSA,L]
+
+  # In case Rails experiences terminal errors
+  # Instead of displaying this message you can supply a file here which will be rendered instead
+  # 
+  # Example:
+  #   ErrorDocument 500 /500.html
+
+  ErrorDocument 500 "<h2>Application error</h2>Rails application failed to start properly"
 
 
 == Debugging Rails
-- 
1.6.0

