From a35468decb1e68e4747c5d597c3ca632d48ed30c Mon Sep 17 00:00:00 2001
From: David Reese <david@whatcould.com>
Date: Sat, 7 Feb 2009 13:40:46 -0600
Subject: [PATCH] do not blast plugin view paths

---
 railties/lib/initializer.rb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index e3811dd..6bfe6cf 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -370,8 +370,8 @@ Run `rake gems:install` to install the missing gems.
     def load_view_paths
       if configuration.frameworks.include?(:action_view)
         if configuration.cache_classes
-          view_path = ActionView::Template::EagerPath.new(configuration.view_path)
-          ActionController::Base.view_paths = view_path if configuration.frameworks.include?(:action_controller)
+          view_paths = ActionController::Base.view_paths.map{|path| ActionView::Template::EagerPath.new(path.to_s)}
+          ActionController::Base.view_paths = view_paths if configuration.frameworks.include?(:action_controller)
           ActionMailer::Base.template_root = view_path if configuration.frameworks.include?(:action_mailer)
         end
       end
-- 
1.6.0

