diff --git a/activesupport/lib/active_support/cache/mem_cache_store.rb b/activesupport/lib/active_support/cache/mem_cache_store.rb
index b3769b8..9176c39 100644
--- a/activesupport/lib/active_support/cache/mem_cache_store.rb
+++ b/activesupport/lib/active_support/cache/mem_cache_store.rb
@@ -68,7 +68,7 @@ module ActiveSupport
       def decrement(key, amount = 1)
         log("decrement", key, amount)
         
-        response = data.decr(key, amount) 
+        response = @data.decr(key, amount)
         response == Response::NOT_FOUND ? nil : response
       rescue MemCache::MemCacheError 
         nil
