diff --git a/railties/lib/commands/plugin.rb b/railties/lib/commands/plugin.rb
index 9ff4739..0f5e611 100644
--- a/railties/lib/commands/plugin.rb
+++ b/railties/lib/commands/plugin.rb
@@ -278,8 +278,8 @@ class Plugin
         base_cmd += " #{options[:revision]}" if options[:revision]
         puts base_cmd if $verbose
         if system(base_cmd)
-          puts "removing: .git" if $verbose
-          rm_rf ".git"
+          puts "removing: .git*" if $verbose
+          rm_rf %w(.git .gitignore)
         else
           rm_rf install_path
         end
