--- /var/lib/gems/1.8/gems/actionpack-2.3.3/lib/action_controller/integration.rb.org	2009-08-15 11:11:38.000000000 +0200
+++ /var/lib/gems/1.8/gems/actionpack-2.3.3/lib/action_controller/integration.rb	2009-08-15 11:11:12.000000000 +0200
@@ -414,6 +414,12 @@
                 multipart_requestify(value, false).each do |subkey, subvalue|
                   p[k + subkey] = subvalue
                 end
+              elsif Array === value
+                value.each_with_index do |element, i|
+                  multipart_requestify(element, false).each do |subkey, subvalue|
+                    p[k + "[#{i}]" + subkey] = subvalue
+                  end
+                end
               else
                 p[k] = value
               end
