diff --git a/activerecord/test/cases/migration_test.rb b/activerecord/test/cases/migration_test.rb
index 2ec3d40..70446c4 100644
--- a/activerecord/test/cases/migration_test.rb
+++ b/activerecord/test/cases/migration_test.rb
@@ -788,6 +788,20 @@ if ActiveRecord::Base.connection.supports_migrations?
       assert_nil Person.new.first_name
     end
 
+    def test_change_column_does_not_lose_decimal_precision_and_scale
+      Person.connection.create_table :testings do |t|
+        t.column :price_on_head, :decimal, :scale => 8, :precision => 2
+      end
+      person_klass = Class.new(Person)
+      person_klass.set_table_name 'testings'
+      person_klass.connection.change_column "testings", "price_on_head", :decimal, :precision => 12, :scale => 5
+      person_klass.reset_column_information
+      assert_equal 12, person_klass.columns_hash["price_on_head"].precision
+      assert_equal 5,  person_klass.columns_hash["price_on_head"].scale
+    ensure
+      Person.connection.drop_table :testings rescue nil
+    end
+
     def test_add_table
       assert !Reminder.table_exists?
 
