--- a/vendor/plugins/acts_as_tsearch/lib/postgres_extensions.rb
+++ b/vendor/plugins/acts_as_tsearch/lib/postgres_extensions.rb
@@ -15,6 +15,7 @@ module ActiveRecord
       end
       
     
+    end
     class PostgreSQLColumn < Column #:nodoc:
       private
         alias_method :simplified_type_base, :simplified_type
@@ -32,7 +33,6 @@ module ActiveRecord
         end
     end
 
-    end
 
   end
 end

