From 24463f262f1a0dda1a549933dcb7ef96a1e228cc Mon Sep 17 00:00:00 2001
From: Chad Woolley <chad@pivotallabs.com>
Date: Mon, 8 Feb 2010 00:01:14 -0700
Subject: [PATCH] Campfire notification for CI

---
 ci/cruise_config.rb |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/ci/cruise_config.rb b/ci/cruise_config.rb
index 9c7fa98..eea8ae1 100644
--- a/ci/cruise_config.rb
+++ b/ci/cruise_config.rb
@@ -6,4 +6,10 @@ Project.configure do |project|
 
   project.build_command = 'sudo gem update --system && ruby ci/ci_build.rb'
   project.email_notifier.from = 'thewoolleyman@gmail.com'
+
+  begin
+    eval File.readlines("#{CRUISE_DATA_ROOT}/campfire_notifier_settings.rb").join
+  rescue
+    CruiseControl::Log.warn("Failed to load Campfire notifier settings")
+  end
 end
-- 
1.6.5.3

