--- a/actionpack/test/dispatch/routing_test.rb
+++ b/actionpack/test/dispatch/routing_test.rb
@@ -187,6 +187,7 @@ class TestRoutingMapper < ActionDispatch::IntegrationTest
       end
 
       resources :sheep
+      resources :taxis
 
       resources :clients do
         namespace :google do
@@ -977,6 +978,14 @@ class TestRoutingMapper < ActionDispatch::IntegrationTest
     end
   end
 
+  def test_named_paths_for_resources_with_irregular_plural_of_the_plural
+    with_test_routes do
+      assert_equal '/taxis/1', taxi_path(1)
+      assert_equal '/taxis', taxis_path
+      assert_raise(NameError) { taxes_path }
+    end
+  end
+
   def test_path_names
     with_test_routes do
       get '/pt/projetos'