From 7522a89d5bf19e82b056e4b93b42940278204bae Mon Sep 17 00:00:00 2001
From: Arya Asemanfar <arya.asemanfar@gmail.com>
Date: Tue, 13 May 2008 14:59:23 -0700
Subject: [PATCH] When raise_on_type_mismatch raises an exception with the same class name, tell them they probably changed the class and haven't restarted mongrel

---
 .../associations/association_proxy.rb              |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/associations/association_proxy.rb b/activerecord/lib/active_record/associations/association_proxy.rb
index 68503a3..f3c5c7a 100644
--- a/activerecord/lib/active_record/associations/association_proxy.rb
+++ b/activerecord/lib/active_record/associations/association_proxy.rb
@@ -210,7 +210,10 @@ module ActiveRecord
 
         def raise_on_type_mismatch(record)
           unless record.is_a?(@reflection.klass)
-            raise ActiveRecord::AssociationTypeMismatch, "#{@reflection.klass} expected, got #{record.class}"
+            possible_reason = if @reflection.klass.to_s == record.class.to_s 
+              " (did you change the #{record.class} class? try restarting mongrel)"
+            end
+            raise ActiveRecord::AssociationTypeMismatch, "#{@reflection.klass} expected, got #{record.class}#{possible_reason}"
           end
         end
 
-- 
1.5.4.4

