diff --git a/activesupport/lib/active_support/json/encoders/string.rb b/activesupport/lib/active_support/json/encoders/string.rb
index 5ef7979..dd90770 100644
--- a/activesupport/lib/active_support/json/encoders/string.rb
+++ b/activesupport/lib/active_support/json/encoders/string.rb
@@ -27,7 +27,7 @@ class String
       ActiveSupport::JSON::Encoding::ESCAPED_CHARS[s]
     }
     json.force_encoding('ascii-8bit') if respond_to?(:force_encoding)
-    json.gsub(/([\xC0-\xDF][\x80-\xBF]|
+    json.gsub(/([\xC2-\xDF][\x80-\xBF]|
              [\xE0-\xEF][\x80-\xBF]{2}|
              [\xF0-\xF7][\x80-\xBF]{3})+/nx) { |s|
       s.unpack("U*").pack("n*").unpack("H*")[0].gsub(/.{4}/, '\\\\u\&')
