From 4c20b5e9a3e86a9e6f5ec84de61153b5e4c2ca4d Mon Sep 17 00:00:00 2001
From: Akira Matsuda <ronnie@dio.jp>
Date: Mon, 27 Jul 2009 08:25:57 +0900
Subject: [PATCH] Ruby 1.9.2 compat: Array#* uses to_str instead of to_s to join values since Ruby 1.9.2

---
 .../abstract/schema_definitions.rb                 |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb b/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb
index 24c734c..eca6b2a 100644
--- a/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb
+++ b/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb
@@ -278,6 +278,7 @@ module ActiveRecord
         column_sql
       end
       alias to_s :to_sql
+      alias to_str :to_sql
 
       private
 
-- 
1.6.3.3

