From 4f6b9d6e1672dd9dd2454861e8d6811a7d5d11c8 Mon Sep 17 00:00:00 2001
From: Diego Carrion <dc.rec1@gmail.com>
Date: Sat, 5 Mar 2011 01:57:25 -0300
Subject: [PATCH] refactored Engine.#find by giving an explicit name to a variable and reducing calls

---
 railties/lib/rails/engine.rb |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/railties/lib/rails/engine.rb b/railties/lib/rails/engine.rb
index 7c26234..0db0aa7 100644
--- a/railties/lib/rails/engine.rb
+++ b/railties/lib/rails/engine.rb
@@ -382,9 +382,9 @@ module Rails
 
       # Finds engine with given path
       def find(path)
-        path = path.to_s
-        Rails::Engine::Railties.engines.find { |r|
-          File.expand_path(r.root.to_s) == File.expand_path(path)
+        expanded_path = File.expand_path path.to_s
+        Rails::Engine::Railties.engines.find { |engine|
+          File.expand_path(engine.root.to_s) == expanded_path
         }
       end
     end
-- 
1.6.5.2

