From 6bdb89198204efba902c727627718e6f9be145dc Mon Sep 17 00:00:00 2001
From: Maksym <max@max.office.waveaccess.kiev.ua>
Date: Mon, 23 Nov 2009 18:05:49 +0200
Subject: [PATCH] possible postgresql schema fix

---
 .../connection_adapters/postgresql_adapter.rb      |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
index 1d52c5e..6924ca0 100644
--- a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
@@ -764,7 +764,7 @@ module ActiveRecord
         end
         if pk
           if sequence
-            quoted_sequence = quote_column_name(sequence)
+            quoted_sequence = quote_table_name(sequence)
 
             select_value <<-end_sql, 'Reset sequence'
               SELECT setval('#{quoted_sequence}', (SELECT COALESCE(MAX(#{quote_column_name pk})+(SELECT increment_by FROM #{quoted_sequence}), (SELECT min_value FROM #{quoted_sequence})) FROM #{quote_table_name(table)}), false)
@@ -780,7 +780,7 @@ module ActiveRecord
         # First try looking for a sequence with a dependency on the
         # given table's primary key.
         result = query(<<-end_sql, 'PK and serial sequence')[0]
-          SELECT attr.attname, seq.relname
+          SELECT attr.attname, name.nspname || '.' || seq.relname as relname
           FROM pg_class      seq,
                pg_attribute  attr,
                pg_depend     dep,
@@ -793,6 +793,7 @@ module ActiveRecord
             AND attr.attrelid     = cons.conrelid
             AND attr.attnum       = cons.conkey[1]
             AND cons.contype      = 'p'
+            AND name.oid          = seq.relnamespace
             AND dep.refobjid      = '#{quote_table_name(table)}'::regclass
         end_sql
 
-- 
1.6.3.3

