diff --git a/vendor/gems/actionmailer-3.0.0.beta/lib/action_mailer/quoting.rb b/vendor/gems/actionmailer-3.0.0.beta/lib/action_mailer/quoting.rb
index 70f636b..61efc39 100644
--- a/vendor/gems/actionmailer-3.0.0.beta/lib/action_mailer/quoting.rb
+++ b/vendor/gems/actionmailer-3.0.0.beta/lib/action_mailer/quoting.rb
@@ -22,7 +22,7 @@ module ActionMailer
     # A quick-and-dirty regexp for determining whether a string contains any
     # characters that need escaping.
     if !defined?(CHARS_NEEDING_QUOTING)
-      CHARS_NEEDING_QUOTING = /[\000-\011\013\014\016-\037\177-\377]/
+      CHARS_NEEDING_QUOTING = Regexp.new '[\000-\011\013\014\016-\037\177-\377]', nil, 'n'
     end
 
     # Quote the given text if it contains any "illegal" characters

