From 1a37897dd2fef8c171510fa063ec17b40f3e9444 Mon Sep 17 00:00:00 2001
From: Joe Van Dyk <joe@fixieconsulting.com>
Date: Sat, 26 Sep 2009 16:46:27 -0700
Subject: [PATCH] Mocha is required for most of the tests

---
 activerecord/Rakefile |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/activerecord/Rakefile b/activerecord/Rakefile
index 1d8062e..8c52e81 100644
--- a/activerecord/Rakefile
+++ b/activerecord/Rakefile
@@ -1,5 +1,6 @@
 require 'rubygems'
 require 'rake'
+require 'mocha'
 require 'rake/testtask'
 require 'rake/rdoctask'
 require 'rake/packagetask'
-- 
1.6.3.2

