From bd27badd510f9adc29a126434bf0f4aa59c93723 Mon Sep 17 00:00:00 2001
From: Xavier Noria <fxn@hashref.com>
Date: Mon, 15 Feb 2010 19:48:23 +0100
Subject: [PATCH] adds rdoc dependency to generate the API

---
 Gemfile |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/Gemfile b/Gemfile
index 7fbf57b..3756e29 100644
--- a/Gemfile
+++ b/Gemfile
@@ -22,6 +22,10 @@ end
 gem "rack-test", "0.5.3", :require => 'rack/test'
 gem "RedCloth", ">= 4.2.2"
 
+group :documentation do
+  gem 'rdoc', '2.1'
+end
+
 if ENV['CI']
   gem "nokogiri", ">= 1.4.0"
 
-- 
1.6.6

