From 1d389f079158d410bad012d24f9a869324d173b2 Mon Sep 17 00:00:00 2001
From: David Reese <david@whatcould.com>
Date: Thu, 6 Aug 2009 12:51:51 -0500
Subject: [PATCH] Make BlankSlate play nice with Ruby 1.9#methods

---
 .../vendor/builder-2.1.2/blankslate.rb             |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/activesupport/lib/active_support/vendor/builder-2.1.2/blankslate.rb b/activesupport/lib/active_support/vendor/builder-2.1.2/blankslate.rb
index da6034d..bc17eeb 100644
--- a/activesupport/lib/active_support/vendor/builder-2.1.2/blankslate.rb
+++ b/activesupport/lib/active_support/vendor/builder-2.1.2/blankslate.rb
@@ -20,7 +20,8 @@ class BlankSlate
     # Hide the method named +name+ in the BlankSlate class.  Don't
     # hide +instance_eval+ or any method beginning with "__".
     def hide(name)
-      if instance_methods.include?(name.to_s) and
+      method_name = RUBY_VERSION < '1.9' ? name.to_s : name.to_sym
+      if instance_methods.include?(method_name) and
         name !~ /^(__|instance_eval)/
         @hidden_methods ||= {}
         @hidden_methods[name.to_sym] = instance_method(name)
-- 
1.6.0

