diff --git a/activerecord/lib/active_record/base.rb b/activerecord/lib/active_record/base.rb
index fa239ba..d97425e 100644
--- a/activerecord/lib/active_record/base.rb
+++ b/activerecord/lib/active_record/base.rb
@@ -684,6 +684,10 @@ module ActiveRecord #:nodoc:
       def set_table_name(value = nil, &block)
         @quoted_table_name = nil
         define_attr_method :table_name, value, &block
+
+        # update memoized variables
+        @arel_table &&= Arel::Table.new(table_name, :engine => arel_engine)
+        @unscoped &&= Relation.new(self, arel_table)
       end
       alias :table_name= :set_table_name
 
