diff --git a/activerecord/lib/active_record/base.rb b/activerecord/lib/active_record/base.rb
index 039e144..371ee40 100755
--- a/activerecord/lib/active_record/base.rb
+++ b/activerecord/lib/active_record/base.rb
@@ -905,7 +905,8 @@ module ActiveRecord #:nodoc:
           sign = increment < 0 ? "-" : "+"
           list << "#{connection.quote_column_name(counter_name)} = COALESCE(#{connection.quote_column_name(counter_name)}, 0) #{sign} #{increment.abs}"
         }.join(", ")
-        update_all(updates, "#{connection.quote_column_name(primary_key)} = #{quote_value(id)}")
+        condition  = id.is_a?(Array)? "IN  (#{id.map { |i| quote_value(i) }.join(',')})" : "= #{quote_value(id)}"
+        update_all(updates, "#{connection.quote_column_name(primary_key)} #{condition}")
       end
 
       # Increment a number field by one, usually representing a count.
