From b9246f64452b2ab2af4423ffeb16ed4631dd4392 Mon Sep 17 00:00:00 2001
From: Trotter Cashion <cashion@gmail.com>
Date: Thu, 9 Jul 2009 12:49:58 -0400
Subject: [PATCH 1/3] Add support for jdbc versions of mysql and postgresql adapters

---
 railties/lib/tasks/databases.rake |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index 0e25673..5b08aaa 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -53,7 +53,7 @@ namespace :db do
       end
     rescue
       case config['adapter']
-      when 'mysql'
+      when /mysql/
         @charset   = ENV['CHARSET']   || 'utf8'
         @collation = ENV['COLLATION'] || 'utf8_general_ci'
         creation_options = {:charset => (config['charset'] || @charset), :collation => (config['collation'] || @collation)}
@@ -79,7 +79,7 @@ namespace :db do
             $stderr.puts "(if you set the charset manually, make sure you have a matching collation)" if config['charset']
           end
         end
-      when 'postgresql'
+      when /postgresql/
         @encoding = config[:encoding] || ENV['CHARSET'] || 'utf8'
         begin
           ActiveRecord::Base.establish_connection(config.merge('database' => 'postgres', 'schema_search_path' => 'public'))
-- 
1.6.3.3


From 65b3b4e84f352e6cb81890f3941008e1d5661a58 Mon Sep 17 00:00:00 2001
From: Trotter Cashion <cashion@gmail.com>
Date: Thu, 9 Jul 2009 12:56:04 -0400
Subject: [PATCH 2/3] Also print a message if your adapters is unsupported

---
 railties/lib/tasks/databases.rake |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index 5b08aaa..14413ae 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -89,6 +89,8 @@ namespace :db do
           $stderr.puts $!, *($!.backtrace)
           $stderr.puts "Couldn't create database for #{config.inspect}"
         end
+      else
+        $stderr.puts "#{config['adapter']} is not supported for db creation"
       end
     else
       $stderr.puts "#{config['database']} already exists"
-- 
1.6.3.3


From ca429d3c4f9009558a5083fbf60442c689167c1c Mon Sep 17 00:00:00 2001
From: Trotter Cashion <cashion@gmail.com>
Date: Wed, 15 Jul 2009 08:39:40 -0400
Subject: [PATCH 3/3] Add support for dropping databases as well

---
 railties/lib/tasks/databases.rake |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index 14413ae..ea5dee7 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -427,14 +427,16 @@ end
 
 def drop_database(config)
   case config['adapter']
-  when 'mysql'
+  when /mysql/
     ActiveRecord::Base.establish_connection(config)
     ActiveRecord::Base.connection.drop_database config['database']
   when /^sqlite/
     FileUtils.rm(File.join(RAILS_ROOT, config['database']))
-  when 'postgresql'
+  when /postgresql/
     ActiveRecord::Base.establish_connection(config.merge('database' => 'postgres', 'schema_search_path' => 'public'))
     ActiveRecord::Base.connection.drop_database config['database']
+  else
+    $stderr.puts "#{config['adapter']} is not supported for db creation"
   end
 end
 
-- 
1.6.3.3

