diff --git a/activerecord/lib/active_record/validations.rb b/activerecord/lib/active_record/validations.rb
index 7b70f78..eeeabe4 100755
--- a/activerecord/lib/active_record/validations.rb
+++ b/activerecord/lib/active_record/validations.rb
@@ -656,7 +656,7 @@ module ActiveRecord
           results = finder_class.with_exclusive_scope do
             connection.select_all(
               construct_finder_sql(
-                :select     => "#{attr_name}",
+                :select     => "#{finder_class.quoted_table_name}.#{attr_name}",
                 :from       => "#{finder_class.quoted_table_name}",
                 :conditions => [condition_sql, *condition_params]
               )
