--- actionmailer/lib/action_mailer/base.rb.orig        2010-09-04 01:45:45.000000000 +0000
+++ actionmailer/lib/action_mailer/base.rb     2010-09-04 01:45:19.000000000 +0000
@@ -687,7 +687,7 @@
 
       def perform_delivery_sendmail(mail)
         sendmail_args = sendmail_settings[:arguments]
-        sendmail_args += " -f \"#{mail['return-path']}\"" if mail['return-path']
+        sendmail_args += " -f \"#{mail['return-path'].spec}\"" if mail['return-path']
         IO.popen("#{sendmail_settings[:location]} #{sendmail_args}","w+") do |sm|
           sm.print(mail.encoded.gsub(/\r/, ''))
           sm.flush