diff --git a/activesupport/lib/active_support/core_ext/file/atomic.rb b/activesupport/lib/active_support/core_ext/file/atomic.rb
index 4d3cf54..0b35e27 100644
--- a/activesupport/lib/active_support/core_ext/file/atomic.rb
+++ b/activesupport/lib/active_support/core_ext/file/atomic.rb
@@ -28,7 +28,8 @@ module ActiveSupport #:nodoc:
           rescue Errno::ENOENT
             # No old permissions, write a temp file to determine the defaults
             check_name = ".permissions_check.#{Thread.current.object_id}.#{Process.pid}.#{rand(1000000)}"
-            new(check_name, "w")
+            check_file = new(check_name, "w")
+            check_file.close
             old_stat = stat(check_name)
             unlink(check_name)
           end
