From 81ee0d63b134d3b915cf6cfaac895c2bc2631c5d Mon Sep 17 00:00:00 2001
From: Ricardo Santos <oterosantos@gmail.com>
Date: Tue, 1 Jul 2008 16:33:40 +0100
Subject: [PATCH] The script/plugin unsource 'Usage' text is now correct

---
 railties/lib/commands/plugin.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/commands/plugin.rb b/railties/lib/commands/plugin.rb
index 105819c..ce4b0d0 100644
--- a/railties/lib/commands/plugin.rb
+++ b/railties/lib/commands/plugin.rb
@@ -632,7 +632,7 @@ module Commands
     def options
       OptionParser.new do |o|
         o.set_summary_indent('  ')
-        o.banner =    "Usage: #{@base_command.script_name} source URI [URI [URI]...]"
+        o.banner =    "Usage: #{@base_command.script_name} unsource URI [URI [URI]...]"
         o.define_head "Remove repositories from the default search list."
         o.separator ""
         o.on_tail("-h", "--help", "Show this help message.") { puts o; exit }
-- 
1.5.4.5

