From 0f9346781887c325f7586357991bd42235faebfb Mon Sep 17 00:00:00 2001
From: Ben Marini <bmarini@gmail.com>
Date: Mon, 12 Oct 2009 20:27:02 -0700
Subject: [PATCH] Removed duplicate attr_accessor :gems from Rails::Configuration

---
 railties/lib/rails/configuration.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/rails/configuration.rb b/railties/lib/rails/configuration.rb
index 4a70a48..1984c3c 100644
--- a/railties/lib/rails/configuration.rb
+++ b/railties/lib/rails/configuration.rb
@@ -4,7 +4,7 @@ module Rails
                   :load_once_paths, :gems_dependencies_loaded, :after_initialize_blocks,
                   :frameworks, :framework_root_path, :root_path, :plugin_paths, :plugins,
                   :plugin_loader, :plugin_locators, :gems, :loaded_plugins, :reload_plugins,
-                  :i18n, :gems, :whiny_nils, :consider_all_requests_local,
+                  :i18n, :whiny_nils, :consider_all_requests_local,
                   :action_controller, :active_record, :action_view, :active_support,
                   :action_mailer, :active_resource,
                   :log_path, :log_level, :logger, :preload_frameworks,
-- 
1.6.4.2

