From f0148daa53f2cfaa8242114aefb8f8c3308a181a Mon Sep 17 00:00:00 2001
From: Jan Varwig <jan@varwig.org>
Date: Wed, 16 Dec 2009 19:56:05 +0100
Subject: [PATCH] Make databases.rake honor settings in the true Rails.configuration instead of working with a blank one

---
 railties/lib/tasks/databases.rake |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index 8b60839..22176bb 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -1,7 +1,6 @@
 namespace :db do
-  task :load_config => :rails_env do
-    require 'active_record'
-    ActiveRecord::Base.configurations = Rails::Configuration.new.database_configuration
+  task :load_config => :environment do
+    ActiveRecord::Base.configurations = Rails.configuration.database_configuration
   end
 
   namespace :create do
-- 
1.6.4.1

