From 81847d4ac012d254fb9e6f8046bda4887d64aaa2 Mon Sep 17 00:00:00 2001
From: Ryan Bigg <radarlistener@gmail.com>
Date: Fri, 8 Oct 2010 16:17:36 +0530
Subject: [PATCH] Remove the final initializers since initializers += will return it anyway. [#3892 state:resolved]

---
 railties/lib/rails/application.rb |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/railties/lib/rails/application.rb b/railties/lib/rails/application.rb
index 5559f49..2c5d166 100644
--- a/railties/lib/rails/application.rb
+++ b/railties/lib/rails/application.rb
@@ -131,7 +131,6 @@ module Rails
       initializers = Bootstrap.initializers_for(self)
       initializers += super
       initializers += Finisher.initializers_for(self)
-      initializers
     end
 
     def config
-- 
1.7.0.4

