Index: flash.rb
===================================================================
--- flash.rb	(revision 1054)
+++ flash.rb	(revision 1056)
@@ -152,7 +152,11 @@
         # to put a new one.
         def flash #:doc:
           unless defined? @_flash
-            @_flash = session["flash"] ||= FlashHash.new
+           if session.send(:loaded?)
+             @_flash = session["flash"] ||= FlashHash.new
+           else
+             @_flash = FlashHash.new
+           end
             @_flash.sweep
           end

