diff --git a/vendor/rails/actionpack/lib/action_controller/url_rewriter.rb b/vendor/rails/actionpack/lib/action_controller/url_rewriter.rb
index bb6cb43..d9d8097 100644
--- a/vendor/rails/actionpack/lib/action_controller/url_rewriter.rb
+++ b/vendor/rails/actionpack/lib/action_controller/url_rewriter.rb
@@ -171,6 +171,8 @@ module ActionController
     private
       # Given a path and options, returns a rewritten URL string
       def rewrite_url(options)
+        options = options.symbolize_keys
+
         rewritten_url = ""
 
         unless options[:only_path]
