diff --git a/railties/guides/rails_guides/generator.rb b/railties/guides/rails_guides/generator.rb
index 2a4714b..cebf6ae 100644
--- a/railties/guides/rails_guides/generator.rb
+++ b/railties/guides/rails_guides/generator.rb
@@ -71,8 +71,8 @@ module RailsGuides
 
       header = textile(header)
 
-      view.content_for(:page_title) { page_title }
-      view.content_for(:header_section) { header }
+      view.content_for(:page_title) { page_title.html_safe! }
+      view.content_for(:header_section) { header.html_safe! }
       new_body
     end
 
@@ -103,7 +103,7 @@ module RailsGuides
       index << '</ol>'
       index << '</div>'
 
-      view.content_for(:index_section) { index }
+      view.content_for(:index_section) { index.html_safe! }
 
       i.result
     end
diff --git a/railties/guides/source/layout.html.erb b/railties/guides/source/layout.html.erb
index eb66366..7dfcf4a 100644
--- a/railties/guides/source/layout.html.erb
+++ b/railties/guides/source/layout.html.erb
@@ -87,7 +87,7 @@
   <div id="container">
     <div class="wrapper">
       <div id="mainCol">
-        <%= yield %>
+        <%= yield.html_safe! %>
       </div>
     </div>
   </div>
