From 81761822b90134bbe190c934309d70e5f2dfde00 Mon Sep 17 00:00:00 2001
From: David Chelimsky <dchelimsky@gmail.com>
Date: Sun, 21 Feb 2010 09:31:23 -0600
Subject: [PATCH] fix small doc bug about running single unit test files

---
 actionpack/RUNNING_UNIT_TESTS |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/actionpack/RUNNING_UNIT_TESTS b/actionpack/RUNNING_UNIT_TESTS
index 16b1198..95a8bc7 100644
--- a/actionpack/RUNNING_UNIT_TESTS
+++ b/actionpack/RUNNING_UNIT_TESTS
@@ -11,7 +11,7 @@ Rake can be found at http://rake.rubyforge.org
 If you only want to run a single test suite, or don't want to bother with Rake,
 you can do so with something like:
 
-   ruby controller/base_tests.rb
+   ruby -Itest test/controller/base_tests.rb
 
 == Dependency on ActiveRecord and database setup
 
-- 
1.6.4.2

