From cbbc4878a7705661f4aa05e6a62debfb2c83ebae Mon Sep 17 00:00:00 2001
From: Hongli Lai (Phusion) <hongli@phusion.nl>
Date: Wed, 3 Dec 2008 19:50:08 +0100
Subject: [PATCH] Require mocha >= 0.9.3, older versions don't work anymore.

---
 activerecord/test/cases/helper.rb |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/activerecord/test/cases/helper.rb b/activerecord/test/cases/helper.rb
index 2382bfe..afba715 100644
--- a/activerecord/test/cases/helper.rb
+++ b/activerecord/test/cases/helper.rb
@@ -24,6 +24,7 @@ end
 
 def uses_mocha(description)
   require 'rubygems'
+  gem 'mocha', '>= 0.9.3'
   require 'mocha'
   yield
 rescue LoadError
-- 
1.6.0.4

