From cb5e7e8c8ab4554aec6b34dc0991857748895df7 Mon Sep 17 00:00:00 2001
From: Chris Griego <cgriego@gmail.com>
Date: Fri, 25 Jun 2010 23:30:19 -0500
Subject: [PATCH] Declare ActiveModel to be dependent on tzinfo since ActiveModel::Observing fails without it.

---
 activemodel/activemodel.gemspec |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/activemodel/activemodel.gemspec b/activemodel/activemodel.gemspec
index 6bd6fe4..8f8223b 100644
--- a/activemodel/activemodel.gemspec
+++ b/activemodel/activemodel.gemspec
@@ -22,4 +22,5 @@ Gem::Specification.new do |s|
   s.add_dependency('activesupport', version)
   s.add_dependency('builder',       '~> 2.1.2')
   s.add_dependency('i18n',          '~> 0.4.1')
+  s.add_dependency('tzinfo',        '~> 0.3.16')
 end
-- 
1.6.5

