--- ./gems/1.8/gems/activerecord-2.2.2/lib/active_record/fixtures.rb.org	2009-07-03 11:34:08.000000000 +0200
+++ ./gems/1.8/gems/activerecord-2.2.2/lib/active_record/fixtures.rb	2009-07-03 12:10:03.000000000 +0200
@@ -773,6 +773,7 @@
   def initialize(fixture, model_class)
     @fixture = fixture
     @model_class = model_class.is_a?(Class) ? model_class : model_class.constantize rescue nil
+    @connection = @model_class ? @model_class.connection : ActiveRecord::Base.connection
   end
 
   def class_name
@@ -792,14 +793,14 @@
   end
 
   def key_list
-    columns = @fixture.keys.collect{ |column_name| ActiveRecord::Base.connection.quote_column_name(column_name) }
+    columns = @fixture.keys.collect{ |column_name| @connection.quote_column_name(column_name) }
     columns.join(", ")
   end
 
   def value_list
     list = @fixture.inject([]) do |fixtures, (key, value)|
       col = model_class.columns_hash[key] if model_class.respond_to?(:ancestors) && model_class.ancestors.include?(ActiveRecord::Base)
-      fixtures << ActiveRecord::Base.connection.quote(value, col).gsub('[^\]\\n', "\n").gsub('[^\]\\r', "\r")
+      fixtures << @connection.quote(value, col).gsub('[^\]\\n', "\n").gsub('[^\]\\r', "\r")
     end
     list * ', '
   end
