--- actionpack-3.0.pre/lib/action_view/helpers/date_helper.rb.orig	2010-01-28 20:16:48.000000000 +0100
+++ actionpack-3.0.pre/lib/action_view/helpers/date_helper.rb	2010-01-28 20:18:47.000000000 +0100
@@ -865,7 +865,7 @@
             :id => input_id_from_type(type),
             :name => input_name_from_type(type),
             :value => value
-          }) + "\n").html_safe!
+          }.merge(@html_options)) + "\n").html_safe!
         end
 
         # Returns the name attribute for the input tag
