From b960de8d323d8d53137b708a15a8dbc1449fde99 Mon Sep 17 00:00:00 2001
From: Roger Pack <rogerpack@roger-packs-macbook-pro.local>
Date: Fri, 23 May 2008 15:57:06 -0600
Subject: [PATCH] add the exception message to the generic 'rubygems failed to load' error output

---
 railties/environments/boot.rb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/railties/environments/boot.rb b/railties/environments/boot.rb
index cd21fb9..ff9982b 100644
--- a/railties/environments/boot.rb
+++ b/railties/environments/boot.rb
@@ -88,8 +88,8 @@ module Rails
           exit 1
         end
 
-      rescue LoadError
-        $stderr.puts %(Rails requires RubyGems >= 0.9.4. Please install RubyGems and try again: http://rubygems.rubyforge.org)
+      rescue LoadError => e
+        $stderr.puts %(Rails requires RubyGems >= 0.9.4. Please install RubyGems and try again: http://rubygems.rubyforge.org --#{e})
         exit 1
       end
 
-- 
1.5.5.1

