From d9422ca0ee717825b467ffc15d601c0b1bfedbbb Mon Sep 17 00:00:00 2001
From: Christopher Currie <christopher@sixslicestudios.com>
Date: Sat, 25 Oct 2008 21:39:49 -0700
Subject: [PATCH] Update non-gems distributions to use sqlite3 as the default, matching the "rails" command default


diff --git a/railties/Rakefile b/railties/Rakefile
index adb6db0..872ea83 100644
--- a/railties/Rakefile
+++ b/railties/Rakefile
@@ -192,7 +192,7 @@ task :copy_configs do
   app_name = "rails"
   socket = nil
   require 'erb'
-  File.open("#{PKG_DESTINATION}/config/database.yml", 'w') {|f| f.write ERB.new(IO.read("configs/databases/mysql.yml"), nil, '-').result(binding)}
+  File.open("#{PKG_DESTINATION}/config/database.yml", 'w') {|f| f.write ERB.new(IO.read("configs/databases/sqlite3.yml"), nil, '-').result(binding)}
   
   cp "configs/routes.rb", "#{PKG_DESTINATION}/config/routes.rb"
 
-- 
1.5.4.3

