:100644 100644 f9fc7a0... 0000000... M	actionpack/test/controller/test_test.rb

diff --git a/actionpack/test/controller/test_test.rb b/actionpack/test/controller/test_test.rb
index f9fc7a0..9b96940 100644
--- a/actionpack/test/controller/test_test.rb
+++ b/actionpack/test/controller/test_test.rb
@@ -490,6 +490,23 @@ XML
     end
   end
 
+  def test_constraints_matching_properly_in_test_requests
+    with_routing do |set|
+      set.draw do |map|
+        resources :users do
+          scope :constraints => {:id => /\d+/} do
+            collection do
+              match "test/:id" => "test_test/test#test_params"
+            end
+          end
+        end
+      end
+
+      get :test_params, :id => "abc123", :user_id => 1
+      assert_response :success
+    end
+  end
+
   def test_assert_realistic_path_parameters
     get :test_params, :id => 20, :foo => Object.new
 
