diff --git a/activerecord/lib/active_record/associations/has_many_through_association.rb b/activerecord/lib/active_record/associations/has_many_through_association.rb
index 6af14a5..1d2928a 100644
--- a/activerecord/lib/active_record/associations/has_many_through_association.rb
+++ b/activerecord/lib/active_record/associations/has_many_through_association.rb
@@ -122,7 +122,8 @@ module ActiveRecord
                 @owner.class.base_class.name.to_s,
                 reflection.klass.columns_hash["#{as}_type"]) }
           elsif reflection.macro == :belongs_to
-            { reflection.klass.primary_key => @owner[reflection.primary_key_name] }
+            { reflection.klass.primary_key =>
+              reflection.klass.quote_value(@owner[reflection.primary_key_name]) }
           else
             { reflection.primary_key_name => owner_quoted_id }
           end

