diff --git a/railties/lib/commands/server.rb b/railties/lib/commands/server.rb
index ebe34a4..f5c371e 100644
--- a/railties/lib/commands/server.rb
+++ b/railties/lib/commands/server.rb
@@ -52,7 +52,6 @@ unless server
 end
 
 puts "=> Booting #{ActiveSupport::Inflector.demodulize(server)}"
-puts "=> Rails #{Rails.version} application starting on http://#{options[:Host]}:#{options[:Port]}#{options[:path]}"
 
 %w(cache pids sessions sockets).each do |dir_to_make|
   FileUtils.mkdir_p(File.join(RAILS_ROOT, 'tmp', dir_to_make))
@@ -86,12 +85,14 @@ else
 end
 
 if options[:path].nil?
-  map_path = "/"
+  map_path = ActionController::Base.relative_url_root || "/"
 else
   ActionController::Base.relative_url_root = options[:path]
   map_path = options[:path]
 end
 
+puts "=> Rails #{Rails.version} application starting on http://#{options[:Host]}:#{options[:Port]}#{map_path}" 
+
 app = Rack::Builder.new {
   use Rails::Rack::LogTailer unless options[:detach]
   use Rails::Rack::Debugger if options[:debugger]
