From 3e9fcc5df1a3048cbcb06e8b92f83f1a1345bb23 Mon Sep 17 00:00:00 2001
From: Matthias Hennemeyer <mhennemeyer@noname.(none)>
Date: Wed, 2 Dec 2009 14:29:45 +0100
Subject: [PATCH] Seperate tests for validate uniqueness and validate uniqueness with utf characters

---
 .../validations/uniqueness_validation_test.rb      |   13 +++++++++++--
 1 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/activerecord/test/cases/validations/uniqueness_validation_test.rb b/activerecord/test/cases/validations/uniqueness_validation_test.rb
index 17ba4e2..45f26b5 100644
--- a/activerecord/test/cases/validations/uniqueness_validation_test.rb
+++ b/activerecord/test/cases/validations/uniqueness_validation_test.rb
@@ -44,13 +44,13 @@ class UniquenessValidationTest < ActiveRecord::TestCase
   def test_validate_uniqueness
     Topic.validates_uniqueness_of(:title)
 
-    t = Topic.new("title" => "I'm uniqué!")
+    t = Topic.new("title" => "I'm unique!")
     assert t.save, "Should save t as unique"
 
     t.content = "Remaining unique"
     assert t.save, "Should still save t as unique"
 
-    t2 = Topic.new("title" => "I'm uniqué!")
+    t2 = Topic.new("title" => "I'm unique!")
     assert !t2.valid?, "Shouldn't be valid"
     assert !t2.save, "Shouldn't save t2 as unique"
     assert_equal ["has already been taken"], t2.errors[:title]
@@ -58,6 +58,15 @@ class UniquenessValidationTest < ActiveRecord::TestCase
     t2.title = "Now Im really also unique"
     assert t2.save, "Should now save t2 as unique"
   end
+  
+  def test_validate_uniqueness_with_utf8_chars
+    Topic.validates_uniqueness_of(:title)
+
+    Topic.create("title"  => "一二三四五")
+    t = Topic.new("title" => "一二三四五")
+    assert !t.valid?, "Shouldn't be valid"
+    assert_equal ["has already been taken"], t.errors[:title]
+  end
 
   def test_validates_uniqueness_with_newline_chars
     Topic.validates_uniqueness_of(:title, :case_sensitive => false)
-- 
1.6.4.2

