From 33a4313e8c89a1125eb51f03a626daf7afe24ed3 Mon Sep 17 00:00:00 2001
From: Josh Nichols <josh@technicalpickles.com>
Date: Sun, 29 Jun 2008 18:52:41 -0400
Subject: [PATCH] #{RAILS_ROOT}/gems/*/tasks/**/*.rake should now be loaded by tasks/rails.rb, so you get your gem's tasks, like you would for plugins.

---
 railties/lib/tasks/rails.rb |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/railties/lib/tasks/rails.rb b/railties/lib/tasks/rails.rb
index 8c2b7f9..17983b1 100644
--- a/railties/lib/tasks/rails.rb
+++ b/railties/lib/tasks/rails.rb
@@ -5,4 +5,5 @@ Dir["#{File.dirname(__FILE__)}/*.rake"].each { |ext| load ext }
 
 # Load any custom rakefile extensions
 Dir["#{RAILS_ROOT}/vendor/plugins/*/**/tasks/**/*.rake"].sort.each { |ext| load ext }
+Dir["#{RAILS_ROOT}/vendor/gems/*/tasks/**/*.rake"].sort.each { |ext| load ext }
 Dir["#{RAILS_ROOT}/lib/tasks/**/*.rake"].sort.each { |ext| load ext }
-- 
1.5.3.8


From 79b32ccac087516593a0dcba247f034aa0643a86 Mon Sep 17 00:00:00 2001
From: Josh Nichols <josh@technicalpickles.com>
Date: Mon, 30 Jun 2008 14:10:12 -0400
Subject: [PATCH] Tasks in gems should live under rails/tasks.

---
 railties/lib/tasks/rails.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/tasks/rails.rb b/railties/lib/tasks/rails.rb
index 17983b1..888c0a7 100644
--- a/railties/lib/tasks/rails.rb
+++ b/railties/lib/tasks/rails.rb
@@ -5,5 +5,5 @@ Dir["#{File.dirname(__FILE__)}/*.rake"].each { |ext| load ext }
 
 # Load any custom rakefile extensions
 Dir["#{RAILS_ROOT}/vendor/plugins/*/**/tasks/**/*.rake"].sort.each { |ext| load ext }
-Dir["#{RAILS_ROOT}/vendor/gems/*/tasks/**/*.rake"].sort.each { |ext| load ext }
+Dir["#{RAILS_ROOT}/vendor/gems/*/rails/tasks/**/*.rake"].sort.each { |ext| load ext }
 Dir["#{RAILS_ROOT}/lib/tasks/**/*.rake"].sort.each { |ext| load ext }
-- 
1.5.3.8

