From a7c8275947cb7ab223f35fe7ab5cfbc85747fc2c Mon Sep 17 00:00:00 2001
From: Andrew Kaspick <andrew@redlinesoftware.com>
Date: Sat, 14 Aug 2010 13:51:24 -0700
Subject: [PATCH] Support multiple submit buttons

---
 src/rails.js |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/src/rails.js b/src/rails.js
index 1d413f4..6b83e7f 100644
--- a/src/rails.js
+++ b/src/rails.js
@@ -80,7 +80,7 @@
     if (element.tagName.toLowerCase() === 'form') {
       method = element.readAttribute('method') || 'post';
       url    = element.readAttribute('action');
-      params = element.serialize();
+      params = element.serialize({submit: element.retrieve('rails:submit-button')});
     } else {
       method = element.readAttribute('data-method') || 'get';
       url    = element.readAttribute('href');
@@ -142,6 +142,11 @@
     event.stop();
   });
 
+  document.on("click", "form input[type=submit]", function(event, button) {
+    // register the pressed submit button
+    event.findElement('form').store('rails:submit-button', button.name || false);
+  });
+
   document.on("submit", function(event) {
     var form = event.findElement(),
         message = form.readAttribute('data-confirm');
-- 
1.6.5.2

