diff --git a/xml_serializer.rb b/xml_serializer.rb.original
index 4e19a12..d171b74 100644
--- a/xml_serializer.rb
+++ b/xml_serializer.rb.original
@@ -305,7 +305,7 @@ module ActiveRecord #:nodoc:
 
       protected
         def compute_type
-          type = @record.class.serialized_attributes.has_key?(name) ? :yaml : @record.class.columns_hash[name].class
+          type = @record.class.serialized_attributes.has_key?(name) ? :yaml : @record.class.columns_hash[name].type
 
           case type
             when :text
