From e651d24bd5a6aa37023d634d82d0bcb6eeb33720 Mon Sep 17 00:00:00 2001
From: Andrew Jeffries <andrew@nextgendevelopment.co.uk>
Date: Wed, 25 Feb 2009 13:09:18 +0000
Subject: [PATCH] DB::Test namespace for Rails rake tasks set RAILS_ENV to test unless it's already set - Lighthouse ticket #1599

---
 railties/lib/tasks/databases.rake |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index 9588fab..331783e 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -297,6 +297,8 @@ namespace :db do
   end
 
   namespace :test do
+    RAILS_ENV ||= "test"
+    
     desc "Recreate the test database from the current schema.rb"
     task :load => 'db:test:purge' do
       ActiveRecord::Base.establish_connection(ActiveRecord::Base.configurations['test'])
-- 
1.6.1.3

