From 9f9d1845168241e11a4cc392cdba9a288ca078fe Mon Sep 17 00:00:00 2001
From: Steven Soroka <ssoroka78@gmail.com>
Date: Tue, 9 Dec 2008 12:33:33 -0600
Subject: [PATCH] This require environment doesn't seem to make sense.  Every time I see it hit the environment is already loaded and it loads it a second time.

---
 railties/lib/tasks/misc.rake |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/tasks/misc.rake b/railties/lib/tasks/misc.rake
index 411750b..a9716ff 100644
--- a/railties/lib/tasks/misc.rake
+++ b/railties/lib/tasks/misc.rake
@@ -1,6 +1,6 @@
 task :default => :test
 task :environment do
-  require(File.join(RAILS_ROOT, 'config', 'environment'))
+  require(File.join(RAILS_ROOT, 'config', 'environment')) unless defined?(Rails)
 end
 
 task :rails_env do
-- 
1.5.5

