--- actionpack-3.0.3/lib/action_view/helpers/form_helper.rb.orig	2010-12-20 12:06:46.900811099 +0100
+++ actionpack-3.0.3/lib/action_view/helpers/form_helper.rb	2010-12-20 12:08:17.628335871 +0100
@@ -539,6 +539,7 @@
           object = record_or_name_or_array
           object_name = ActiveModel::Naming.singular(object)
         end
+        object_name = options[:as] || object_name
 
         builder = options[:builder] || ActionView::Base.default_form_builder
         capture(builder.new(object_name, object, self, options, block), &block)
