commit 9a4fa967a1644104b86382ffc719c85b277e89cc
Author: Jacob Lewallen <jlewallen@gmail.com>
Date:   Wed Jul 7 10:55:40 2010 +0900

    Set destroyed=true in opt locking's destroy

diff --git a/activerecord/lib/active_record/locking/optimistic.rb b/activerecord/lib/active_record/locking/optimistic.rb
index ceb0902..b6f87a5 100644
--- a/activerecord/lib/active_record/locking/optimistic.rb
+++ b/activerecord/lib/active_record/locking/optimistic.rb
@@ -124,6 +124,7 @@ module ActiveRecord
             end
           end
 
+          @destroyed = true
           freeze
         end
 
diff --git a/activerecord/test/cases/locking_test.rb b/activerecord/test/cases/locking_test.rb
index 66874cd..b87498a 100644
--- a/activerecord/test/cases/locking_test.rb
+++ b/activerecord/test/cases/locking_test.rb
@@ -54,6 +54,7 @@ class OptimisticLockingTest < ActiveRecord::TestCase
 
     assert p1.destroy
     assert_equal true, p1.frozen?
+    assert_equal true, p1.destroyed?
     assert_raises(ActiveRecord::RecordNotFound) { Person.find(1) }
   end
 
