From 3dfb6f3fcb6c149038ad282812711ed677019b5f Mon Sep 17 00:00:00 2001
From: wadestebbings <wade@pobox.com>
Date: Thu, 24 Jun 2010 15:43:26 -0500
Subject: [PATCH] use prefix/suffix when searching for schema_migrations

---
 .../abstract/schema_statements.rb                  |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
index 6157791..6195176 100644
--- a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
+++ b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
@@ -366,7 +366,7 @@ module ActiveRecord
       def initialize_schema_migrations_table
         sm_table = ActiveRecord::Migrator.schema_migrations_table_name
 
-        unless tables.detect { |t| t == sm_table }
+        unless tables.detect { |t| Base.table_name_prefix + t + Base.table_name_suffix == sm_table }
           create_table(sm_table, :id => false) do |schema_migrations_table|
             schema_migrations_table.column :version, :string, :null => false
           end
-- 
1.7.0.3

