From b56a729bd1048bbde628d18c850331ac78cac5f2 Mon Sep 17 00:00:00 2001
From: Todd Persen and John Pignata <pivotal@prince.(none)>
Date: Wed, 16 Dec 2009 12:37:27 -0500
Subject: [PATCH] Always raise first exception thrown in transaction rescue block

---
 .../abstract/database_statements.rb                |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb b/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb
index be89873..b86f598 100644
--- a/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb
+++ b/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb
@@ -145,7 +145,7 @@ module ActiveRecord
               rollback_to_savepoint
             end
           end
-          raise unless database_transaction_rollback.is_a?(ActiveRecord::Rollback)
+          raise database_transaction_rollback unless database_transaction_rollback.is_a?(ActiveRecord::Rollback)
         end
       ensure
         @transaction_joinable = last_transaction_joinable
--
1.6.5.3

