diff --git a/activerecord/lib/active_record/base.rb b/activerecord/lib/active_record/base.rb
index 0087896..6f86bac 100644
--- a/activerecord/lib/active_record/base.rb
+++ b/activerecord/lib/active_record/base.rb
@@ -1166,12 +1166,21 @@ MSG
         #
         #   Article.new.published    # => true
         #   Article.create.published # => true
-        def default_scope(options = {})
-          self.default_scoping << construct_finder_arel(options, default_scoping.pop)
+
+        def default_scope(scope_options = {})
+          self.default_scoping << lambda {
+            construct_finder_arel(
+              scope_options.is_a?(Proc) ? scope_options.call : scope_options)
+          }
         end
 
         def current_scoped_methods #:nodoc:
-          scoped_methods.last
+          method = scoped_methods.last
+          if method.is_a? Proc
+            unscoped(&method)
+          else
+            method
+          end
         end
 
         # Returns the class type of the record using the current module as a prefix. So descendants of
