From b4d5b880a57799c317de67ce78707e4741a31905 Mon Sep 17 00:00:00 2001
From: Chad Woolley <thewoolleyman@gmail.com>
Date: Sat, 29 Aug 2009 23:43:53 -0700
Subject: [PATCH] CI config updates: do not send CI emails unless explicitly enabled, use 'gem update --system', and send emails from an address which can post to the core list

---
 ci/cruise_config.rb |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/ci/cruise_config.rb b/ci/cruise_config.rb
index 439a93d..9c7fa98 100644
--- a/ci/cruise_config.rb
+++ b/ci/cruise_config.rb
@@ -1,9 +1,9 @@
 Project.configure do |project|
   # Send email notifications about broken and fixed builds to core mailing list
-  if Socket.gethostname =~ /ci.rubyonrails.org/
+  if Socket.gethostname =~ /ci.rubyonrails.org/ && ENV['ENABLE_RAILS_CI_EMAILS'] == 'true'
     project.email_notifier.emails = ['rubyonrails-core@googlegroups.com']
   end
 
-  project.build_command = 'sudo update_rubygems && ruby ci/ci_build.rb'
-  project.email_notifier.from = 'thewoolleyman+railsci@gmail.com'
+  project.build_command = 'sudo gem update --system && ruby ci/ci_build.rb'
+  project.email_notifier.from = 'thewoolleyman@gmail.com'
 end
-- 
1.6.3.3

