From 03d35d9a734601000886006cad2f019fb0e2c0bf Mon Sep 17 00:00:00 2001
From: Robert Sosinski <rsosinski@slingmedia.com>
Date: Fri, 26 Nov 2010 20:41:37 -0500
Subject: [PATCH] adding tsvector datatype to postgresql adapter

---
 .../connection_adapters/postgresql_adapter.rb      |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
index ccc5085..cd515c8 100644
--- a/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb
@@ -104,6 +104,9 @@ module ActiveRecord
             # UUID type
             when 'uuid'
               :string
+            # TSVector type
+            when 'tsvector'
+              :string
             # Small and big integer types
             when /^(?:small|big)int$/
               :integer
@@ -203,7 +206,8 @@ module ActiveRecord
         :date        => { :name => "date" },
         :binary      => { :name => "bytea" },
         :boolean     => { :name => "boolean" },
-        :xml         => { :name => "xml" }
+        :xml         => { :name => "xml" },
+        :tsvector    => { :name => "tsvector" }
       }
 
       # Returns 'PostgreSQL' as adapter name for identification purposes.
-- 
1.7.1.1

