From edad02b1d1baabf90be25a3033a2c89d97ed248c Mon Sep 17 00:00:00 2001
From: Anthony Crumley <anthony.crumley@gmail.com>
Date: Sun, 3 May 2009 16:05:53 -0500
Subject: [PATCH] Fixed chicken and egg scenario using tap to add activesupport path.

---
 activerecord/lib/active_record.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/lib/active_record.rb b/activerecord/lib/active_record.rb
index 1982ffc..d2f77f6 100644
--- a/activerecord/lib/active_record.rb
+++ b/activerecord/lib/active_record.rb
@@ -21,7 +21,7 @@
 # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #++
 
-"#{File.dirname(__FILE__)}/../../activesupport/lib".tap do |path|
+["#{File.dirname(__FILE__)}/../../activesupport/lib"].each do |path|
   $:.unshift(path) if File.directory?(path)
 end
 require 'active_support'
-- 
1.5.5.1

