--- actionmailer-2.3.8/lib/action_mailer/part.rb    2010-05-25 14:20:16.656697409 +0200
+++ part.rb     2010-06-24 11:42:09.017614089 +0200
@@ -68,7 +68,7 @@
         # Also don't set filename and name when there is none (like in
         # non-attachment parts)
         if content_disposition == "attachment"
-          ctype_attrs.delete "charset"
+          ctype_attrs.delete "charset" unless real_content_type =~ /(text\/*|xml)/
           part.set_content_type(real_content_type, nil,
             squish("name" => filename).merge(ctype_attrs))
           part.set_content_disposition(content_disposition,
