From 25b511bc58d82d0ac998f23e67e547324b47b17b Mon Sep 17 00:00:00 2001
From: Michel Belleville <michel.belleville@gmail.com>
Date: Sun, 16 May 2010 23:15:27 +0430
Subject: [PATCH] Fixes Rack Dependency in 2-3-stable

---
 actionpack/Rakefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/actionpack/Rakefile b/actionpack/Rakefile
index 23e251f..446c2ee 100644
--- a/actionpack/Rakefile
+++ b/actionpack/Rakefile
@@ -80,7 +80,7 @@ spec = Gem::Specification.new do |s|
   s.requirements << 'none'
 
   s.add_dependency('activesupport', '= 2.3.6' + PKG_BUILD)
-  s.add_dependency('rack', '~> 1.0.0')
+  s.add_dependency('rack', '~> 1.0.1')
 
   s.require_path = 'lib'
   s.autorequire = 'action_controller'
-- 
1.6.5.2

