diff --git a/activerecord/test/cases/associations/has_many_associations_test.rb b/activerecord/test/cases/associations/has_many_associations_test.rb
index 6f0190e..70e05e8 100644
--- a/activerecord/test/cases/associations/has_many_associations_test.rb
+++ b/activerecord/test/cases/associations/has_many_associations_test.rb
@@ -272,6 +272,12 @@ class HasManyAssociationsTest < ActiveRecord::TestCase
     end
   end
 
+  def test_my_creation
+    firm = Firm.new(:name => "BigShot Inc.", :client_name => "SmallShots Inc.")
+    assert firm.clients.first
+    assert_equal "SmallShots Inc.", firm.clients.first.name
+  end
+
   def test_adding_a_mismatch_class
     assert_raises(ActiveRecord::AssociationTypeMismatch) { companies(:first_firm).clients_of_firm << nil }
     assert_raises(ActiveRecord::AssociationTypeMismatch) { companies(:first_firm).clients_of_firm << 1 }
diff --git a/activerecord/test/models/company.rb b/activerecord/test/models/company.rb
index 3d76dfd..af49de5 100755
--- a/activerecord/test/models/company.rb
+++ b/activerecord/test/models/company.rb
@@ -44,6 +44,10 @@ class Firm < Company
 
   has_one :account, :foreign_key => "firm_id", :dependent => :destroy
   has_one :readonly_account, :foreign_key => "firm_id", :class_name => "Account", :readonly => true
+
+  def client_name=(name)
+    clients.build(:name => name)
+  end
 end
 
 class DependentFirm < Company
