From 80b721840cfbc9c1fbe38e6670db1bfa81ae6d08 Mon Sep 17 00:00:00 2001
From: Ryan Kinderman <ryan@kinderman.net>
Date: Mon, 26 May 2008 22:15:57 -0500
Subject: [PATCH] changed load order for Rake task files so that plugin tasks are loaded before application tasks, so that the application can modify tasks provided by plugins

---
 railties/lib/tasks/rails.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/tasks/rails.rb b/railties/lib/tasks/rails.rb
index bfcf5bc..8c2b7f9 100644
--- a/railties/lib/tasks/rails.rb
+++ b/railties/lib/tasks/rails.rb
@@ -4,5 +4,5 @@ $VERBOSE = nil
 Dir["#{File.dirname(__FILE__)}/*.rake"].each { |ext| load ext }
 
 # Load any custom rakefile extensions
-Dir["#{RAILS_ROOT}/lib/tasks/**/*.rake"].sort.each { |ext| load ext }
 Dir["#{RAILS_ROOT}/vendor/plugins/*/**/tasks/**/*.rake"].sort.each { |ext| load ext }
+Dir["#{RAILS_ROOT}/lib/tasks/**/*.rake"].sort.each { |ext| load ext }
-- 
1.5.4.5

