From a597faadd8f36fe46bbf76968fbde7cdcc76eb44 Mon Sep 17 00:00:00 2001
From: Anil Wadghule <anildigital@gmail.com>
Date: Tue, 25 May 2010 12:21:42 +0530
Subject: [PATCH] Add files to .gitignore of rails 2-3-stable branch [#4687 state:resolved]

---
 .gitignore |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/.gitignore b/.gitignore
index d5ad877..970e7f8 100644
--- a/.gitignore
+++ b/.gitignore
@@ -8,6 +8,7 @@ activerecord/doc
 actionpack/doc
 actionmailer/doc
 activesupport/doc
+activesupport/test/fixtures/
 activeresource/pkg
 activerecord/pkg
 actionpack/pkg
@@ -18,3 +19,5 @@ railties/test/500.html
 railties/doc/guides/html/images
 railties/doc/guides/html/stylesheets
 railties/guides/output
+railties/log/
+railties/tmp/
-- 
1.6.6.1

