Index: activerecord/lib/active_record/associations/has_and_belongs_to_many_association.rb
===================================================================
--- activerecord/lib/active_record/associations/has_and_belongs_to_many_association.rb	(revision 2162)
+++ activerecord/lib/active_record/associations/has_and_belongs_to_many_association.rb	(working copy)
@@ -32,7 +32,7 @@
         def construct_find_options!(options)
           options[:joins]      = @join_sql
           options[:readonly]   = finding_with_ambiguous_select?(options[:select] || @reflection.options[:select])
-          options[:select]   ||= (@reflection.options[:select] || '*')
+          options[:select]   ||= @reflection.options[:select]
         end
         
         def count_records
