--- /usr/pkg/lib/ruby/gems/1.8/gems/activesupport-2.0.2/lib/active_support/core_ext/array/conversions.rb-orig	2008-09-25 00:13:09.000000000 +0200
+++ /usr/pkg/lib/ruby/gems/1.8/gems/activesupport-2.0.2/lib/active_support/core_ext/array/conversions.rb	2008-09-25 00:13:19.000000000 +0200
@@ -83,7 +83,7 @@
           else
             xml.tag!(root, options[:skip_types] ? {} : {:type => "array"}) {
               yield xml if block_given?
-              each { |e| e.to_xml(opts.merge!({ :skip_instruct => true })) }
+              each { |e| e.to_xml(opts.merge({ :skip_instruct => true })) }
             }
           end
         end
