From 3f4c64607226286eff926dadc2e646ec43d73fd0 Mon Sep 17 00:00:00 2001
From: Kyle Hargraves <philodespotos@gmail.com>
Date: Sun, 4 May 2008 01:31:36 -0500
Subject: [PATCH] Restore dump_schema_information connection adapter method

---
 .../abstract/schema_statements.rb                  |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
index 6aae556..1854197 100644
--- a/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
+++ b/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb
@@ -295,6 +295,12 @@ module ActiveRecord
       def structure_dump
       end
 
+      def dump_schema_information #:nodoc:
+        sm_table = ActiveRecord::Migrator.schema_migrations_table_name
+        migrated = select_values("SELECT version FROM #{sm_table}")
+        migrated.map { |v| "INSERT INTO #{sm_table} (version) VALUES ('#{v}');" }.join("\n")
+      end
+
       # Should not be called normally, but this operation is non-destructive.
       # The migrations module handles this automatically.
       def initialize_schema_migrations_table
-- 
1.5.4.5

