From 679e05556be6902fb60144f342392f265d2e7cbc Mon Sep 17 00:00:00 2001
From: Nathan Zook <blogger@pierian-spring.net>
Date: Fri, 26 Feb 2010 00:27:51 -0600
Subject: [PATCH] Update document for initializing databases for active_record testing

---
 activerecord/RUNNING_UNIT_TESTS |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/activerecord/RUNNING_UNIT_TESTS b/activerecord/RUNNING_UNIT_TESTS
index 39fc867..716de45 100644
--- a/activerecord/RUNNING_UNIT_TESTS
+++ b/activerecord/RUNNING_UNIT_TESTS
@@ -30,7 +30,7 @@ you can do so with:
 That'll run the base suite using the MySQL-Ruby adapter.  Some tests rely on the schema
 being initialized - you can initialize the schema with:
 
-  rake test_mysql TEST=test/cases/aaa_create_tables_test.rb
+  rake mysql:build_databases
 
 
 
-- 
1.6.0.4

