From dbb8b295589a4b8a9886548a6a4a5eb1932a96c7 Mon Sep 17 00:00:00 2001
From: Santiago Pastorino <santiago@wyeworks.com>
Date: Sun, 17 Jan 2010 04:21:35 -0200
Subject: [PATCH] fixes the failure of config.plugins without :all option tests

---
 railties/lib/rails/application.rb |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/railties/lib/rails/application.rb b/railties/lib/rails/application.rb
index 4d05f81..0e48715 100644
--- a/railties/lib/rails/application.rb
+++ b/railties/lib/rails/application.rb
@@ -93,9 +93,7 @@ module Rails
     def plugins
       @plugins ||= begin
         plugin_names = (config.plugins || [:all]).map { |p| p.to_sym }
-        Railtie.plugins.select { |p|
-          plugin_names.include?(:all) || plugin_names.include?(p.plugin_name)
-        }.map { |p| p.new } + Plugin.all(plugin_names, config.paths.vendor.plugins)
+        Railtie.plugins.map(&:new) + Plugin.all(plugin_names, config.paths.vendor.plugins)
       end
     end
 
-- 
1.6.5

