From 7b2ec1aa356645228c919b13dac96720864aac4f Mon Sep 17 00:00:00 2001
From: Akira Matsuda <ronnie@dio.jp>
Date: Fri, 20 Feb 2009 01:27:43 +0900
Subject: [PATCH] Ruby 1.9 compat: Avoid using the return value of FileUtils.mkdir_p, as it does not return a String but an Array in Ruby 1.9

---
 railties/lib/commands/plugin.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/commands/plugin.rb b/railties/lib/commands/plugin.rb
index 9ff4739..881221d 100644
--- a/railties/lib/commands/plugin.rb
+++ b/railties/lib/commands/plugin.rb
@@ -267,7 +267,7 @@ class Plugin
     
     def install_using_git(options = {})
       root = rails_env.root
-      install_path = mkdir_p "#{root}/vendor/plugins/#{name}"
+      mkdir_p(install_path = "#{root}/vendor/plugins/#{name}")
       Dir.chdir install_path do
         init_cmd = "git init"
         init_cmd += " -q" if options[:quiet] and not $verbose
-- 
1.6.1.2

