From bca4d407fe8325facd573ff971f6868938b39f54 Mon Sep 17 00:00:00 2001
From: Michael Twentyman <20@kluster.com>
Date: Fri, 16 Jan 2009 14:25:50 -0500
Subject: [PATCH] don't load dispatcher if gem dependencies are missing

---
 railties/lib/initializer.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index b57c46e..5c2952c 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -577,7 +577,7 @@ Run `rake gems:install` to install the missing gems.
     end
 
     def prepare_dispatcher
-      return unless configuration.frameworks.include?(:action_controller)
+      return unless configuration.frameworks.include?(:action_controller) && gems_dependencies_loaded
       require 'dispatcher' unless defined?(::Dispatcher)
       Dispatcher.define_dispatcher_callbacks(configuration.cache_classes)
       Dispatcher.new(Rails.logger).send :run_callbacks, :prepare_dispatch
-- 
1.5.6.1.204.g699135

