From 02bf1982205ff5abdbe48c462bf62bfe5eab7a2e Mon Sep 17 00:00:00 2001
From: Steven Soroka <ssoroka78@gmail.com>
Date: Thu, 1 May 2008 10:02:05 -0500
Subject: [PATCH] Ticket #21, resolves db:structure:dump broken by 8a5a9dc


Signed-off-by: Steven Soroka <ssoroka78@gmail.com>
---
 railties/lib/tasks/databases.rake |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/railties/lib/tasks/databases.rake b/railties/lib/tasks/databases.rake
index e39f9ca..a85aafe 100644
--- a/railties/lib/tasks/databases.rake
+++ b/railties/lib/tasks/databases.rake
@@ -256,10 +256,6 @@ namespace :db do
       else
         raise "Task not supported by '#{abcs["test"]["adapter"]}'"
       end
-
-      if ActiveRecord::Base.connection.supports_migrations?
-        File.open("db/#{RAILS_ENV}_structure.sql", "a") { |f| f << ActiveRecord::Base.connection.dump_schema_information }
-      end
     end
   end
 
-- 
1.5.5

