From eaff1c6586d88daaec224c41b21ea1deb658dc5f Mon Sep 17 00:00:00 2001
From: Gaspard Bucher <gaspard@teti.ch>
Date: Fri, 13 Feb 2009 23:20:06 +0100
Subject: [PATCH] Added a configuration option to set the timezone of the mysql connection.

---
 .../connection_adapters/mysql_adapter.rb           |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
index 9300df2..37f9cbe 100644
--- a/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
+++ b/activerecord/lib/active_record/connection_adapters/mysql_adapter.rb
@@ -152,6 +152,7 @@ module ActiveRecord
     # * <tt>:password</tt> - Defaults to nothing.
     # * <tt>:database</tt> - The name of the database. No default, must be provided.
     # * <tt>:encoding</tt> - (Optional) Sets the client encoding by executing "SET NAMES <encoding>" after connection.
+    # * <tt>:timezone</tt> - (Optional) Sets the client timezone by executing "SET TIME_ZONE <timezone>" after connection.
     # * <tt>:reconnect</tt> - Defaults to false (See MySQL documentation: http://dev.mysql.com/doc/refman/5.0/en/auto-reconnect.html).
     # * <tt>:sslca</tt> - Necessary to use MySQL with an SSL connection.
     # * <tt>:sslkey</tt> - Necessary to use MySQL with an SSL connection.
@@ -584,6 +585,9 @@ module ActiveRecord
         def configure_connection
           encoding = @config[:encoding]
           execute("SET NAMES '#{encoding}'") if encoding
+          
+          timezone = @config[:timezone]
+          execute("SET TIME_ZONE = '#{timezone}'") if timezone
 
           # By default, MySQL 'where id is null' selects the last inserted id.
           # Turn this off. http://dev.rubyonrails.org/ticket/6778
-- 
1.6.0

