diff --git a/activerecord/lib/active_record/associations.rb b/activerecord/lib/active_record/associations.rb
index c7cb6eb..ef969b4 100755
--- a/activerecord/lib/active_record/associations.rb
+++ b/activerecord/lib/active_record/associations.rb
@@ -166,11 +166,11 @@ module ActiveRecord
     #
     # Use +has_one+ in the base, and +belongs_to+ in the associated model.
     #
-    #   class Employee < ActiveRecord::Base
-    #     has_one :office
-    #   end
     #   class Office < ActiveRecord::Base
-    #     belongs_to :employee    # foreign key - employee_id
+    #     has_one :employee
+    #   end
+    #   class Employee < ActiveRecord::Base
+    #     belongs_to :office    # foreign key - office_id
     #   end
     #
     # === One-to-many
