diff --git a/app/models/post.rb b/app/models/post.rb
new file mode 100644
index 0000000..4f00777
--- /dev/null
+++ b/app/models/post.rb
@@ -0,0 +1,2 @@
+class Past < ActiveRecord::Base
+end
\ No newline at end of file
diff --git a/app/models/user.rb b/app/models/user.rb
new file mode 100644
index 0000000..a770e88
--- /dev/null
+++ b/app/models/user.rb
@@ -0,0 +1,3 @@
+class User < ActiveRecord::Base
+  has_many :posts, :dependent => :destroy
+end
\ No newline at end of file
diff --git a/app/models/user_observer.rb b/app/models/user_observer.rb
new file mode 100644
index 0000000..6f0c05a
--- /dev/null
+++ b/app/models/user_observer.rb
@@ -0,0 +1,2 @@
+class UserObserver < ActiveRecord::Observer
+end
\ No newline at end of file
diff --git a/config/database.yml b/config/database.yml
index b444b03..b83ad68 100644
--- a/config/database.yml
+++ b/config/database.yml
@@ -1,19 +1,20 @@
-# SQLite version 3.x
-#   gem install sqlite3-ruby (not necessary on OS X Leopard)
 development:
-  adapter: sqlite3
-  database: db/development.sqlite3
-  timeout: 5000
+  adapter: mysql
+  database: ticket63_development
+  username: root
+  password:
+  socket: /tmp/mysql.sock
 
-# Warning: The database defined as 'test' will be erased and
-# re-generated from your development database when you run 'rake'.
-# Do not set this db to the same as development or production.
 test:
-  adapter: sqlite3
-  database: db/test.sqlite3
-  timeout: 5000
+  adapter: mysql
+  database: ticket63_test
+  username: root
+  password:
+  socket: /tmp/mysql.sock
 
 production:
-  adapter: sqlite3
-  database: db/production.sqlite3
-  timeout: 5000
+  adapter: mysql
+  database: ticket63_production
+  username: root
+  password: 
+  socket: /tmp/mysql.sock
diff --git a/config/environment.rb b/config/environment.rb
index 17c17c7..fa86daf 100644
--- a/config/environment.rb
+++ b/config/environment.rb
@@ -63,5 +63,5 @@ Rails::Initializer.run do |config|
   # config.active_record.schema_format = :sql
 
   # Activate observers that should always be running
-  # config.active_record.observers = :cacher, :garbage_collector
+  config.active_record.observers = :user_observer
 end
\ No newline at end of file
