From 91dc12fcbe449c5cd79d20c9b1b6dd52793cf98e Mon Sep 17 00:00:00 2001
From: Tekin Suleyman <tekin@tekin.co.uk>
Date: Sat, 28 Feb 2009 15:23:39 +0000
Subject: [PATCH] Updated actionpack/CHANGELOG to reflect changes to form option helpers

---
 actionpack/CHANGELOG |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/actionpack/CHANGELOG b/actionpack/CHANGELOG
index 809e90f..4d19bf1 100644
--- a/actionpack/CHANGELOG
+++ b/actionpack/CHANGELOG
@@ -1,5 +1,7 @@
 *2.3.1 [RC2] (February ?, 2009)*
 
+* Form option helpers now support disabled option tags and use of lambdas for selecting/disabling option tags from collections #837 [Tekin]
+
 * Added ability to pass in :public => true to fresh_when, stale?, and expires_in to make the request proxy cachable #2095 [Gregg Pollack]
 
 * Fixed that passing a custom form builder would be forwarded to nested fields_for calls #2023 [Eloy Duran/Nate Wiger]
-- 
1.5.5.4

