From 016f2d2d2d4a8cd9a5de73da6a789b2ea3e40c64 Mon Sep 17 00:00:00 2001
From: dmathieu <42@dmathieu.com>
Date: Wed, 16 Dec 2009 16:11:14 +0100
Subject: [PATCH] :vendored_at is deprecated. Using :path

---
 Gemfile |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Gemfile b/Gemfile
index 87bf43b..d979add 100644
--- a/Gemfile
+++ b/Gemfile
@@ -1,9 +1,9 @@
 gem "rake", ">= 0.8.7"
 gem "mocha", ">= 0.9.8"
 
-gem "rails", "3.0.pre", :vendored_at => "railties"
+gem "rails", "3.0.pre", :path => "railties"
 %w(activesupport activemodel actionpack actionmailer activerecord activeresource).each do |lib|
-  gem lib, '3.0.pre', :vendored_at => lib
+  gem lib, '3.0.pre', :path => lib
 end
 
 # AR
-- 
1.6.3.3

