diff --git a/activerecord/lib/active_record/base.rb b/activerecord/lib/active_record/base.rb
index 29c2995..0580db1 100644
--- a/activerecord/lib/active_record/base.rb
+++ b/activerecord/lib/active_record/base.rb
@@ -1892,6 +1892,11 @@ module ActiveRecord #:nodoc:
           with_scope(method_scoping, :overwrite, &block)
         end
 
+        # sentry for ActiveRecord::NamedScope::Scope#with
+        def with(&block)
+          block.call if block
+        end
+
         def subclasses #:nodoc:
           @@subclasses[self] ||= []
           @@subclasses[self] + extra = @@subclasses[self].inject([]) {|list, subclass| list + subclass.subclasses }
diff --git a/activerecord/lib/active_record/named_scope.rb b/activerecord/lib/active_record/named_scope.rb
index d5a1c5f..3808da2 100644
--- a/activerecord/lib/active_record/named_scope.rb
+++ b/activerecord/lib/active_record/named_scope.rb
@@ -148,6 +148,14 @@ module ActiveRecord
         end
       end
 
+      def with(&block)
+        create = proxy_options[:conditions].is_a?(Hash) ?
+          proxy_options[:conditions] : {}
+        with_scope :find => proxy_options, :create => create do
+          proxy_scope.send :with, &block
+        end
+      end
+
       protected
       def proxy_found
         @found || load_found
