From 20a08ccf13ac469e50194a01010240ae7ab91583 Mon Sep 17 00:00:00 2001
From: stevestmartin <kuprishuz@gmail.com>
Date: Tue, 13 Apr 2010 11:54:36 -0400
Subject: [PATCH] throw an exception when csrf_token is not found closes #4084

---
 src/rails.js |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/rails.js b/src/rails.js
index d8ce0d8..1be4449 100644
--- a/src/rails.js
+++ b/src/rails.js
@@ -45,14 +45,15 @@ document.observe("dom:loaded", function() {
       form.appendChild(field);
     }
 
-    if (csrf_param) {
+    if (csrf_param && csrf_token) {
       var param = csrf_param.readAttribute('content'),
           token = csrf_token.readAttribute('content'),
           field = new Element('input', { type: 'hidden', name: param, value: token });
       form.appendChild(field);
+      form.submit();
+    } else {
+      throw "No CSRF token found (is csrf_meta_tag in your layout?).";
     }
-
-    form.submit();
   }
 
   document.observe("click", function(event) {
-- 
1.6.3.3

