From bfde31fa8809532e6f8de9419481d2303fad6b29 Mon Sep 17 00:00:00 2001
From: Dalibor Nasevic <dalibor.nasevic@gmail.com>
Date: Sun, 6 Mar 2011 20:03:06 +0100
Subject: [PATCH] Fixed indentation in actionmailer base_test

---
 actionmailer/test/base_test.rb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/actionmailer/test/base_test.rb b/actionmailer/test/base_test.rb
index 1b793d2..6a7931d 100644
--- a/actionmailer/test/base_test.rb
+++ b/actionmailer/test/base_test.rb
@@ -153,8 +153,8 @@ class BaseTest < ActiveSupport::TestCase
     assert_equal(2, email.parts.length)
     assert_equal("multipart/related", email.mime_type)
     assert_equal("multipart/alternative", email.parts[0].mime_type)
-      assert_equal("text/plain", email.parts[0].parts[0].mime_type)
-      assert_equal("text/html",  email.parts[0].parts[1].mime_type)
+    assert_equal("text/plain", email.parts[0].parts[0].mime_type)
+    assert_equal("text/html",  email.parts[0].parts[1].mime_type)
     assert_equal("logo.png", email.parts[1].filename)
   end
 
-- 
1.7.0.4

