From b886d829bbf2c3d71997aefb65fd4286b2074687 Mon Sep 17 00:00:00 2001
From: Emilio Tagua <miloops@gmail.com>
Date: Mon, 24 Aug 2009 19:54:37 -0300
Subject: [PATCH] timestamps gives a created_at field not created_on.

---
 activerecord/examples/performance.rb |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/activerecord/examples/performance.rb b/activerecord/examples/performance.rb
index 12153b4..6841a04 100755
--- a/activerecord/examples/performance.rb
+++ b/activerecord/examples/performance.rb
@@ -75,13 +75,13 @@ else
   puts 'Inserting 10,000 users and exhibits...'
   10_000.times do
     user = User.create(
-      :created_on => today,
+      :created_at => today,
       :name       => Faker::Name.name,
       :email      => Faker::Internet.email
     )
 
     Exhibit.create(
-      :created_on => today,
+      :created_at => today,
       :name       => Faker::Company.name,
       :user       => user,
       :notes      => notes
-- 
1.6.0.4

