From a2da8cdeeaaa12bfd56f08549407ebb4f8ec7f44 Mon Sep 17 00:00:00 2001
From: Matthew Brindley <matt@mattbrindley.com>
Date: Mon, 12 May 2008 13:20:54 +1200
Subject: [PATCH] Changed ScaffoldGenerator to create public/stylesheets if it doesn't exist.

---
 .../components/scaffold/scaffold_generator.rb      |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/railties/lib/rails_generator/generators/components/scaffold/scaffold_generator.rb b/railties/lib/rails_generator/generators/components/scaffold/scaffold_generator.rb
index 864a0bc..e2902bf 100644
--- a/railties/lib/rails_generator/generators/components/scaffold/scaffold_generator.rb
+++ b/railties/lib/rails_generator/generators/components/scaffold/scaffold_generator.rb
@@ -34,7 +34,7 @@ class ScaffoldGenerator < Rails::Generator::NamedBase
       m.class_collisions(controller_class_path, "#{controller_class_name}Controller", "#{controller_class_name}Helper")
       m.class_collisions(class_path, "#{class_name}")
 
-      # Controller, helper, views, and test directories.
+      # Controller, helper, views, test and stylesheets directories.
       m.directory(File.join('app/models', class_path))
       m.directory(File.join('app/controllers', controller_class_path))
       m.directory(File.join('app/helpers', controller_class_path))
@@ -42,6 +42,7 @@ class ScaffoldGenerator < Rails::Generator::NamedBase
       m.directory(File.join('app/views/layouts', controller_class_path))
       m.directory(File.join('test/functional', controller_class_path))
       m.directory(File.join('test/unit', class_path))
+      m.directory(File.join('public/stylesheets', class_path))
 
       for action in scaffold_views
         m.template(
-- 
1.5.5.1

