diff --git a/actionpack/lib/action_view/base.rb b/actionpack/lib/action_view/base.rb
index b30b565..396cf14 100644
--- a/actionpack/lib/action_view/base.rb
+++ b/actionpack/lib/action_view/base.rb
@@ -170,7 +170,7 @@ module ActionView #:nodoc:
     cattr_accessor :cache_template_loading
     
     def self.cache_template_extensions=(*args)
-      ActiveSupport::Deprecation.warn("config.action_view.cache_template_extensions option has been deprecated and has no affect. " <<
+      ActiveSupport::Deprecation.warn("config.action_view.cache_template_extensions option has been deprecated and has no effect. " <<
                                        "Please remove it from your config files.", caller)
     end
 
