From bc49ac93427814b2ce252c0e98adaedffc4d218e Mon Sep 17 00:00:00 2001
From: Mike Gunderloy <MikeG1@larkfarm.com>
Date: Sat, 10 Jan 2009 08:43:29 -0600
Subject: [PATCH] Make ActioMailer quoting test play nice with Ruby 1.9

---
 actionmailer/test/quoting_test.rb |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/actionmailer/test/quoting_test.rb b/actionmailer/test/quoting_test.rb
index 2650efc..2fee137 100644
--- a/actionmailer/test/quoting_test.rb
+++ b/actionmailer/test/quoting_test.rb
@@ -48,8 +48,10 @@ class QuotingTest < Test::Unit::TestCase
 
     result = execute_in_sandbox(<<-CODE)
       $:.unshift(File.dirname(__FILE__) + "/../lib/")
-      $KCODE = 'u'
-      require 'jcode'
+      if RUBY_VERSION < '1.9'
+        $KCODE = 'u'
+        require 'jcode'
+      end
       require 'action_mailer/quoting'
       include ActionMailer::Quoting
       quoted_printable(#{original.inspect}, "UTF-8")
-- 
1.5.4.3

