From 5a6bc1e0c59d3f8548d2fd13cf0245f0733a6fe4 Mon Sep 17 00:00:00 2001
From: Matthew Rudy Jacobs <matthewrudyjacobs@gmail.com>
Date: Mon, 5 Jan 2009 01:34:26 +0000
Subject: [PATCH 1/2] when we run rake rails:freeze:edge update the value of the "REVISION" file rather than creating a new file "REVISION_ce706..." each time

---
 railties/lib/tasks/framework.rake |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/railties/lib/tasks/framework.rake b/railties/lib/tasks/framework.rake
index 191c936..5dcdace 100644
--- a/railties/lib/tasks/framework.rake
+++ b/railties/lib/tasks/framework.rake
@@ -64,7 +64,10 @@ namespace :rails do
           rm_f goner
         end
 
-        touch "rails/REVISION_#{latest_revision}"
+        puts "Frozen to git revision #{latest_revision}"
+        File.open('rails/REVISION', 'w') do |revision|
+          revision.puts latest_revision
+        end
       end
 
       puts 'Updating current scripts, javascripts, and configuration settings'
-- 
1.7.0.6


From 47bb137e30c71dc872c954dac0439fce1e8cfd29 Mon Sep 17 00:00:00 2001
From: Luke Chadwick <luke.a.chadwick@gmail.com>
Date: Fri, 7 May 2010 21:03:45 +1000
Subject: [PATCH 2/2] REVISION file is also used by freeze_edge_version in railties/builtin/rails_info/rails/info.rb

---
 railties/builtin/rails_info/rails/info.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/builtin/rails_info/rails/info.rb b/railties/builtin/rails_info/rails/info.rb
index 8c858d2..e18fcbb 100644
--- a/railties/builtin/rails_info/rails/info.rb
+++ b/railties/builtin/rails_info/rails/info.rb
@@ -38,7 +38,7 @@ module Rails
       def freeze_edge_version
         if File.exist?(rails_vendor_root)
           begin
-            Dir[File.join(rails_vendor_root, 'REVISION_*')].first.scan(/_(\d+)$/).first.first
+            File.readlines(File.join(rails_vendor_root,'REVISION')).first.strip
           rescue
             Dir[File.join(rails_vendor_root, 'TAG_*')].first.scan(/_(.+)$/).first.first rescue 'unknown'
           end
-- 
1.7.0.6

