From aeaa21ab878c44c688d4c4f79c85ac6487c6a57a Mon Sep 17 00:00:00 2001
From: raggi <jftucker@gmail.com>
Date: Sun, 12 Oct 2008 14:29:20 +0100
Subject: [PATCH] Added manage_gems deprecation fix as done in rake

---
 railties/lib/tasks/framework.rake |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/tasks/framework.rake b/railties/lib/tasks/framework.rake
index 66ab78c..4f5d0a8 100644
--- a/railties/lib/tasks/framework.rake
+++ b/railties/lib/tasks/framework.rake
@@ -5,7 +5,7 @@ namespace :rails do
       deps = %w(actionpack activerecord actionmailer activesupport activeresource)
       require 'rubygems'
       require 'rubygems/gem_runner'
-      Gem.manage_gems
+      Gem.manage_gems if Gem::RubyGemsVersion < '1.2.0'
 
       rails = (version = ENV['VERSION']) ?
         Gem.cache.find_name('rails', "= #{version}").first :
-- 
1.5.6.rc3.7.g336d0.dirty

