From 863b190738e83c15777ee670cfbbe316243a8130 Mon Sep 17 00:00:00 2001
From: joshsmoore <joshsmoore@gmail.com>
Date: Mon, 21 Dec 2009 14:05:05 +0800
Subject: [PATCH] Patched the gems_generators_paths so that it would check for generators gems included with bundler.

---
 railties/lib/rails/generators.rb |   14 +++++++++++---
 1 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/railties/lib/rails/generators.rb b/railties/lib/rails/generators.rb
index 0e66c9f..ccd67ee 100644
--- a/railties/lib/rails/generators.rb
+++ b/railties/lib/rails/generators.rb
@@ -88,10 +88,18 @@ module Rails
     end
 
     def self.gems_generators_paths #:nodoc:
-      return [] unless defined?(Gem) && Gem.respond_to?(:loaded_specs)
-      Gem.loaded_specs.inject([]) do |paths, (name, spec)|
-        paths += Dir[File.join(spec.full_gem_path, "lib/{generators,rails_generators}")]
+      paths = []
+      #puts defined?(Gem) && Gem.respond_to?(:loaded_specs)
+      if defined?(Gem) && Gem.respond_to?(:loaded_specs)
+        Gem.loaded_specs.inject([]) do |paths, (name, spec)|
+          paths += Dir[File.join(spec.full_gem_path, "lib/{generators,rails_generators}")]
+        end
+      elsif !Gem.path.nil?
+        Dir.new("#{Gem.path}/gems").each do |dir|
+          paths += Dir[File.join(Gem.path, 'gems', dir, "{lib/generators,rails_generators}")]
+        end
       end
+      return paths
     end
 
     def self.plugins_generators_paths #:nodoc:
-- 
1.6.2.4

