diff -Naur activerecord-2.1.0.orig/lib/active_record/connection_adapters/postgresql_adapter.rb activerecord-2.1.0/lib/active_record/connection_adapters/postgresql_adapter.rb
--- activerecord-2.1.0.orig/lib/active_record/connection_adapters/postgresql_adapter.rb	2008-06-17 11:18:02.000000000 +0200
+++ activerecord-2.1.0/lib/active_record/connection_adapters/postgresql_adapter.rb	2008-06-17 11:10:49.000000000 +0200
@@ -514,7 +514,14 @@
       # Example:
       #   drop_database 'matt_development'
       def drop_database(name) #:nodoc:
-        execute "DROP DATABASE IF EXISTS #{name}"
+         if postgresql_version >= 80200
+            execute "DROP DATABASE IF EXISTS #{quote_table_name(name)}"
+          else
+            begin
+              execute "DROP DATABASE #{quote_table_name(name)}"
+            rescue ActiveRecord::StatementInvalid
+            end
+          end
       end
 
 
