diff --git a/actionpack/lib/action_dispatch/middleware/session/abstract_store.rb b/actionpack/lib/action_dispatch/middleware/session/abstract_store.rb
index 03761b1..a876863 100644
--- a/actionpack/lib/action_dispatch/middleware/session/abstract_store.rb
+++ b/actionpack/lib/action_dispatch/middleware/session/abstract_store.rb
@@ -45,6 +45,7 @@ module ActionDispatch
             ActiveSupport::Deprecation.warn('use replace instead', caller)
             replace({})
           else
+            load! unless @loaded
             super(hash.stringify_keys)
           end
         end
@@ -54,6 +55,7 @@ module ActionDispatch
             ActiveSupport::Deprecation.warn('use clear instead', caller)
             clear
           else
+            load! unless @loaded
             super(key.to_s)
           end
         end
