From a9ffb5052cfb6781e89f259758ab4583d584bd76 Mon Sep 17 00:00:00 2001
From: Piotr Sarnacki <drogus@gmail.com>
Date: Tue, 7 Sep 2010 22:59:09 +0200
Subject: [PATCH 4/4] Set default asset_path only for engine

---
 railties/lib/rails/engine.rb |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/railties/lib/rails/engine.rb b/railties/lib/rails/engine.rb
index 10df9b3..bbf0d64 100644
--- a/railties/lib/rails/engine.rb
+++ b/railties/lib/rails/engine.rb
@@ -429,7 +429,7 @@ module Rails
     end
 
     initializer :append_asset_paths do
-      config.asset_path ||= "/#{engine_name}%s"
+      config.asset_path ||= "/#{engine_name}%s" unless self.kind_of?(Rails::Application)
 
       public_path = config.paths.public.to_a.first
       if config.compiled_asset_path && File.exist?(public_path)
-- 
1.7.2

