From 05bc21cf6ed77e1248092f9fef6996bc3cd324c7 Mon Sep 17 00:00:00 2001
From: Matt Jones <al2o3cr@gmail.com>
Date: Wed, 6 May 2009 19:35:54 -0500
Subject: [PATCH] Remove stray call to gems:unpack in gems:build:force

---
 railties/lib/tasks/gems.rake |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/railties/lib/tasks/gems.rake b/railties/lib/tasks/gems.rake
index efadb1d..7cf7061 100644
--- a/railties/lib/tasks/gems.rake
+++ b/railties/lib/tasks/gems.rake
@@ -27,8 +27,7 @@ namespace :gems do
     desc "Force the build of all gems"
     task :force do
       $gems_build_rake_task = true
-      Rake::Task['gems:unpack'].invoke
-      current_gems.each { |gem| gem.build(:force => true) }
+      frozen_gems.each { |gem| gem.build(:force => true) }
     end
   end
 
-- 
1.5.3.1

